// react-image-displayer/dist/BaseSpinner-NHLIzqRl.js
import { jsxs as tr, jsx as Nt } from "react/jsx-runtime";
import { createContext as pt, useLayoutEffect as er, useEffect as Ee, useContext as F, useRef as ge, useInsertionEffect as nr, useCallback as As, useMemo as zt, forwardRef as sr, Fragment as Le, createElement as ir, useId as rr, Component as or } from "react";
import { r as ar } from "./index-C47FX26S.js";
const Fu = (t, e) => {
  if (e)
    return e;
  if (!t || t[0] === "none")
    return {};
  const n = { duration: t[1] };
  return t[0] === "scale" ? {
    initial: { opacity: 0.5, scale: 0.5 },
    animate: { opacity: 1, scale: 1 },
    transition: n
  } : t[0] === "fullscale" ? {
    initial: { opacity: 0, scale: 0 },
    animate: { opacity: 1, scale: 1 },
    transition: n
  } : t[0] === "fade" ? {
    initial: { opacity: 0 },
    animate: { opacity: 1 },
    transition: n
  } : t[0] === "blur" ? {
    initial: { opacity: 0, filter: "blur(10px)" },
    animate: { opacity: 1, filter: "blur(0px)" },
    transition: n
  } : t[0] === "left" ? {
    initial: { x: "-10%" },
    animate: { x: 0 },
    transition: n
  } : t[0] === "right" ? {
    initial: { x: "10%" },
    animate: { x: 0 },
    transition: n
  } : t[0] === "top" ? {
    initial: { y: "-10%" },
    animate: { y: 0 },
    transition: n
  } : t[0] === "bottom" ? {
    initial: { y: "10%" },
    animate: { y: 0 },
    transition: n
  } : t[0] === "fullleft" ? {
    initial: { x: "-100%" },
    animate: { x: 0 },
    transition: n
  } : t[0] === "fullright" ? {
    initial: { x: "100%" },
    animate: { x: 0 },
    transition: n
  } : t[0] === "fulltop" ? {
    initial: { y: "-100%" },
    animate: { y: 0 },
    transition: n
  } : t[0] === "fullbottom" ? {
    initial: { y: "100%" },
    animate: { y: 0 },
    transition: n
  } : {};
}, Vs = pt({
  transformPagePoint: (t) => t,
  isStatic: !1,
  reducedMotion: "never"
}), Ht = pt({}), Fe = pt(null), Be = typeof window < "u", lr = Be ? er : Ee, Cs = pt({ strict: !1 }), Yt = (t) => t.replace(/([a-z])([A-Z])/gu, "$1-$2").toLowerCase(), ur = "framerAppearId", Ds = "data-" + Yt(ur), cr = {
  skipAnimations: !1,
  useManualTiming: !1
};
function hr(t) {
  let e = /* @__PURE__ */ new Set(), n = /* @__PURE__ */ new Set(), s = !1, i = !1;
  const o = /* @__PURE__ */ new WeakSet();
  let r = {
    delta: 0,
    timestamp: 0,
    isProcessing: !1
  };
  function a(u) {
    o.has(u) && (l.schedule(u), t()), u(r);
  }
  const l = {
    /**
     * Schedule a process to run on the next frame.
     */
    schedule: (u, c = !1, h = !1) => {
      const d = h && s ? e : n;
      return c && o.add(u), d.has(u) || d.add(u), u;
    },
    /**
     * Cancel the provided callback from running on the next frame.
     */
    cancel: (u) => {
      n.delete(u), o.delete(u);
    },
    /**
     * Execute all schedule callbacks.
     */
    process: (u) => {
      if (r = u, s) {
        i = !0;
        return;
      }
      s = !0, [e, n] = [n, e], n.clear(), e.forEach(a), s = !1, i && (i = !1, l.process(u));
    }
  };
  return l;
}
const Ft = [
  "read",
  // Read
  "resolveKeyframes",
  // Write/Read/Write/Read
  "update",
  // Compute
  "preRender",
  // Compute
  "render",
  // Write
  "postRender"
  // Compute
], fr = 40;
function Ms(t, e) {
  let n = !1, s = !0;
  const i = {
    delta: 0,
    timestamp: 0,
    isProcessing: !1
  }, o = () => n = !0, r = Ft.reduce((g, v) => (g[v] = hr(o), g), {}), { read: a, resolveKeyframes: l, update: u, preRender: c, render: h, postRender: f } = r, d = () => {
    const g = performance.now();
    n = !1, i.delta = s ? 1e3 / 60 : Math.max(Math.min(g - i.timestamp, fr), 1), i.timestamp = g, i.isProcessing = !0, a.process(i), l.process(i), u.process(i), c.process(i), h.process(i), f.process(i), i.isProcessing = !1, n && e && (s = !1, t(d));
  }, p = () => {
    n = !0, s = !0, i.isProcessing || t(d);
  };
  return { schedule: Ft.reduce((g, v) => {
    const y = r[v];
    return g[v] = (P, S = !1, C = !1) => (n || p(), y.schedule(P, S, C)), g;
  }, {}), cancel: (g) => {
    for (let v = 0; v < Ft.length; v++)
      r[Ft[v]].cancel(g);
  }, state: i, steps: r };
}
const { schedule: ke, cancel: Bu } = Ms(queueMicrotask, !1);
function at(t) {
  return t && typeof t == "object" && Object.prototype.hasOwnProperty.call(t, "current");
}
const Rs = pt({});
let un = !1;
function dr(t, e, n, s, i) {
  const { visualElement: o } = F(Ht), r = F(Cs), a = F(Fe), l = F(Vs).reducedMotion, u = ge();
  s = s || r.renderer, !u.current && s && (u.current = s(t, {
    visualState: e,
    parent: o,
    props: n,
    presenceContext: a,
    blockInitialAnimation: a ? a.initial === !1 : !1,
    reducedMotionConfig: l
  }));
  const c = u.current, h = F(Rs);
  c && !c.projection && i && (c.type === "html" || c.type === "svg") && mr(u.current, n, i, h), nr(() => {
    c && c.update(n, a);
  });
  const f = ge(!!(n[Ds] && !window.HandoffComplete));
  return lr(() => {
    c && (c.updateFeatures(), ke.render(c.render), f.current && c.animationState && c.animationState.animateChanges());
  }), Ee(() => {
    c && (!f.current && c.animationState && c.animationState.animateChanges(), f.current && (f.current = !1, un || (un = !0, queueMicrotask(pr))));
  }), c;
}
function pr() {
  window.HandoffComplete = !0;
}
function mr(t, e, n, s) {
  const { layoutId: i, layout: o, drag: r, dragConstraints: a, layoutScroll: l, layoutRoot: u } = e;
  t.projection = new n(t.latestValues, e["data-framer-portal-id"] ? void 0 : Es(t.parent)), t.projection.setOptions({
    layoutId: i,
    layout: o,
    alwaysMeasureLayout: !!r || a && at(a),
    visualElement: t,
    /**
     * TODO: Update options in an effect. This could be tricky as it'll be too late
     * to update by the time layout animations run.
     * We also need to fix this safeToRemove by linking it up to the one returned by usePresence,
     * ensuring it gets called if there's no potential layout animations.
     *
     */
    animationType: typeof o == "string" ? o : "both",
    initialPromotionConfig: s,
    layoutScroll: l,
    layoutRoot: u
  });
}
function Es(t) {
  if (t)
    return t.options.allowProjection !== !1 ? t.projection : Es(t.parent);
}
function gr(t, e, n) {
  return As(
    (s) => {
      s && t.mount && t.mount(s), e && (s ? e.mount(s) : e.unmount()), n && (typeof n == "function" ? n(s) : at(n) && (n.current = s));
    },
    /**
     * Only pass a new ref callback to React if we've received a visual element
     * factory. Otherwise we'll be mounting/remounting every time externalRef
     * or other dependencies change.
     */
    [e]
  );
}
function wt(t) {
  return typeof t == "string" || Array.isArray(t);
}
function At(t) {
  return t !== null && typeof t == "object" && typeof t.start == "function";
}
const Oe = [
  "animate",
  "whileInView",
  "whileFocus",
  "whileHover",
  "whileTap",
  "whileDrag",
  "exit"
], je = ["initial", ...Oe];
function Xt(t) {
  return At(t.animate) || je.some((e) => wt(t[e]));
}
function Ls(t) {
  return !!(Xt(t) || t.variants);
}
function yr(t, e) {
  if (Xt(t)) {
    const { initial: n, animate: s } = t;
    return {
      initial: n === !1 || wt(n) ? n : void 0,
      animate: wt(s) ? s : void 0
    };
  }
  return t.inherit !== !1 ? e : {};
}
function vr(t) {
  const { initial: e, animate: n } = yr(t, F(Ht));
  return zt(() => ({ initial: e, animate: n }), [cn(e), cn(n)]);
}
function cn(t) {
  return Array.isArray(t) ? t.join(" ") : t;
}
const hn = {
  animation: [
    "animate",
    "variants",
    "whileHover",
    "whileTap",
    "exit",
    "whileInView",
    "whileFocus",
    "whileDrag"
  ],
  exit: ["exit"],
  drag: ["drag", "dragControls"],
  focus: ["whileFocus"],
  hover: ["whileHover", "onHoverStart", "onHoverEnd"],
  tap: ["whileTap", "onTap", "onTapStart", "onTapCancel"],
  pan: ["onPan", "onPanStart", "onPanSessionStart", "onPanEnd"],
  inView: ["whileInView", "onViewportEnter", "onViewportLeave"],
  layout: ["layout", "layoutId"]
}, ft = {};
for (const t in hn)
  ft[t] = {
    isEnabled: (e) => hn[t].some((n) => !!e[n])
  };
function xr(t) {
  for (const e in t)
    ft[e] = {
      ...ft[e],
      ...t[e]
    };
}
const Fs = pt({}), Tr = Symbol.for("motionComponentSymbol"), M = (t) => t;
let mt = M, q = M;
process.env.NODE_ENV !== "production" && (mt = (t, e) => {
  !t && typeof console < "u" && console.warn(e);
}, q = (t, e) => {
  if (!t)
    throw new Error(e);
});
function Pr({ preloadedFeatures: t, createVisualElement: e, useRender: n, useVisualState: s, Component: i }) {
  t && xr(t);
  function o(a, l) {
    let u;
    const c = {
      ...F(Vs),
      ...a,
      layoutId: Sr(a)
    }, { isStatic: h } = c, f = vr(a), d = s(a, h);
    if (!h && Be) {
      br(c, t);
      const p = wr(c);
      u = p.MeasureLayout, f.visualElement = dr(i, d, c, e, p.ProjectionNode);
    }
    return tr(Ht.Provider, { value: f, children: [u && f.visualElement ? Nt(u, { visualElement: f.visualElement, ...c }) : null, n(i, a, gr(d, f.visualElement, l), d, h, f.visualElement)] });
  }
  const r = sr(o);
  return r[Tr] = i, r;
}
function Sr({ layoutId: t }) {
  const e = F(Fs).id;
  return e && t !== void 0 ? e + "-" + t : t;
}
function br(t, e) {
  const n = F(Cs).strict;
  if (process.env.NODE_ENV !== "production" && e && n) {
    const s = "You have rendered a `motion` component within a `LazyMotion` component. This will break tree shaking. Import and render a `m` component instead.";
    t.ignoreStrict ? mt(!1, s) : q(!1, s);
  }
}
function wr(t) {
  const { drag: e, layout: n } = ft;
  if (!e && !n)
    return {};
  const s = { ...e, ...n };
  return {
    MeasureLayout: e != null && e.isEnabled(t) || n != null && n.isEnabled(t) ? s.MeasureLayout : void 0,
    ProjectionNode: s.ProjectionNode
  };
}
function Ar(t) {
  function e(s, i = {}) {
    return Pr(t(s, i));
  }
  if (typeof Proxy > "u")
    return e;
  const n = /* @__PURE__ */ new Map();
  return new Proxy(e, {
    /**
     * Called when `motion` is referenced with a prop: `motion.div`, `motion.input` etc.
     * The prop name is passed through as `key` and we can use that to generate a `motion`
     * DOM component with that name.
     */
    get: (s, i) => (n.has(i) || n.set(i, e(i)), n.get(i))
  });
}
const Vr = [
  "animate",
  "circle",
  "defs",
  "desc",
  "ellipse",
  "g",
  "image",
  "line",
  "filter",
  "marker",
  "mask",
  "metadata",
  "path",
  "pattern",
  "polygon",
  "polyline",
  "rect",
  "stop",
  "switch",
  "symbol",
  "svg",
  "text",
  "tspan",
  "use",
  "view"
];
function Ie(t) {
  return (
    /**
     * If it's not a string, it's a custom React component. Currently we only support
     * HTML custom React components.
     */
    typeof t != "string" || /**
     * If it contains a dash, the element is a custom HTML webcomponent.
     */
    t.includes("-") ? !1 : (
      /**
       * If it's in our list of lowercase SVG tags, it's an SVG component
       */
      !!(Vr.indexOf(t) > -1 || /**
       * If it contains a capital letter, it's an SVG component
       */
      /[A-Z]/u.test(t))
    )
  );
}
const Ut = {};
function Cr(t) {
  Object.assign(Ut, t);
}
const Mt = [
  "transformPerspective",
  "x",
  "y",
  "z",
  "translateX",
  "translateY",
  "translateZ",
  "scale",
  "scaleX",
  "scaleY",
  "rotate",
  "rotateX",
  "rotateY",
  "rotateZ",
  "skew",
  "skewX",
  "skewY"
], Q = new Set(Mt);
function Bs(t, { layout: e, layoutId: n }) {
  return Q.has(t) || t.startsWith("origin") || (e || n !== void 0) && (!!Ut[t] || t === "opacity");
}
const E = (t) => !!(t && t.getVelocity), ks = (t, e) => e && typeof t == "number" ? e.transform(t) : t, Z = (t, e, n) => n > e ? e : n < t ? t : n, gt = {
  test: (t) => typeof t == "number",
  parse: parseFloat,
  transform: (t) => t
}, Tt = {
  ...gt,
  transform: (t) => Z(0, 1, t)
}, Bt = {
  ...gt,
  default: 1
}, Pt = (t) => Math.round(t * 1e5) / 1e5, Ne = /-?(?:\d+(?:\.\d+)?|\.\d+)/gu, Dr = /(?:#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\))/giu, Mr = /^(?:#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\))$/iu;
function Rt(t) {
  return typeof t == "string";
}
function Rr(t) {
  return t == null;
}
const Et = (t) => ({
  test: (e) => Rt(e) && e.endsWith(t) && e.split(" ").length === 1,
  parse: parseFloat,
  transform: (e) => `${e}${t}`
}), Y = Et("deg"), N = Et("%"), T = Et("px"), Er = Et("vh"), Lr = Et("vw"), fn = {
  ...N,
  parse: (t) => N.parse(t) / 100,
  transform: (t) => N.transform(t * 100)
}, dn = {
  ...gt,
  transform: Math.round
}, Ue = {
  // Border props
  borderWidth: T,
  borderTopWidth: T,
  borderRightWidth: T,
  borderBottomWidth: T,
  borderLeftWidth: T,
  borderRadius: T,
  radius: T,
  borderTopLeftRadius: T,
  borderTopRightRadius: T,
  borderBottomRightRadius: T,
  borderBottomLeftRadius: T,
  // Positioning props
  width: T,
  maxWidth: T,
  height: T,
  maxHeight: T,
  size: T,
  top: T,
  right: T,
  bottom: T,
  left: T,
  // Spacing props
  padding: T,
  paddingTop: T,
  paddingRight: T,
  paddingBottom: T,
  paddingLeft: T,
  margin: T,
  marginTop: T,
  marginRight: T,
  marginBottom: T,
  marginLeft: T,
  // Transform props
  rotate: Y,
  rotateX: Y,
  rotateY: Y,
  rotateZ: Y,
  scale: Bt,
  scaleX: Bt,
  scaleY: Bt,
  scaleZ: Bt,
  skew: Y,
  skewX: Y,
  skewY: Y,
  distance: T,
  translateX: T,
  translateY: T,
  translateZ: T,
  x: T,
  y: T,
  z: T,
  perspective: T,
  transformPerspective: T,
  opacity: Tt,
  originX: fn,
  originY: fn,
  originZ: T,
  // Misc
  zIndex: dn,
  backgroundPositionX: T,
  backgroundPositionY: T,
  // SVG
  fillOpacity: Tt,
  strokeOpacity: Tt,
  numOctaves: dn
}, Fr = {
  x: "translateX",
  y: "translateY",
  z: "translateZ",
  transformPerspective: "perspective"
}, Br = Mt.length;
function kr(t, e, n) {
  let s = "", i = !0;
  for (let o = 0; o < Br; o++) {
    const r = Mt[o], a = t[r];
    if (a === void 0)
      continue;
    let l = !0;
    if (typeof a == "number" ? l = a === (r.startsWith("scale") ? 1 : 0) : l = parseFloat(a) === 0, l || (i = !1), !l || n) {
      const u = Fr[r] || r, c = ks(a, Ue[r]);
      s += `${u}(${c}) `, n && (e[r] = c);
    }
  }
  return s = s.trim(), n ? s = n(e, i ? "" : s) : i && (s = "none"), s;
}
const Os = (t) => (e) => typeof e == "string" && e.startsWith(t), js = Os("--"), Or = Os("var(--"), _e = (t) => Or(t) ? jr.test(t.split("/*")[0].trim()) : !1, jr = /var\(--(?:[\w-]+\s*|[\w-]+\s*,(?:\s*[^)(\s]|\s*\((?:[^)(]|\([^)(]*\))*\))+\s*)\)$/iu;
function We(t, e, n) {
  const { style: s, vars: i, transformOrigin: o } = t;
  let r = !1, a = !1;
  for (const l in e) {
    const u = e[l];
    if (Q.has(l)) {
      r = !0;
      continue;
    } else if (js(l)) {
      i[l] = u;
      continue;
    } else {
      const c = ks(u, Ue[l]);
      l.startsWith("origin") ? (a = !0, o[l] = c) : s[l] = c;
    }
  }
  if (e.transform || (r || n ? s.transform = kr(e, t.transform, n) : s.transform && (s.transform = "none")), a) {
    const { originX: l = "50%", originY: u = "50%", originZ: c = 0 } = o;
    s.transformOrigin = `${l} ${u} ${c}`;
  }
}
const Ke = () => ({
  style: {},
  transform: {},
  transformOrigin: {},
  vars: {}
});
function Is(t, e, n) {
  for (const s in e)
    !E(e[s]) && !Bs(s, n) && (t[s] = e[s]);
}
function Ir({ transformTemplate: t }, e) {
  return zt(() => {
    const n = Ke();
    return We(n, e, t), Object.assign({}, n.vars, n.style);
  }, [e]);
}
function Nr(t, e) {
  const n = t.style || {}, s = {};
  return Is(s, n, t), Object.assign(s, Ir(t, e)), s;
}
function Ur(t, e) {
  const n = {}, s = Nr(t, e);
  return t.drag && t.dragListener !== !1 && (n.draggable = !1, s.userSelect = s.WebkitUserSelect = s.WebkitTouchCallout = "none", s.touchAction = t.drag === !0 ? "none" : `pan-${t.drag === "x" ? "y" : "x"}`), t.tabIndex === void 0 && (t.onTap || t.onTapStart || t.whileTap) && (n.tabIndex = 0), n.style = s, n;
}
const _r = /* @__PURE__ */ new Set([
  "animate",
  "exit",
  "variants",
  "initial",
  "style",
  "values",
  "variants",
  "transition",
  "transformTemplate",
  "custom",
  "inherit",
  "onBeforeLayoutMeasure",
  "onAnimationStart",
  "onAnimationComplete",
  "onUpdate",
  "onDragStart",
  "onDrag",
  "onDragEnd",
  "onMeasureDragConstraints",
  "onDirectionLock",
  "onDragTransitionEnd",
  "_dragX",
  "_dragY",
  "onHoverStart",
  "onHoverEnd",
  "onViewportEnter",
  "onViewportLeave",
  "globalTapTarget",
  "ignoreStrict",
  "viewport"
]);
function _t(t) {
  return t.startsWith("while") || t.startsWith("drag") && t !== "draggable" || t.startsWith("layout") || t.startsWith("onTap") || t.startsWith("onPan") || t.startsWith("onLayout") || _r.has(t);
}
let Ns = (t) => !_t(t);
function Wr(t) {
  t && (Ns = (e) => e.startsWith("on") ? !_t(e) : t(e));
}
try {
  Wr(require("@emotion/is-prop-valid").default);
} catch {
}
function Kr(t, e, n) {
  const s = {};
  for (const i in t)
    i === "values" && typeof t.values == "object" || (Ns(i) || n === !0 && _t(i) || !e && !_t(i) || // If trying to use native HTML drag events, forward drag listeners
    t.draggable && i.startsWith("onDrag")) && (s[i] = t[i]);
  return s;
}
function pn(t, e, n) {
  return typeof t == "string" ? t : T.transform(e + n * t);
}
function $r(t, e, n) {
  const s = pn(e, t.x, t.width), i = pn(n, t.y, t.height);
  return `${s} ${i}`;
}
const Gr = {
  offset: "stroke-dashoffset",
  array: "stroke-dasharray"
}, zr = {
  offset: "strokeDashoffset",
  array: "strokeDasharray"
};
function Hr(t, e, n = 1, s = 0, i = !0) {
  t.pathLength = 1;
  const o = i ? Gr : zr;
  t[o.offset] = T.transform(-s);
  const r = T.transform(e), a = T.transform(n);
  t[o.array] = `${r} ${a}`;
}
function $e(t, {
  attrX: e,
  attrY: n,
  attrScale: s,
  originX: i,
  originY: o,
  pathLength: r,
  pathSpacing: a = 1,
  pathOffset: l = 0,
  // This is object creation, which we try to avoid per-frame.
  ...u
}, c, h) {
  if (We(t, u, h), c) {
    t.style.viewBox && (t.attrs.viewBox = t.style.viewBox);
    return;
  }
  t.attrs = t.style, t.style = {};
  const { attrs: f, style: d, dimensions: p } = t;
  f.transform && (p && (d.transform = f.transform), delete f.transform), p && (i !== void 0 || o !== void 0 || d.transform) && (d.transformOrigin = $r(p, i !== void 0 ? i : 0.5, o !== void 0 ? o : 0.5)), e !== void 0 && (f.x = e), n !== void 0 && (f.y = n), s !== void 0 && (f.scale = s), r !== void 0 && Hr(f, r, a, l, !1);
}
const Us = () => ({
  ...Ke(),
  attrs: {}
}), Ge = (t) => typeof t == "string" && t.toLowerCase() === "svg";
function Yr(t, e, n, s) {
  const i = zt(() => {
    const o = Us();
    return $e(o, e, Ge(s), t.transformTemplate), {
      ...o.attrs,
      style: { ...o.style }
    };
  }, [e]);
  if (t.style) {
    const o = {};
    Is(o, t.style, t), i.style = { ...o, ...i.style };
  }
  return i;
}
function Xr(t = !1) {
  return (n, s, i, { latestValues: o }, r) => {
    const l = (Ie(n) ? Yr : Ur)(s, o, r, n), u = Kr(s, typeof n == "string", t), c = n !== Le ? { ...u, ...l, ref: i } : {}, { children: h } = s, f = zt(() => E(h) ? h.get() : h, [h]);
    return ir(n, {
      ...c,
      children: f
    });
  };
}
function _s(t, { style: e, vars: n }, s, i) {
  Object.assign(t.style, e, i && i.getProjectionStyles(s));
  for (const o in n)
    t.style.setProperty(o, n[o]);
}
const Ws = /* @__PURE__ */ new Set([
  "baseFrequency",
  "diffuseConstant",
  "kernelMatrix",
  "kernelUnitLength",
  "keySplines",
  "keyTimes",
  "limitingConeAngle",
  "markerHeight",
  "markerWidth",
  "numOctaves",
  "targetX",
  "targetY",
  "surfaceScale",
  "specularConstant",
  "specularExponent",
  "stdDeviation",
  "tableValues",
  "viewBox",
  "gradientTransform",
  "pathLength",
  "startOffset",
  "textLength",
  "lengthAdjust"
]);
function Ks(t, e, n, s) {
  _s(t, e, void 0, s);
  for (const i in e.attrs)
    t.setAttribute(Ws.has(i) ? i : Yt(i), e.attrs[i]);
}
function ze(t, e, n) {
  var s;
  const { style: i } = t, o = {};
  for (const r in i)
    (E(i[r]) || e.style && E(e.style[r]) || Bs(r, t) || ((s = n == null ? void 0 : n.getValue(r)) === null || s === void 0 ? void 0 : s.liveStyle) !== void 0) && (o[r] = i[r]);
  return n && i && typeof i.willChange == "string" && (n.applyWillChange = !1), o;
}
function $s(t, e, n) {
  const s = ze(t, e, n);
  for (const i in t)
    if (E(t[i]) || E(e[i])) {
      const o = Mt.indexOf(i) !== -1 ? "attr" + i.charAt(0).toUpperCase() + i.substring(1) : i;
      s[o] = t[i];
    }
  return s;
}
function mn(t) {
  const e = [{}, {}];
  return t == null || t.values.forEach((n, s) => {
    e[0][s] = n.get(), e[1][s] = n.getVelocity();
  }), e;
}
function He(t, e, n, s) {
  if (typeof e == "function") {
    const [i, o] = mn(s);
    e = e(n !== void 0 ? n : t.custom, i, o);
  }
  if (typeof e == "string" && (e = t.variants && t.variants[e]), typeof e == "function") {
    const [i, o] = mn(s);
    e = e(n !== void 0 ? n : t.custom, i, o);
  }
  return e;
}
function qr(t) {
  const e = ge(null);
  return e.current === null && (e.current = t()), e.current;
}
const ye = (t) => Array.isArray(t), Zr = (t) => !!(t && typeof t == "object" && t.mix && t.toValue), Jr = (t) => ye(t) ? t[t.length - 1] || 0 : t;
function Ot(t) {
  const e = E(t) ? t.get() : t;
  return Zr(e) ? e.toValue() : e;
}
const Gs = /* @__PURE__ */ new Set([
  "opacity",
  "clipPath",
  "filter",
  "transform"
  // TODO: Can be accelerated but currently disabled until https://issues.chromium.org/issues/41491098 is resolved
  // or until we implement support for linear() easing.
  // "background-color"
]);
function zs(t) {
  if (Q.has(t))
    return "transform";
  if (Gs.has(t))
    return Yt(t);
}
function qt(t, e) {
  t.indexOf(e) === -1 && t.push(e);
}
function Zt(t, e) {
  const n = t.indexOf(e);
  n > -1 && t.splice(n, 1);
}
function Qr({ applyWillChange: t = !1, scrapeMotionValuesFromProps: e, createRenderState: n, onMount: s }, i, o, r, a) {
  const l = {
    latestValues: eo(i, o, r, a ? !1 : t, e),
    renderState: n()
  };
  return s && (l.mount = (u) => s(i, u, l)), l;
}
const Hs = (t) => (e, n) => {
  const s = F(Ht), i = F(Fe), o = () => Qr(t, e, s, i, n);
  return n ? o() : qr(o);
};
function to(t, e) {
  const n = zs(e);
  n && qt(t, n);
}
function gn(t, e, n) {
  const s = Array.isArray(e) ? e : [e];
  for (let i = 0; i < s.length; i++) {
    const o = He(t, s[i]);
    if (o) {
      const { transitionEnd: r, transition: a, ...l } = o;
      n(l, r);
    }
  }
}
function eo(t, e, n, s, i) {
  var o;
  const r = {}, a = [], l = s && ((o = t.style) === null || o === void 0 ? void 0 : o.willChange) === void 0, u = i(t, {});
  for (const x in u)
    r[x] = Ot(u[x]);
  let { initial: c, animate: h } = t;
  const f = Xt(t), d = Ls(t);
  e && d && !f && t.inherit !== !1 && (c === void 0 && (c = e.initial), h === void 0 && (h = e.animate));
  let p = n ? n.initial === !1 : !1;
  p = p || c === !1;
  const m = p ? h : c;
  return m && typeof m != "boolean" && !At(m) && gn(t, m, (x, g) => {
    for (const v in x) {
      let y = x[v];
      if (Array.isArray(y)) {
        const P = p ? y.length - 1 : 0;
        y = y[P];
      }
      y !== null && (r[v] = y);
    }
    for (const v in g)
      r[v] = g[v];
  }), l && (h && c !== !1 && !At(h) && gn(t, h, (x) => {
    for (const g in x)
      to(a, g);
  }), a.length && (r.willChange = a.join(","))), r;
}
const { schedule: w, cancel: z, state: D, steps: se } = Ms(typeof requestAnimationFrame < "u" ? requestAnimationFrame : M, !0), no = {
  useVisualState: Hs({
    scrapeMotionValuesFromProps: $s,
    createRenderState: Us,
    onMount: (t, e, { renderState: n, latestValues: s }) => {
      w.read(() => {
        try {
          n.dimensions = typeof e.getBBox == "function" ? e.getBBox() : e.getBoundingClientRect();
        } catch {
          n.dimensions = {
            x: 0,
            y: 0,
            width: 0,
            height: 0
          };
        }
      }), w.render(() => {
        $e(n, s, Ge(e.tagName), t.transformTemplate), Ks(e, n);
      });
    }
  })
}, so = {
  useVisualState: Hs({
    applyWillChange: !0,
    scrapeMotionValuesFromProps: ze,
    createRenderState: Ke
  })
};
function io(t, { forwardMotionProps: e = !1 }, n, s) {
  return {
    ...Ie(t) ? no : so,
    preloadedFeatures: n,
    useRender: Xr(e),
    createVisualElement: s,
    Component: t
  };
}
function _(t, e, n, s = { passive: !0 }) {
  return t.addEventListener(e, n, s), () => t.removeEventListener(e, n);
}
const Ys = (t) => t.pointerType === "mouse" ? typeof t.button != "number" || t.button <= 0 : t.isPrimary !== !1;
function Jt(t, e = "page") {
  return {
    point: {
      x: t[`${e}X`],
      y: t[`${e}Y`]
    }
  };
}
const ro = (t) => (e) => Ys(e) && t(e, Jt(e));
function W(t, e, n, s) {
  return _(t, e, ro(n), s);
}
const oo = (t, e) => (n) => e(t(n)), K = (...t) => t.reduce(oo);
function Xs(t) {
  let e = null;
  return () => {
    const n = () => {
      e = null;
    };
    return e === null ? (e = t, n) : !1;
  };
}
const yn = Xs("dragHorizontal"), vn = Xs("dragVertical");
function qs(t) {
  let e = !1;
  if (t === "y")
    e = vn();
  else if (t === "x")
    e = yn();
  else {
    const n = yn(), s = vn();
    n && s ? e = () => {
      n(), s();
    } : (n && n(), s && s());
  }
  return e;
}
function Zs() {
  const t = qs(!0);
  return t ? (t(), !1) : !0;
}
class tt {
  constructor(e) {
    this.isMounted = !1, this.node = e;
  }
  update() {
  }
}
function xn(t, e) {
  const n = e ? "pointerenter" : "pointerleave", s = e ? "onHoverStart" : "onHoverEnd", i = (o, r) => {
    if (o.pointerType === "touch" || Zs())
      return;
    const a = t.getProps();
    t.animationState && a.whileHover && t.animationState.setActive("whileHover", e);
    const l = a[s];
    l && w.postRender(() => l(o, r));
  };
  return W(t.current, n, i, {
    passive: !t.getProps()[s]
  });
}
class ao extends tt {
  mount() {
    this.unmount = K(xn(this.node, !0), xn(this.node, !1));
  }
  unmount() {
  }
}
class lo extends tt {
  constructor() {
    super(...arguments), this.isActive = !1;
  }
  onFocus() {
    let e = !1;
    try {
      e = this.node.current.matches(":focus-visible");
    } catch {
      e = !0;
    }
    !e || !this.node.animationState || (this.node.animationState.setActive("whileFocus", !0), this.isActive = !0);
  }
  onBlur() {
    !this.isActive || !this.node.animationState || (this.node.animationState.setActive("whileFocus", !1), this.isActive = !1);
  }
  mount() {
    this.unmount = K(_(this.node.current, "focus", () => this.onFocus()), _(this.node.current, "blur", () => this.onBlur()));
  }
  unmount() {
  }
}
const Js = (t, e) => e ? t === e ? !0 : Js(t, e.parentElement) : !1;
function ie(t, e) {
  if (!e)
    return;
  const n = new PointerEvent("pointer" + t);
  e(n, Jt(n));
}
class uo extends tt {
  constructor() {
    super(...arguments), this.removeStartListeners = M, this.removeEndListeners = M, this.removeAccessibleListeners = M, this.startPointerPress = (e, n) => {
      if (this.isPressing)
        return;
      this.removeEndListeners();
      const s = this.node.getProps(), o = W(window, "pointerup", (a, l) => {
        if (!this.checkPressEnd())
          return;
        const { onTap: u, onTapCancel: c, globalTapTarget: h } = this.node.getProps(), f = !h && !Js(this.node.current, a.target) ? c : u;
        f && w.update(() => f(a, l));
      }, {
        passive: !(s.onTap || s.onPointerUp)
      }), r = W(window, "pointercancel", (a, l) => this.cancelPress(a, l), {
        passive: !(s.onTapCancel || s.onPointerCancel)
      });
      this.removeEndListeners = K(o, r), this.startPress(e, n);
    }, this.startAccessiblePress = () => {
      const e = (o) => {
        if (o.key !== "Enter" || this.isPressing)
          return;
        const r = (a) => {
          a.key !== "Enter" || !this.checkPressEnd() || ie("up", (l, u) => {
            const { onTap: c } = this.node.getProps();
            c && w.postRender(() => c(l, u));
          });
        };
        this.removeEndListeners(), this.removeEndListeners = _(this.node.current, "keyup", r), ie("down", (a, l) => {
          this.startPress(a, l);
        });
      }, n = _(this.node.current, "keydown", e), s = () => {
        this.isPressing && ie("cancel", (o, r) => this.cancelPress(o, r));
      }, i = _(this.node.current, "blur", s);
      this.removeAccessibleListeners = K(n, i);
    };
  }
  startPress(e, n) {
    this.isPressing = !0;
    const { onTapStart: s, whileTap: i } = this.node.getProps();
    i && this.node.animationState && this.node.animationState.setActive("whileTap", !0), s && w.postRender(() => s(e, n));
  }
  checkPressEnd() {
    return this.removeEndListeners(), this.isPressing = !1, this.node.getProps().whileTap && this.node.animationState && this.node.animationState.setActive("whileTap", !1), !Zs();
  }
  cancelPress(e, n) {
    if (!this.checkPressEnd())
      return;
    const { onTapCancel: s } = this.node.getProps();
    s && w.postRender(() => s(e, n));
  }
  mount() {
    const e = this.node.getProps(), n = W(e.globalTapTarget ? window : this.node.current, "pointerdown", this.startPointerPress, {
      passive: !(e.onTapStart || e.onPointerStart)
    }), s = _(this.node.current, "focus", this.startAccessiblePress);
    this.removeStartListeners = K(n, s);
  }
  unmount() {
    this.removeStartListeners(), this.removeEndListeners(), this.removeAccessibleListeners();
  }
}
const ve = /* @__PURE__ */ new WeakMap(), re = /* @__PURE__ */ new WeakMap(), co = (t) => {
  const e = ve.get(t.target);
  e && e(t);
}, ho = (t) => {
  t.forEach(co);
};
function fo({ root: t, ...e }) {
  const n = t || document;
  re.has(n) || re.set(n, {});
  const s = re.get(n), i = JSON.stringify(e);
  return s[i] || (s[i] = new IntersectionObserver(ho, { root: t, ...e })), s[i];
}
function po(t, e, n) {
  const s = fo(e);
  return ve.set(t, n), s.observe(t), () => {
    ve.delete(t), s.unobserve(t);
  };
}
const mo = {
  some: 0,
  all: 1
};
class go extends tt {
  constructor() {
    super(...arguments), this.hasEnteredView = !1, this.isInView = !1;
  }
  startObserver() {
    this.unmount();
    const { viewport: e = {} } = this.node.getProps(), { root: n, margin: s, amount: i = "some", once: o } = e, r = {
      root: n ? n.current : void 0,
      rootMargin: s,
      threshold: typeof i == "number" ? i : mo[i]
    }, a = (l) => {
      const { isIntersecting: u } = l;
      if (this.isInView === u || (this.isInView = u, o && !u && this.hasEnteredView))
        return;
      u && (this.hasEnteredView = !0), this.node.animationState && this.node.animationState.setActive("whileInView", u);
      const { onViewportEnter: c, onViewportLeave: h } = this.node.getProps(), f = u ? c : h;
      f && f(l);
    };
    return po(this.node.current, r, a);
  }
  mount() {
    this.startObserver();
  }
  update() {
    if (typeof IntersectionObserver > "u")
      return;
    const { props: e, prevProps: n } = this.node;
    ["amount", "margin", "root"].some(yo(e, n)) && this.startObserver();
  }
  unmount() {
  }
}
function yo({ viewport: t = {} }, { viewport: e = {} } = {}) {
  return (n) => t[n] !== e[n];
}
const vo = {
  inView: {
    Feature: go
  },
  tap: {
    Feature: uo
  },
  focus: {
    Feature: lo
  },
  hover: {
    Feature: ao
  }
};
function Qs(t, e) {
  if (!Array.isArray(e))
    return !1;
  const n = e.length;
  if (n !== t.length)
    return !1;
  for (let s = 0; s < n; s++)
    if (e[s] !== t[s])
      return !1;
  return !0;
}
function Qt(t, e, n) {
  const s = t.getProps();
  return He(s, e, n !== void 0 ? n : s.custom, t);
}
const $ = (t) => t * 1e3, G = (t) => t / 1e3, xo = {
  type: "spring",
  stiffness: 500,
  damping: 25,
  restSpeed: 10
}, To = (t) => ({
  type: "spring",
  stiffness: 550,
  damping: t === 0 ? 2 * Math.sqrt(550) : 30,
  restSpeed: 10
}), Po = {
  type: "keyframes",
  duration: 0.8
}, So = {
  type: "keyframes",
  ease: [0.25, 0.1, 0.35, 1],
  duration: 0.3
}, bo = (t, { keyframes: e }) => e.length > 2 ? Po : Q.has(t) ? t.startsWith("scale") ? To(e[1]) : xo : So;
function wo({ when: t, delay: e, delayChildren: n, staggerChildren: s, staggerDirection: i, repeat: o, repeatType: r, repeatDelay: a, from: l, elapsed: u, ...c }) {
  return !!Object.keys(c).length;
}
function Ye(t, e) {
  return t[e] || t.default || t;
}
const Ao = (t) => t !== null;
function te(t, { repeat: e, repeatType: n = "loop" }, s) {
  const i = t.filter(Ao), o = e && n !== "loop" && e % 2 === 1 ? 0 : i.length - 1;
  return !o || s === void 0 ? i[o] : s;
}
let jt;
function Vo() {
  jt = void 0;
}
const X = {
  now: () => (jt === void 0 && X.set(D.isProcessing || cr.useManualTiming ? D.timestamp : performance.now()), jt),
  set: (t) => {
    jt = t, queueMicrotask(Vo);
  }
}, ti = (t) => /^0[^.\s]+$/u.test(t);
function Co(t) {
  return typeof t == "number" ? t === 0 : t !== null ? t === "none" || t === "0" || ti(t) : !0;
}
const ei = (t) => /^-?(?:\d+(?:\.\d+)?|\.\d+)$/u.test(t), Do = (
  // eslint-disable-next-line redos-detector/no-unsafe-regex -- false positive, as it can match a lot of words
  /^var\(--(?:([\w-]+)|([\w-]+), ?([a-zA-Z\d ()%#.,-]+))\)/u
);
function Mo(t) {
  const e = Do.exec(t);
  if (!e)
    return [,];
  const [, n, s, i] = e;
  return [`--${n ?? s}`, i];
}
const Ro = 4;
function ni(t, e, n = 1) {
  q(n <= Ro, `Max CSS variable fallback depth detected in property "${t}". This may indicate a circular fallback dependency.`);
  const [s, i] = Mo(t);
  if (!s)
    return;
  const o = window.getComputedStyle(e).getPropertyValue(s);
  if (o) {
    const r = o.trim();
    return ei(r) ? parseFloat(r) : r;
  }
  return _e(i) ? ni(i, e, n + 1) : i;
}
const Eo = /* @__PURE__ */ new Set([
  "width",
  "height",
  "top",
  "left",
  "right",
  "bottom",
  "x",
  "y",
  "translateX",
  "translateY"
]), Tn = (t) => t === gt || t === T, Pn = (t, e) => parseFloat(t.split(", ")[e]), Sn = (t, e) => (n, { transform: s }) => {
  if (s === "none" || !s)
    return 0;
  const i = s.match(/^matrix3d\((.+)\)$/u);
  if (i)
    return Pn(i[1], e);
  {
    const o = s.match(/^matrix\((.+)\)$/u);
    return o ? Pn(o[1], t) : 0;
  }
}, Lo = /* @__PURE__ */ new Set(["x", "y", "z"]), Fo = Mt.filter((t) => !Lo.has(t));
function Bo(t) {
  const e = [];
  return Fo.forEach((n) => {
    const s = t.getValue(n);
    s !== void 0 && (e.push([n, s.get()]), s.set(n.startsWith("scale") ? 1 : 0));
  }), e;
}
const dt = {
  // Dimensions
  width: ({ x: t }, { paddingLeft: e = "0", paddingRight: n = "0" }) => t.max - t.min - parseFloat(e) - parseFloat(n),
  height: ({ y: t }, { paddingTop: e = "0", paddingBottom: n = "0" }) => t.max - t.min - parseFloat(e) - parseFloat(n),
  top: (t, { top: e }) => parseFloat(e),
  left: (t, { left: e }) => parseFloat(e),
  bottom: ({ y: t }, { top: e }) => parseFloat(e) + (t.max - t.min),
  right: ({ x: t }, { left: e }) => parseFloat(e) + (t.max - t.min),
  // Transform
  x: Sn(4, 13),
  y: Sn(5, 14)
};
dt.translateX = dt.x;
dt.translateY = dt.y;
const si = (t) => (e) => e.test(t), ko = {
  test: (t) => t === "auto",
  parse: (t) => t
}, ii = [gt, T, N, Y, Lr, Er, ko], bn = (t) => ii.find(si(t)), rt = /* @__PURE__ */ new Set();
let xe = !1, Te = !1;
function ri() {
  if (Te) {
    const t = Array.from(rt).filter((s) => s.needsMeasurement), e = new Set(t.map((s) => s.element)), n = /* @__PURE__ */ new Map();
    e.forEach((s) => {
      const i = Bo(s);
      i.length && (n.set(s, i), s.render());
    }), t.forEach((s) => s.measureInitialState()), e.forEach((s) => {
      s.render();
      const i = n.get(s);
      i && i.forEach(([o, r]) => {
        var a;
        (a = s.getValue(o)) === null || a === void 0 || a.set(r);
      });
    }), t.forEach((s) => s.measureEndState()), t.forEach((s) => {
      s.suspendedScrollY !== void 0 && window.scrollTo(0, s.suspendedScrollY);
    });
  }
  Te = !1, xe = !1, rt.forEach((t) => t.complete()), rt.clear();
}
function oi() {
  rt.forEach((t) => {
    t.readKeyframes(), t.needsMeasurement && (Te = !0);
  });
}
function Oo() {
  oi(), ri();
}
class Xe {
  constructor(e, n, s, i, o, r = !1) {
    this.isComplete = !1, this.isAsync = !1, this.needsMeasurement = !1, this.isScheduled = !1, this.unresolvedKeyframes = [...e], this.onComplete = n, this.name = s, this.motionValue = i, this.element = o, this.isAsync = r;
  }
  scheduleResolve() {
    this.isScheduled = !0, this.isAsync ? (rt.add(this), xe || (xe = !0, w.read(oi), w.resolveKeyframes(ri))) : (this.readKeyframes(), this.complete());
  }
  readKeyframes() {
    const { unresolvedKeyframes: e, name: n, element: s, motionValue: i } = this;
    for (let o = 0; o < e.length; o++)
      if (e[o] === null)
        if (o === 0) {
          const r = i == null ? void 0 : i.get(), a = e[e.length - 1];
          if (r !== void 0)
            e[0] = r;
          else if (s && n) {
            const l = s.readValue(n, a);
            l != null && (e[0] = l);
          }
          e[0] === void 0 && (e[0] = a), i && r === void 0 && i.set(e[0]);
        } else
          e[o] = e[o - 1];
  }
  setFinalKeyframe() {
  }
  measureInitialState() {
  }
  renderEndStyles() {
  }
  measureEndState() {
  }
  complete() {
    this.isComplete = !0, this.onComplete(this.unresolvedKeyframes, this.finalKeyframe), rt.delete(this);
  }
  cancel() {
    this.isComplete || (this.isScheduled = !1, rt.delete(this));
  }
  resume() {
    this.isComplete || this.scheduleResolve();
  }
}
const qe = (t, e) => (n) => !!(Rt(n) && Mr.test(n) && n.startsWith(t) || e && !Rr(n) && Object.prototype.hasOwnProperty.call(n, e)), ai = (t, e, n) => (s) => {
  if (!Rt(s))
    return s;
  const [i, o, r, a] = s.match(Ne);
  return {
    [t]: parseFloat(i),
    [e]: parseFloat(o),
    [n]: parseFloat(r),
    alpha: a !== void 0 ? parseFloat(a) : 1
  };
}, jo = (t) => Z(0, 255, t), oe = {
  ...gt,
  transform: (t) => Math.round(jo(t))
}, it = {
  test: qe("rgb", "red"),
  parse: ai("red", "green", "blue"),
  transform: ({ red: t, green: e, blue: n, alpha: s = 1 }) => "rgba(" + oe.transform(t) + ", " + oe.transform(e) + ", " + oe.transform(n) + ", " + Pt(Tt.transform(s)) + ")"
};
function Io(t) {
  let e = "", n = "", s = "", i = "";
  return t.length > 5 ? (e = t.substring(1, 3), n = t.substring(3, 5), s = t.substring(5, 7), i = t.substring(7, 9)) : (e = t.substring(1, 2), n = t.substring(2, 3), s = t.substring(3, 4), i = t.substring(4, 5), e += e, n += n, s += s, i += i), {
    red: parseInt(e, 16),
    green: parseInt(n, 16),
    blue: parseInt(s, 16),
    alpha: i ? parseInt(i, 16) / 255 : 1
  };
}
const Pe = {
  test: qe("#"),
  parse: Io,
  transform: it.transform
}, lt = {
  test: qe("hsl", "hue"),
  parse: ai("hue", "saturation", "lightness"),
  transform: ({ hue: t, saturation: e, lightness: n, alpha: s = 1 }) => "hsla(" + Math.round(t) + ", " + N.transform(Pt(e)) + ", " + N.transform(Pt(n)) + ", " + Pt(Tt.transform(s)) + ")"
}, R = {
  test: (t) => it.test(t) || Pe.test(t) || lt.test(t),
  parse: (t) => it.test(t) ? it.parse(t) : lt.test(t) ? lt.parse(t) : Pe.parse(t),
  transform: (t) => Rt(t) ? t : t.hasOwnProperty("red") ? it.transform(t) : lt.transform(t)
};
function No(t) {
  var e, n;
  return isNaN(t) && Rt(t) && (((e = t.match(Ne)) === null || e === void 0 ? void 0 : e.length) || 0) + (((n = t.match(Dr)) === null || n === void 0 ? void 0 : n.length) || 0) > 0;
}
const li = "number", ui = "color", Uo = "var", _o = "var(", wn = "${}", Wo = /var\s*\(\s*--(?:[\w-]+\s*|[\w-]+\s*,(?:\s*[^)(\s]|\s*\((?:[^)(]|\([^)(]*\))*\))+\s*)\)|#[\da-f]{3,8}|(?:rgb|hsl)a?\((?:-?[\d.]+%?[,\s]+){2}-?[\d.]+%?\s*(?:[,/]\s*)?(?:\b\d+(?:\.\d+)?|\.\d+)?%?\)|-?(?:\d+(?:\.\d+)?|\.\d+)/giu;
function Vt(t) {
  const e = t.toString(), n = [], s = {
    color: [],
    number: [],
    var: []
  }, i = [];
  let o = 0;
  const a = e.replace(Wo, (l) => (R.test(l) ? (s.color.push(o), i.push(ui), n.push(R.parse(l))) : l.startsWith(_o) ? (s.var.push(o), i.push(Uo), n.push(l)) : (s.number.push(o), i.push(li), n.push(parseFloat(l))), ++o, wn)).split(wn);
  return { values: n, split: a, indexes: s, types: i };
}
function ci(t) {
  return Vt(t).values;
}
function hi(t) {
  const { split: e, types: n } = Vt(t), s = e.length;
  return (i) => {
    let o = "";
    for (let r = 0; r < s; r++)
      if (o += e[r], i[r] !== void 0) {
        const a = n[r];
        a === li ? o += Pt(i[r]) : a === ui ? o += R.transform(i[r]) : o += i[r];
      }
    return o;
  };
}
const Ko = (t) => typeof t == "number" ? 0 : t;
function $o(t) {
  const e = ci(t);
  return hi(t)(e.map(Ko));
}
const J = {
  test: No,
  parse: ci,
  createTransformer: hi,
  getAnimatableNone: $o
}, Go = /* @__PURE__ */ new Set(["brightness", "contrast", "saturate", "opacity"]);
function zo(t) {
  const [e, n] = t.slice(0, -1).split("(");
  if (e === "drop-shadow")
    return t;
  const [s] = n.match(Ne) || [];
  if (!s)
    return t;
  const i = n.replace(s, "");
  let o = Go.has(e) ? 1 : 0;
  return s !== n && (o *= 100), e + "(" + o + i + ")";
}
const Ho = /\b([a-z-]*)\(.*?\)/gu, Se = {
  ...J,
  getAnimatableNone: (t) => {
    const e = t.match(Ho);
    return e ? e.map(zo).join(" ") : t;
  }
}, Yo = {
  ...Ue,
  // Color props
  color: R,
  backgroundColor: R,
  outlineColor: R,
  fill: R,
  stroke: R,
  // Border props
  borderColor: R,
  borderTopColor: R,
  borderRightColor: R,
  borderBottomColor: R,
  borderLeftColor: R,
  filter: Se,
  WebkitFilter: Se
}, Ze = (t) => Yo[t];
function fi(t, e) {
  let n = Ze(t);
  return n !== Se && (n = J), n.getAnimatableNone ? n.getAnimatableNone(e) : void 0;
}
const Xo = /* @__PURE__ */ new Set(["auto", "none", "0"]);
function qo(t, e, n) {
  let s = 0, i;
  for (; s < t.length && !i; ) {
    const o = t[s];
    typeof o == "string" && !Xo.has(o) && Vt(o).values.length && (i = t[s]), s++;
  }
  if (i && n)
    for (const o of e)
      t[o] = fi(n, i);
}
class di extends Xe {
  constructor(e, n, s, i) {
    super(e, n, s, i, i == null ? void 0 : i.owner, !0);
  }
  readKeyframes() {
    const { unresolvedKeyframes: e, element: n, name: s } = this;
    if (!n.current)
      return;
    super.readKeyframes();
    for (let l = 0; l < e.length; l++) {
      let u = e[l];
      if (typeof u == "string" && (u = u.trim(), _e(u))) {
        const c = ni(u, n.current);
        c !== void 0 && (e[l] = c), l === e.length - 1 && (this.finalKeyframe = u);
      }
    }
    if (this.resolveNoneKeyframes(), !Eo.has(s) || e.length !== 2)
      return;
    const [i, o] = e, r = bn(i), a = bn(o);
    if (r !== a)
      if (Tn(r) && Tn(a))
        for (let l = 0; l < e.length; l++) {
          const u = e[l];
          typeof u == "string" && (e[l] = parseFloat(u));
        }
      else
        this.needsMeasurement = !0;
  }
  resolveNoneKeyframes() {
    const { unresolvedKeyframes: e, name: n } = this, s = [];
    for (let i = 0; i < e.length; i++)
      Co(e[i]) && s.push(i);
    s.length && qo(e, s, n);
  }
  measureInitialState() {
    const { element: e, unresolvedKeyframes: n, name: s } = this;
    if (!e.current)
      return;
    s === "height" && (this.suspendedScrollY = window.pageYOffset), this.measuredOrigin = dt[s](e.measureViewportBox(), window.getComputedStyle(e.current)), n[0] = this.measuredOrigin;
    const i = n[n.length - 1];
    i !== void 0 && e.getValue(s, i).jump(i, !1);
  }
  measureEndState() {
    var e;
    const { element: n, name: s, unresolvedKeyframes: i } = this;
    if (!n.current)
      return;
    const o = n.getValue(s);
    o && o.jump(this.measuredOrigin, !1);
    const r = i.length - 1, a = i[r];
    i[r] = dt[s](n.measureViewportBox(), window.getComputedStyle(n.current)), a !== null && this.finalKeyframe === void 0 && (this.finalKeyframe = a), !((e = this.removedTransforms) === null || e === void 0) && e.length && this.removedTransforms.forEach(([l, u]) => {
      n.getValue(l).set(u);
    }), this.resolveNoneKeyframes();
  }
}
function pi(t) {
  let e;
  return () => (e === void 0 && (e = t()), e);
}
const An = (t, e) => e === "zIndex" ? !1 : !!(typeof t == "number" || Array.isArray(t) || typeof t == "string" && // It's animatable if we have a string
(J.test(t) || t === "0") && // And it contains numbers and/or colors
!t.startsWith("url("));
function Zo(t) {
  const e = t[0];
  if (t.length === 1)
    return !0;
  for (let n = 0; n < t.length; n++)
    if (t[n] !== e)
      return !0;
}
function Jo(t, e, n, s) {
  const i = t[0];
  if (i === null)
    return !1;
  if (e === "display" || e === "visibility")
    return !0;
  const o = t[t.length - 1], r = An(i, e), a = An(o, e);
  return mt(r === a, `You are trying to animate ${e} from "${i}" to "${o}". ${i} is not an animatable value - to enable this animation set ${i} to a value animatable to ${o} via the \`style\` property.`), !r || !a ? !1 : Zo(t) || n === "spring" && s;
}
class mi {
  constructor({ autoplay: e = !0, delay: n = 0, type: s = "keyframes", repeat: i = 0, repeatDelay: o = 0, repeatType: r = "loop", ...a }) {
    this.isStopped = !1, this.hasAttemptedResolve = !1, this.options = {
      autoplay: e,
      delay: n,
      type: s,
      repeat: i,
      repeatDelay: o,
      repeatType: r,
      ...a
    }, this.updateFinishedPromise();
  }
  /**
   * A getter for resolved data. If keyframes are not yet resolved, accessing
   * this.resolved will synchronously flush all pending keyframe resolvers.
   * This is a deoptimisation, but at its worst still batches read/writes.
   */
  get resolved() {
    return !this._resolved && !this.hasAttemptedResolve && Oo(), this._resolved;
  }
  /**
   * A method to be called when the keyframes resolver completes. This method
   * will check if its possible to run the animation and, if not, skip it.
   * Otherwise, it will call initPlayback on the implementing class.
   */
  onKeyframesResolved(e, n) {
    this.hasAttemptedResolve = !0;
    const { name: s, type: i, velocity: o, delay: r, onComplete: a, onUpdate: l, isGenerator: u } = this.options;
    if (!u && !Jo(e, s, i, o))
      if (r)
        this.options.duration = 0;
      else {
        l == null || l(te(e, this.options, n)), a == null || a(), this.resolveFinishedPromise();
        return;
      }
    const c = this.initPlayback(e, n);
    c !== !1 && (this._resolved = {
      keyframes: e,
      finalKeyframe: n,
      ...c
    }, this.onPostResolved());
  }
  onPostResolved() {
  }
  /**
   * Allows the returned animation to be awaited or promise-chained. Currently
   * resolves when the animation finishes at all but in a future update could/should
   * reject if its cancels.
   */
  then(e, n) {
    return this.currentFinishedPromise.then(e, n);
  }
  updateFinishedPromise() {
    this.currentFinishedPromise = new Promise((e) => {
      this.resolveFinishedPromise = e;
    });
  }
}
function gi(t, e) {
  return e ? t * (1e3 / e) : 0;
}
const Qo = 5;
function yi(t, e, n) {
  const s = Math.max(e - Qo, 0);
  return gi(n - t(s), e - s);
}
const ae = 1e-3, ta = 0.01, Vn = 10, ea = 0.05, na = 1;
function sa({ duration: t = 800, bounce: e = 0.25, velocity: n = 0, mass: s = 1 }) {
  let i, o;
  mt(t <= $(Vn), "Spring duration must be 10 seconds or less");
  let r = 1 - e;
  r = Z(ea, na, r), t = Z(ta, Vn, G(t)), r < 1 ? (i = (u) => {
    const c = u * r, h = c * t, f = c - n, d = be(u, r), p = Math.exp(-h);
    return ae - f / d * p;
  }, o = (u) => {
    const h = u * r * t, f = h * n + n, d = Math.pow(r, 2) * Math.pow(u, 2) * t, p = Math.exp(-h), m = be(Math.pow(u, 2), r);
    return (-i(u) + ae > 0 ? -1 : 1) * ((f - d) * p) / m;
  }) : (i = (u) => {
    const c = Math.exp(-u * t), h = (u - n) * t + 1;
    return -ae + c * h;
  }, o = (u) => {
    const c = Math.exp(-u * t), h = (n - u) * (t * t);
    return c * h;
  });
  const a = 5 / t, l = ra(i, o, a);
  if (t = $(t), isNaN(l))
    return {
      stiffness: 100,
      damping: 10,
      duration: t
    };
  {
    const u = Math.pow(l, 2) * s;
    return {
      stiffness: u,
      damping: r * 2 * Math.sqrt(s * u),
      duration: t
    };
  }
}
const ia = 12;
function ra(t, e, n) {
  let s = n;
  for (let i = 1; i < ia; i++)
    s = s - t(s) / e(s);
  return s;
}
function be(t, e) {
  return t * Math.sqrt(1 - e * e);
}
const oa = ["duration", "bounce"], aa = ["stiffness", "damping", "mass"];
function Cn(t, e) {
  return e.some((n) => t[n] !== void 0);
}
function la(t) {
  let e = {
    velocity: 0,
    stiffness: 100,
    damping: 10,
    mass: 1,
    isResolvedFromDuration: !1,
    ...t
  };
  if (!Cn(t, aa) && Cn(t, oa)) {
    const n = sa(t);
    e = {
      ...e,
      ...n,
      mass: 1
    }, e.isResolvedFromDuration = !0;
  }
  return e;
}
function vi({ keyframes: t, restDelta: e, restSpeed: n, ...s }) {
  const i = t[0], o = t[t.length - 1], r = { done: !1, value: i }, { stiffness: a, damping: l, mass: u, duration: c, velocity: h, isResolvedFromDuration: f } = la({
    ...s,
    velocity: -G(s.velocity || 0)
  }), d = h || 0, p = l / (2 * Math.sqrt(a * u)), m = o - i, x = G(Math.sqrt(a / u)), g = Math.abs(m) < 5;
  n || (n = g ? 0.01 : 2), e || (e = g ? 5e-3 : 0.5);
  let v;
  if (p < 1) {
    const y = be(x, p);
    v = (P) => {
      const S = Math.exp(-p * x * P);
      return o - S * ((d + p * x * m) / y * Math.sin(y * P) + m * Math.cos(y * P));
    };
  } else if (p === 1)
    v = (y) => o - Math.exp(-x * y) * (m + (d + x * m) * y);
  else {
    const y = x * Math.sqrt(p * p - 1);
    v = (P) => {
      const S = Math.exp(-p * x * P), C = Math.min(y * P, 300);
      return o - S * ((d + p * x * m) * Math.sinh(C) + y * m * Math.cosh(C)) / y;
    };
  }
  return {
    calculatedDuration: f && c || null,
    next: (y) => {
      const P = v(y);
      if (f)
        r.done = y >= c;
      else {
        let S = d;
        y !== 0 && (p < 1 ? S = yi(v, y, P) : S = 0);
        const C = Math.abs(S) <= n, H = Math.abs(o - P) <= e;
        r.done = C && H;
      }
      return r.value = r.done ? o : P, r;
    }
  };
}
function Dn({ keyframes: t, velocity: e = 0, power: n = 0.8, timeConstant: s = 325, bounceDamping: i = 10, bounceStiffness: o = 500, modifyTarget: r, min: a, max: l, restDelta: u = 0.5, restSpeed: c }) {
  const h = t[0], f = {
    done: !1,
    value: h
  }, d = (b) => a !== void 0 && b < a || l !== void 0 && b > l, p = (b) => a === void 0 ? l : l === void 0 || Math.abs(a - b) < Math.abs(l - b) ? a : l;
  let m = n * e;
  const x = h + m, g = r === void 0 ? x : r(x);
  g !== x && (m = g - h);
  const v = (b) => -m * Math.exp(-b / s), y = (b) => g + v(b), P = (b) => {
    const k = v(b), O = y(b);
    f.done = Math.abs(k) <= u, f.value = f.done ? g : O;
  };
  let S, C;
  const H = (b) => {
    d(f.value) && (S = b, C = vi({
      keyframes: [f.value, p(f.value)],
      velocity: yi(y, b, f.value),
      // TODO: This should be passing * 1000
      damping: i,
      stiffness: o,
      restDelta: u,
      restSpeed: c
    }));
  };
  return H(0), {
    calculatedDuration: null,
    next: (b) => {
      let k = !1;
      return !C && S === void 0 && (k = !0, P(b), H(b)), S !== void 0 && b >= S ? C.next(b - S) : (!k && P(b), f);
    }
  };
}
const xi = (t, e, n) => (((1 - 3 * n + 3 * e) * t + (3 * n - 6 * e)) * t + 3 * e) * t, ua = 1e-7, ca = 12;
function ha(t, e, n, s, i) {
  let o, r, a = 0;
  do
    r = e + (n - e) / 2, o = xi(r, s, i) - t, o > 0 ? n = r : e = r;
  while (Math.abs(o) > ua && ++a < ca);
  return r;
}
function Lt(t, e, n, s) {
  if (t === e && n === s)
    return M;
  const i = (o) => ha(o, 0, 1, t, n);
  return (o) => o === 0 || o === 1 ? o : xi(i(o), e, s);
}
const fa = Lt(0.42, 0, 1, 1), da = Lt(0, 0, 0.58, 1), Ti = Lt(0.42, 0, 0.58, 1), pa = (t) => Array.isArray(t) && typeof t[0] != "number", Pi = (t) => (e) => e <= 0.5 ? t(2 * e) / 2 : (2 - t(2 * (1 - e))) / 2, Si = (t) => (e) => 1 - t(1 - e), Je = (t) => 1 - Math.sin(Math.acos(t)), bi = Si(Je), ma = Pi(Je), wi = Lt(0.33, 1.53, 0.69, 0.99), Qe = Si(wi), ga = Pi(Qe), ya = (t) => (t *= 2) < 1 ? 0.5 * Qe(t) : 0.5 * (2 - Math.pow(2, -10 * (t - 1))), Mn = {
  linear: M,
  easeIn: fa,
  easeInOut: Ti,
  easeOut: da,
  circIn: Je,
  circInOut: ma,
  circOut: bi,
  backIn: Qe,
  backInOut: ga,
  backOut: wi,
  anticipate: ya
}, Rn = (t) => {
  if (Array.isArray(t)) {
    q(t.length === 4, "Cubic bezier arrays must contain four numerical values.");
    const [e, n, s, i] = t;
    return Lt(e, n, s, i);
  } else if (typeof t == "string")
    return q(Mn[t] !== void 0, `Invalid easing type '${t}'`), Mn[t];
  return t;
}, Ct = (t, e, n) => {
  const s = e - t;
  return s === 0 ? 1 : (n - t) / s;
}, A = (t, e, n) => t + (e - t) * n;
function le(t, e, n) {
  return n < 0 && (n += 1), n > 1 && (n -= 1), n < 1 / 6 ? t + (e - t) * 6 * n : n < 1 / 2 ? e : n < 2 / 3 ? t + (e - t) * (2 / 3 - n) * 6 : t;
}
function va({ hue: t, saturation: e, lightness: n, alpha: s }) {
  t /= 360, e /= 100, n /= 100;
  let i = 0, o = 0, r = 0;
  if (!e)
    i = o = r = n;
  else {
    const a = n < 0.5 ? n * (1 + e) : n + e - n * e, l = 2 * n - a;
    i = le(l, a, t + 1 / 3), o = le(l, a, t), r = le(l, a, t - 1 / 3);
  }
  return {
    red: Math.round(i * 255),
    green: Math.round(o * 255),
    blue: Math.round(r * 255),
    alpha: s
  };
}
function Wt(t, e) {
  return (n) => n > 0 ? e : t;
}
const ue = (t, e, n) => {
  const s = t * t, i = n * (e * e - s) + s;
  return i < 0 ? 0 : Math.sqrt(i);
}, xa = [Pe, it, lt], Ta = (t) => xa.find((e) => e.test(t));
function En(t) {
  const e = Ta(t);
  if (mt(!!e, `'${t}' is not an animatable color. Use the equivalent color code instead.`), !e)
    return !1;
  let n = e.parse(t);
  return e === lt && (n = va(n)), n;
}
const Ln = (t, e) => {
  const n = En(t), s = En(e);
  if (!n || !s)
    return Wt(t, e);
  const i = { ...n };
  return (o) => (i.red = ue(n.red, s.red, o), i.green = ue(n.green, s.green, o), i.blue = ue(n.blue, s.blue, o), i.alpha = A(n.alpha, s.alpha, o), it.transform(i));
}, we = /* @__PURE__ */ new Set(["none", "hidden"]);
function Pa(t, e) {
  return we.has(t) ? (n) => n <= 0 ? t : e : (n) => n >= 1 ? e : t;
}
function Sa(t, e) {
  return (n) => A(t, e, n);
}
function tn(t) {
  return typeof t == "number" ? Sa : typeof t == "string" ? _e(t) ? Wt : R.test(t) ? Ln : Aa : Array.isArray(t) ? Ai : typeof t == "object" ? R.test(t) ? Ln : ba : Wt;
}
function Ai(t, e) {
  const n = [...t], s = n.length, i = t.map((o, r) => tn(o)(o, e[r]));
  return (o) => {
    for (let r = 0; r < s; r++)
      n[r] = i[r](o);
    return n;
  };
}
function ba(t, e) {
  const n = { ...t, ...e }, s = {};
  for (const i in n)
    t[i] !== void 0 && e[i] !== void 0 && (s[i] = tn(t[i])(t[i], e[i]));
  return (i) => {
    for (const o in s)
      n[o] = s[o](i);
    return n;
  };
}
function wa(t, e) {
  var n;
  const s = [], i = { color: 0, var: 0, number: 0 };
  for (let o = 0; o < e.values.length; o++) {
    const r = e.types[o], a = t.indexes[r][i[r]], l = (n = t.values[a]) !== null && n !== void 0 ? n : 0;
    s[o] = l, i[r]++;
  }
  return s;
}
const Aa = (t, e) => {
  const n = J.createTransformer(e), s = Vt(t), i = Vt(e);
  return s.indexes.var.length === i.indexes.var.length && s.indexes.color.length === i.indexes.color.length && s.indexes.number.length >= i.indexes.number.length ? we.has(t) && !i.values.length || we.has(e) && !s.values.length ? Pa(t, e) : K(Ai(wa(s, i), i.values), n) : (mt(!0, `Complex values '${t}' and '${e}' too different to mix. Ensure all colors are of the same type, and that each contains the same quantity of number and color values. Falling back to instant transition.`), Wt(t, e));
};
function Vi(t, e, n) {
  return typeof t == "number" && typeof e == "number" && typeof n == "number" ? A(t, e, n) : tn(t)(t, e);
}
function Va(t, e, n) {
  const s = [], i = n || Vi, o = t.length - 1;
  for (let r = 0; r < o; r++) {
    let a = i(t[r], t[r + 1]);
    if (e) {
      const l = Array.isArray(e) ? e[r] || M : e;
      a = K(l, a);
    }
    s.push(a);
  }
  return s;
}
function Ca(t, e, { clamp: n = !0, ease: s, mixer: i } = {}) {
  const o = t.length;
  if (q(o === e.length, "Both input and output ranges must be the same length"), o === 1)
    return () => e[0];
  if (o === 2 && t[0] === t[1])
    return () => e[1];
  t[0] > t[o - 1] && (t = [...t].reverse(), e = [...e].reverse());
  const r = Va(e, s, i), a = r.length, l = (u) => {
    let c = 0;
    if (a > 1)
      for (; c < t.length - 2 && !(u < t[c + 1]); c++)
        ;
    const h = Ct(t[c], t[c + 1], u);
    return r[c](h);
  };
  return n ? (u) => l(Z(t[0], t[o - 1], u)) : l;
}
function Da(t, e) {
  const n = t[t.length - 1];
  for (let s = 1; s <= e; s++) {
    const i = Ct(0, e, s);
    t.push(A(n, 1, i));
  }
}
function Ma(t) {
  const e = [0];
  return Da(e, t.length - 1), e;
}
function Ra(t, e) {
  return t.map((n) => n * e);
}
function Ea(t, e) {
  return t.map(() => e || Ti).splice(0, t.length - 1);
}
function Kt({ duration: t = 300, keyframes: e, times: n, ease: s = "easeInOut" }) {
  const i = pa(s) ? s.map(Rn) : Rn(s), o = {
    done: !1,
    value: e[0]
  }, r = Ra(
    // Only use the provided offsets if they're the correct length
    // TODO Maybe we should warn here if there's a length mismatch
    n && n.length === e.length ? n : Ma(e),
    t
  ), a = Ca(r, e, {
    ease: Array.isArray(i) ? i : Ea(e, i)
  });
  return {
    calculatedDuration: t,
    next: (l) => (o.value = a(l), o.done = l >= t, o)
  };
}
const Fn = 2e4;
function La(t) {
  let e = 0;
  const n = 50;
  let s = t.next(e);
  for (; !s.done && e < Fn; )
    e += n, s = t.next(e);
  return e >= Fn ? 1 / 0 : e;
}
const Fa = (t) => {
  const e = ({ timestamp: n }) => t(n);
  return {
    start: () => w.update(e, !0),
    stop: () => z(e),
    /**
     * If we're processing this frame we can use the
     * framelocked timestamp to keep things in sync.
     */
    now: () => D.isProcessing ? D.timestamp : X.now()
  };
}, Ba = {
  decay: Dn,
  inertia: Dn,
  tween: Kt,
  keyframes: Kt,
  spring: vi
}, ka = (t) => t / 100;
class en extends mi {
  constructor({ KeyframeResolver: e = Xe, ...n }) {
    super(n), this.holdTime = null, this.startTime = null, this.cancelTime = null, this.currentTime = 0, this.playbackSpeed = 1, this.pendingPlayState = "running", this.state = "idle", this.stop = () => {
      if (this.resolver.cancel(), this.isStopped = !0, this.state === "idle")
        return;
      this.teardown();
      const { onStop: a } = this.options;
      a && a();
    };
    const { name: s, motionValue: i, keyframes: o } = this.options, r = (a, l) => this.onKeyframesResolved(a, l);
    s && i && i.owner ? this.resolver = i.owner.resolveKeyframes(o, r, s, i) : this.resolver = new e(o, r, s, i), this.resolver.scheduleResolve();
  }
  initPlayback(e) {
    const { type: n = "keyframes", repeat: s = 0, repeatDelay: i = 0, repeatType: o, velocity: r = 0 } = this.options, a = Ba[n] || Kt;
    let l, u;
    a !== Kt && typeof e[0] != "number" && (process.env.NODE_ENV !== "production" && q(e.length === 2, `Only two keyframes currently supported with spring and inertia animations. Trying to animate ${e}`), l = K(ka, Vi(e[0], e[1])), e = [0, 100]);
    const c = a({ ...this.options, keyframes: e });
    o === "mirror" && (u = a({
      ...this.options,
      keyframes: [...e].reverse(),
      velocity: -r
    })), c.calculatedDuration === null && (c.calculatedDuration = La(c));
    const { calculatedDuration: h } = c, f = h + i, d = f * (s + 1) - i;
    return {
      generator: c,
      mirroredGenerator: u,
      mapPercentToKeyframes: l,
      calculatedDuration: h,
      resolvedDuration: f,
      totalDuration: d
    };
  }
  onPostResolved() {
    const { autoplay: e = !0 } = this.options;
    this.play(), this.pendingPlayState === "paused" || !e ? this.pause() : this.state = this.pendingPlayState;
  }
  tick(e, n = !1) {
    const { resolved: s } = this;
    if (!s) {
      const { keyframes: b } = this.options;
      return { done: !0, value: b[b.length - 1] };
    }
    const { finalKeyframe: i, generator: o, mirroredGenerator: r, mapPercentToKeyframes: a, keyframes: l, calculatedDuration: u, totalDuration: c, resolvedDuration: h } = s;
    if (this.startTime === null)
      return o.next(0);
    const { delay: f, repeat: d, repeatType: p, repeatDelay: m, onUpdate: x } = this.options;
    this.speed > 0 ? this.startTime = Math.min(this.startTime, e) : this.speed < 0 && (this.startTime = Math.min(e - c / this.speed, this.startTime)), n ? this.currentTime = e : this.holdTime !== null ? this.currentTime = this.holdTime : this.currentTime = Math.round(e - this.startTime) * this.speed;
    const g = this.currentTime - f * (this.speed >= 0 ? 1 : -1), v = this.speed >= 0 ? g < 0 : g > c;
    this.currentTime = Math.max(g, 0), this.state === "finished" && this.holdTime === null && (this.currentTime = c);
    let y = this.currentTime, P = o;
    if (d) {
      const b = Math.min(this.currentTime, c) / h;
      let k = Math.floor(b), O = b % 1;
      !O && b >= 1 && (O = 1), O === 1 && k--, k = Math.min(k, d + 1), !!(k % 2) && (p === "reverse" ? (O = 1 - O, m && (O -= m / h)) : p === "mirror" && (P = r)), y = Z(0, 1, O) * h;
    }
    const S = v ? { done: !1, value: l[0] } : P.next(y);
    a && (S.value = a(S.value));
    let { done: C } = S;
    !v && u !== null && (C = this.speed >= 0 ? this.currentTime >= c : this.currentTime <= 0);
    const H = this.holdTime === null && (this.state === "finished" || this.state === "running" && C);
    return H && i !== void 0 && (S.value = te(l, this.options, i)), x && x(S.value), H && this.finish(), S;
  }
  get duration() {
    const { resolved: e } = this;
    return e ? G(e.calculatedDuration) : 0;
  }
  get time() {
    return G(this.currentTime);
  }
  set time(e) {
    e = $(e), this.currentTime = e, this.holdTime !== null || this.speed === 0 ? this.holdTime = e : this.driver && (this.startTime = this.driver.now() - e / this.speed);
  }
  get speed() {
    return this.playbackSpeed;
  }
  set speed(e) {
    const n = this.playbackSpeed !== e;
    this.playbackSpeed = e, n && (this.time = G(this.currentTime));
  }
  play() {
    if (this.resolver.isScheduled || this.resolver.resume(), !this._resolved) {
      this.pendingPlayState = "running";
      return;
    }
    if (this.isStopped)
      return;
    const { driver: e = Fa, onPlay: n } = this.options;
    this.driver || (this.driver = e((i) => this.tick(i))), n && n();
    const s = this.driver.now();
    this.holdTime !== null ? this.startTime = s - this.holdTime : (!this.startTime || this.state === "finished") && (this.startTime = s), this.state === "finished" && this.updateFinishedPromise(), this.cancelTime = this.startTime, this.holdTime = null, this.state = "running", this.driver.start();
  }
  pause() {
    var e;
    if (!this._resolved) {
      this.pendingPlayState = "paused";
      return;
    }
    this.state = "paused", this.holdTime = (e = this.currentTime) !== null && e !== void 0 ? e : 0;
  }
  complete() {
    this.state !== "running" && this.play(), this.pendingPlayState = this.state = "finished", this.holdTime = null;
  }
  finish() {
    this.teardown(), this.state = "finished";
    const { onComplete: e } = this.options;
    e && e();
  }
  cancel() {
    this.cancelTime !== null && this.tick(this.cancelTime), this.teardown(), this.updateFinishedPromise();
  }
  teardown() {
    this.state = "idle", this.stopDriver(), this.resolveFinishedPromise(), this.updateFinishedPromise(), this.startTime = this.cancelTime = null, this.resolver.cancel();
  }
  stopDriver() {
    this.driver && (this.driver.stop(), this.driver = void 0);
  }
  sample(e) {
    return this.startTime = 0, this.tick(e, !0);
  }
}
const Ci = (t) => Array.isArray(t) && typeof t[0] == "number";
function Di(t) {
  return !!(!t || typeof t == "string" && t in nn || Ci(t) || Array.isArray(t) && t.every(Di));
}
const vt = ([t, e, n, s]) => `cubic-bezier(${t}, ${e}, ${n}, ${s})`, nn = {
  linear: "linear",
  ease: "ease",
  easeIn: "ease-in",
  easeOut: "ease-out",
  easeInOut: "ease-in-out",
  circIn: vt([0, 0.65, 0.55, 1]),
  circOut: vt([0.55, 0, 1, 0.45]),
  backIn: vt([0.31, 0.01, 0.66, -0.59]),
  backOut: vt([0.33, 1.53, 0.69, 0.99])
};
function Oa(t) {
  return Mi(t) || nn.easeOut;
}
function Mi(t) {
  if (t)
    return Ci(t) ? vt(t) : Array.isArray(t) ? t.map(Oa) : nn[t];
}
function ja(t, e, n, { delay: s = 0, duration: i = 300, repeat: o = 0, repeatType: r = "loop", ease: a, times: l } = {}) {
  const u = { [e]: n };
  l && (u.offset = l);
  const c = Mi(a);
  return Array.isArray(c) && (u.easing = c), t.animate(u, {
    delay: s,
    duration: i,
    easing: Array.isArray(c) ? "linear" : c,
    fill: "both",
    iterations: o + 1,
    direction: r === "reverse" ? "alternate" : "normal"
  });
}
const Ia = pi(() => Object.hasOwnProperty.call(Element.prototype, "animate")), $t = 10, Na = 2e4;
function Ua(t) {
  return t.type === "spring" || !Di(t.ease);
}
function _a(t, e) {
  const n = new en({
    ...e,
    keyframes: t,
    repeat: 0,
    delay: 0,
    isGenerator: !0
  });
  let s = { done: !1, value: t[0] };
  const i = [];
  let o = 0;
  for (; !s.done && o < Na; )
    s = n.sample(o), i.push(s.value), o += $t;
  return {
    times: void 0,
    keyframes: i,
    duration: o - $t,
    ease: "linear"
  };
}
class Bn extends mi {
  constructor(e) {
    super(e);
    const { name: n, motionValue: s, keyframes: i } = this.options;
    this.resolver = new di(i, (o, r) => this.onKeyframesResolved(o, r), n, s), this.resolver.scheduleResolve();
  }
  initPlayback(e, n) {
    var s;
    let { duration: i = 300, times: o, ease: r, type: a, motionValue: l, name: u } = this.options;
    if (!(!((s = l.owner) === null || s === void 0) && s.current))
      return !1;
    if (Ua(this.options)) {
      const { onComplete: h, onUpdate: f, motionValue: d, ...p } = this.options, m = _a(e, p);
      e = m.keyframes, e.length === 1 && (e[1] = e[0]), i = m.duration, o = m.times, r = m.ease, a = "keyframes";
    }
    const c = ja(l.owner.current, u, e, { ...this.options, duration: i, times: o, ease: r });
    return c.startTime = X.now(), this.pendingTimeline ? (c.timeline = this.pendingTimeline, this.pendingTimeline = void 0) : c.onfinish = () => {
      const { onComplete: h } = this.options;
      l.set(te(e, this.options, n)), h && h(), this.cancel(), this.resolveFinishedPromise();
    }, {
      animation: c,
      duration: i,
      times: o,
      type: a,
      ease: r,
      keyframes: e
    };
  }
  get duration() {
    const { resolved: e } = this;
    if (!e)
      return 0;
    const { duration: n } = e;
    return G(n);
  }
  get time() {
    const { resolved: e } = this;
    if (!e)
      return 0;
    const { animation: n } = e;
    return G(n.currentTime || 0);
  }
  set time(e) {
    const { resolved: n } = this;
    if (!n)
      return;
    const { animation: s } = n;
    s.currentTime = $(e);
  }
  get speed() {
    const { resolved: e } = this;
    if (!e)
      return 1;
    const { animation: n } = e;
    return n.playbackRate;
  }
  set speed(e) {
    const { resolved: n } = this;
    if (!n)
      return;
    const { animation: s } = n;
    s.playbackRate = e;
  }
  get state() {
    const { resolved: e } = this;
    if (!e)
      return "idle";
    const { animation: n } = e;
    return n.playState;
  }
  /**
   * Replace the default DocumentTimeline with another AnimationTimeline.
   * Currently used for scroll animations.
   */
  attachTimeline(e) {
    if (!this._resolved)
      this.pendingTimeline = e;
    else {
      const { resolved: n } = this;
      if (!n)
        return M;
      const { animation: s } = n;
      s.timeline = e, s.onfinish = null;
    }
    return M;
  }
  play() {
    if (this.isStopped)
      return;
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n } = e;
    n.playState === "finished" && this.updateFinishedPromise(), n.play();
  }
  pause() {
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n } = e;
    n.pause();
  }
  stop() {
    if (this.resolver.cancel(), this.isStopped = !0, this.state === "idle")
      return;
    this.resolveFinishedPromise(), this.updateFinishedPromise();
    const { resolved: e } = this;
    if (!e)
      return;
    const { animation: n, keyframes: s, duration: i, type: o, ease: r, times: a } = e;
    if (n.playState === "idle" || n.playState === "finished")
      return;
    if (this.time) {
      const { motionValue: u, onUpdate: c, onComplete: h, ...f } = this.options, d = new en({
        ...f,
        keyframes: s,
        duration: i,
        type: o,
        ease: r,
        times: a,
        isGenerator: !0
      }), p = $(this.time);
      u.setWithVelocity(d.sample(p - $t).value, d.sample(p).value, $t);
    }
    const { onStop: l } = this.options;
    l && l(), this.cancel();
  }
  complete() {
    const { resolved: e } = this;
    e && e.animation.finish();
  }
  cancel() {
    const { resolved: e } = this;
    e && e.animation.cancel();
  }
  static supports(e) {
    const { motionValue: n, name: s, repeatDelay: i, repeatType: o, damping: r, type: a } = e;
    return Ia() && s && Gs.has(s) && n && n.owner && n.owner.current instanceof HTMLElement && /**
     * If we're outputting values to onUpdate then we can't use WAAPI as there's
     * no way to read the value from WAAPI every frame.
     */
    !n.owner.getProps().onUpdate && !i && o !== "mirror" && r !== 0 && a !== "inertia";
  }
}
function Wa(t, e) {
  let n;
  const s = () => {
    const { currentTime: i } = e, r = (i === null ? 0 : i.value) / 100;
    n !== r && t(r), n = r;
  };
  return w.update(s, !0), () => z(s);
}
const Ka = pi(() => window.ScrollTimeline !== void 0);
class $a {
  constructor(e) {
    this.stop = () => this.runAll("stop"), this.animations = e.filter(Boolean);
  }
  then(e, n) {
    return Promise.all(this.animations).then(e).catch(n);
  }
  /**
   * TODO: Filter out cancelled or stopped animations before returning
   */
  getAll(e) {
    return this.animations[0][e];
  }
  setAll(e, n) {
    for (let s = 0; s < this.animations.length; s++)
      this.animations[s][e] = n;
  }
  attachTimeline(e) {
    const n = this.animations.map((s) => {
      if (Ka() && s.attachTimeline)
        s.attachTimeline(e);
      else
        return s.pause(), Wa((i) => {
          s.time = s.duration * i;
        }, e);
    });
    return () => {
      n.forEach((s, i) => {
        s && s(), this.animations[i].stop();
      });
    };
  }
  get time() {
    return this.getAll("time");
  }
  set time(e) {
    this.setAll("time", e);
  }
  get speed() {
    return this.getAll("speed");
  }
  set speed(e) {
    this.setAll("speed", e);
  }
  get duration() {
    let e = 0;
    for (let n = 0; n < this.animations.length; n++)
      e = Math.max(e, this.animations[n].duration);
    return e;
  }
  runAll(e) {
    this.animations.forEach((n) => n[e]());
  }
  play() {
    this.runAll("play");
  }
  pause() {
    this.runAll("pause");
  }
  cancel() {
    this.runAll("cancel");
  }
  complete() {
    this.runAll("complete");
  }
}
const sn = (t, e, n, s = {}, i, o, r) => (a) => {
  const l = Ye(s, t) || {}, u = l.delay || s.delay || 0;
  let { elapsed: c = 0 } = s;
  c = c - $(u);
  let h = {
    keyframes: Array.isArray(n) ? n : [null, n],
    ease: "easeOut",
    velocity: e.getVelocity(),
    ...l,
    delay: -c,
    onUpdate: (d) => {
      e.set(d), l.onUpdate && l.onUpdate(d);
    },
    onComplete: () => {
      a(), l.onComplete && l.onComplete(), r && r();
    },
    onStop: r,
    name: t,
    motionValue: e,
    element: o ? void 0 : i
  };
  wo(l) || (h = {
    ...h,
    ...bo(t, h)
  }), h.duration && (h.duration = $(h.duration)), h.repeatDelay && (h.repeatDelay = $(h.repeatDelay)), h.from !== void 0 && (h.keyframes[0] = h.from);
  let f = !1;
  if ((h.type === !1 || h.duration === 0 && !h.repeatDelay) && (h.duration = 0, h.delay === 0 && (f = !0)), f && !o && e.get() !== void 0) {
    const d = te(h.keyframes, l);
    if (d !== void 0)
      return w.update(() => {
        h.onUpdate(d), h.onComplete();
      }), new $a([]);
  }
  return !o && Bn.supports(h) ? new Bn(h) : new en(h);
};
class rn {
  constructor() {
    this.subscriptions = [];
  }
  add(e) {
    return qt(this.subscriptions, e), () => Zt(this.subscriptions, e);
  }
  notify(e, n, s) {
    const i = this.subscriptions.length;
    if (i)
      if (i === 1)
        this.subscriptions[0](e, n, s);
      else
        for (let o = 0; o < i; o++) {
          const r = this.subscriptions[o];
          r && r(e, n, s);
        }
  }
  getSize() {
    return this.subscriptions.length;
  }
  clear() {
    this.subscriptions.length = 0;
  }
}
const kn = /* @__PURE__ */ new Set();
function on(t, e, n) {
  t || kn.has(e) || (console.warn(e), kn.add(e));
}
const On = 30, Ga = (t) => !isNaN(parseFloat(t));
class Ri {
  /**
   * @param init - The initiating value
   * @param config - Optional configuration options
   *
   * -  `transformer`: A function to transform incoming values with.
   *
   * @internal
   */
  constructor(e, n = {}) {
    this.version = "11.3.20", this.canTrackVelocity = null, this.events = {}, this.updateAndNotify = (s, i = !0) => {
      const o = X.now();
      this.updatedAt !== o && this.setPrevFrameValue(), this.prev = this.current, this.setCurrent(s), this.current !== this.prev && this.events.change && this.events.change.notify(this.current), i && this.events.renderRequest && this.events.renderRequest.notify(this.current);
    }, this.hasAnimated = !1, this.setCurrent(e), this.owner = n.owner;
  }
  setCurrent(e) {
    this.current = e, this.updatedAt = X.now(), this.canTrackVelocity === null && e !== void 0 && (this.canTrackVelocity = Ga(this.current));
  }
  setPrevFrameValue(e = this.current) {
    this.prevFrameValue = e, this.prevUpdatedAt = this.updatedAt;
  }
  /**
   * Adds a function that will be notified when the `MotionValue` is updated.
   *
   * It returns a function that, when called, will cancel the subscription.
   *
   * When calling `onChange` inside a React component, it should be wrapped with the
   * `useEffect` hook. As it returns an unsubscribe function, this should be returned
   * from the `useEffect` function to ensure you don't add duplicate subscribers..
   *
   * ```jsx
   * export const MyComponent = () => {
   *   const x = useMotionValue(0)
   *   const y = useMotionValue(0)
   *   const opacity = useMotionValue(1)
   *
   *   useEffect(() => {
   *     function updateOpacity() {
   *       const maxXY = Math.max(x.get(), y.get())
   *       const newOpacity = transform(maxXY, [0, 100], [1, 0])
   *       opacity.set(newOpacity)
   *     }
   *
   *     const unsubscribeX = x.on("change", updateOpacity)
   *     const unsubscribeY = y.on("change", updateOpacity)
   *
   *     return () => {
   *       unsubscribeX()
   *       unsubscribeY()
   *     }
   *   }, [])
   *
   *   return <motion.div style={{ x }} />
   * }
   * ```
   *
   * @param subscriber - A function that receives the latest value.
   * @returns A function that, when called, will cancel this subscription.
   *
   * @deprecated
   */
  onChange(e) {
    return process.env.NODE_ENV !== "production" && on(!1, 'value.onChange(callback) is deprecated. Switch to value.on("change", callback).'), this.on("change", e);
  }
  on(e, n) {
    this.events[e] || (this.events[e] = new rn());
    const s = this.events[e].add(n);
    return e === "change" ? () => {
      s(), w.read(() => {
        this.events.change.getSize() || this.stop();
      });
    } : s;
  }
  clearListeners() {
    for (const e in this.events)
      this.events[e].clear();
  }
  /**
   * Attaches a passive effect to the `MotionValue`.
   *
   * @internal
   */
  attach(e, n) {
    this.passiveEffect = e, this.stopPassiveEffect = n;
  }
  /**
   * Sets the state of the `MotionValue`.
   *
   * @remarks
   *
   * ```jsx
   * const x = useMotionValue(0)
   * x.set(10)
   * ```
   *
   * @param latest - Latest value to set.
   * @param render - Whether to notify render subscribers. Defaults to `true`
   *
   * @public
   */
  set(e, n = !0) {
    !n || !this.passiveEffect ? this.updateAndNotify(e, n) : this.passiveEffect(e, this.updateAndNotify);
  }
  setWithVelocity(e, n, s) {
    this.set(n), this.prev = void 0, this.prevFrameValue = e, this.prevUpdatedAt = this.updatedAt - s;
  }
  /**
   * Set the state of the `MotionValue`, stopping any active animations,
   * effects, and resets velocity to `0`.
   */
  jump(e, n = !0) {
    this.updateAndNotify(e), this.prev = e, this.prevUpdatedAt = this.prevFrameValue = void 0, n && this.stop(), this.stopPassiveEffect && this.stopPassiveEffect();
  }
  /**
   * Returns the latest state of `MotionValue`
   *
   * @returns - The latest state of `MotionValue`
   *
   * @public
   */
  get() {
    return this.current;
  }
  /**
   * @public
   */
  getPrevious() {
    return this.prev;
  }
  /**
   * Returns the latest velocity of `MotionValue`
   *
   * @returns - The latest velocity of `MotionValue`. Returns `0` if the state is non-numerical.
   *
   * @public
   */
  getVelocity() {
    const e = X.now();
    if (!this.canTrackVelocity || this.prevFrameValue === void 0 || e - this.updatedAt > On)
      return 0;
    const n = Math.min(this.updatedAt - this.prevUpdatedAt, On);
    return gi(parseFloat(this.current) - parseFloat(this.prevFrameValue), n);
  }
  /**
   * Registers a new animation to control this `MotionValue`. Only one
   * animation can drive a `MotionValue` at one time.
   *
   * ```jsx
   * value.start()
   * ```
   *
   * @param animation - A function that starts the provided animation
   *
   * @internal
   */
  start(e) {
    return this.stop(), new Promise((n) => {
      this.hasAnimated = !0, this.animation = e(n), this.events.animationStart && this.events.animationStart.notify();
    }).then(() => {
      this.events.animationComplete && this.events.animationComplete.notify(), this.clearAnimation();
    });
  }
  /**
   * Stop the currently active animation.
   *
   * @public
   */
  stop() {
    this.animation && (this.animation.stop(), this.events.animationCancel && this.events.animationCancel.notify()), this.clearAnimation();
  }
  /**
   * Returns `true` if this value is currently animating.
   *
   * @public
   */
  isAnimating() {
    return !!this.animation;
  }
  clearAnimation() {
    delete this.animation;
  }
  /**
   * Destroy and clean up subscribers to this `MotionValue`.
   *
   * The `MotionValue` hooks like `useMotionValue` and `useTransform` automatically
   * handle the lifecycle of the returned `MotionValue`, so this method is only necessary if you've manually
   * created a `MotionValue` via the `motionValue` function.
   *
   * @public
   */
  destroy() {
    this.clearListeners(), this.stop(), this.stopPassiveEffect && this.stopPassiveEffect();
  }
}
function Dt(t, e) {
  return new Ri(t, e);
}
function za(t, e, n) {
  t.hasValue(e) ? t.getValue(e).set(n) : t.addValue(e, Dt(n));
}
function Ha(t, e) {
  const n = Qt(t, e);
  let { transitionEnd: s = {}, transition: i = {}, ...o } = n || {};
  o = { ...o, ...s };
  for (const r in o) {
    const a = Jr(o[r]);
    za(t, r, a);
  }
}
function Ei(t) {
  return t.getProps()[Ds];
}
class Ya extends Ri {
  constructor() {
    super(...arguments), this.output = [], this.counts = /* @__PURE__ */ new Map();
  }
  add(e) {
    const n = zs(e);
    if (!n)
      return;
    const s = this.counts.get(n) || 0;
    this.counts.set(n, s + 1), s === 0 && (this.output.push(n), this.update());
    let i = !1;
    return () => {
      if (i)
        return;
      i = !0;
      const o = this.counts.get(n) - 1;
      this.counts.set(n, o), o === 0 && (Zt(this.output, n), this.update());
    };
  }
  update() {
    this.set(this.output.length ? this.output.join(", ") : "auto");
  }
}
function Xa(t) {
  return !!(E(t) && t.add);
}
function Ae(t, e) {
  var n;
  if (!t.applyWillChange)
    return;
  let s = t.getValue("willChange");
  if (!s && !(!((n = t.props.style) === null || n === void 0) && n.willChange) && (s = new Ya("auto"), t.addValue("willChange", s)), Xa(s))
    return s.add(e);
}
function qa({ protectedKeys: t, needsAnimating: e }, n) {
  const s = t.hasOwnProperty(n) && e[n] !== !0;
  return e[n] = !1, s;
}
function Li(t, e, { delay: n = 0, transitionOverride: s, type: i } = {}) {
  var o;
  let { transition: r = t.getDefaultTransition(), transitionEnd: a, ...l } = e;
  s && (r = s);
  const u = [], c = i && t.animationState && t.animationState.getState()[i];
  for (const h in l) {
    const f = t.getValue(h, (o = t.latestValues[h]) !== null && o !== void 0 ? o : null), d = l[h];
    if (d === void 0 || c && qa(c, h))
      continue;
    const p = {
      delay: n,
      elapsed: 0,
      ...Ye(r || {}, h)
    };
    let m = !1;
    if (window.HandoffAppearAnimations) {
      const g = Ei(t);
      if (g) {
        const v = window.HandoffAppearAnimations(g, h, f, w);
        v !== null && (p.elapsed = v, m = !0);
      }
    }
    f.start(sn(h, f, d, t.shouldReduceMotion && Q.has(h) ? { type: !1 } : p, t, m, Ae(t, h)));
    const x = f.animation;
    x && u.push(x);
  }
  return a && Promise.all(u).then(() => {
    w.update(() => {
      a && Ha(t, a);
    });
  }), u;
}
function Ve(t, e, n = {}) {
  var s;
  const i = Qt(t, e, n.type === "exit" ? (s = t.presenceContext) === null || s === void 0 ? void 0 : s.custom : void 0);
  let { transition: o = t.getDefaultTransition() || {} } = i || {};
  n.transitionOverride && (o = n.transitionOverride);
  const r = i ? () => Promise.all(Li(t, i, n)) : () => Promise.resolve(), a = t.variantChildren && t.variantChildren.size ? (u = 0) => {
    const { delayChildren: c = 0, staggerChildren: h, staggerDirection: f } = o;
    return Za(t, e, c + u, h, f, n);
  } : () => Promise.resolve(), { when: l } = o;
  if (l) {
    const [u, c] = l === "beforeChildren" ? [r, a] : [a, r];
    return u().then(() => c());
  } else
    return Promise.all([r(), a(n.delay)]);
}
function Za(t, e, n = 0, s = 0, i = 1, o) {
  const r = [], a = (t.variantChildren.size - 1) * s, l = i === 1 ? (u = 0) => u * s : (u = 0) => a - u * s;
  return Array.from(t.variantChildren).sort(Ja).forEach((u, c) => {
    u.notify("AnimationStart", e), r.push(Ve(u, e, {
      ...o,
      delay: n + l(c)
    }).then(() => u.notify("AnimationComplete", e)));
  }), Promise.all(r);
}
function Ja(t, e) {
  return t.sortNodePosition(e);
}
function Qa(t, e, n = {}) {
  t.notify("AnimationStart", e);
  let s;
  if (Array.isArray(e)) {
    const i = e.map((o) => Ve(t, o, n));
    s = Promise.all(i);
  } else if (typeof e == "string")
    s = Ve(t, e, n);
  else {
    const i = typeof e == "function" ? Qt(t, e, n.custom) : e;
    s = Promise.all(Li(t, i, n));
  }
  return s.then(() => {
    t.notify("AnimationComplete", e);
  });
}
const tl = [...Oe].reverse(), el = Oe.length;
function nl(t) {
  return (e) => Promise.all(e.map(({ animation: n, options: s }) => Qa(t, n, s)));
}
function sl(t) {
  let e = nl(t), n = jn(), s = !0;
  const i = (l) => (u, c) => {
    var h;
    const f = Qt(t, c, l === "exit" ? (h = t.presenceContext) === null || h === void 0 ? void 0 : h.custom : void 0);
    if (f) {
      const { transition: d, transitionEnd: p, ...m } = f;
      u = { ...u, ...m, ...p };
    }
    return u;
  };
  function o(l) {
    e = l(t);
  }
  function r(l) {
    const u = t.getProps(), c = t.getVariantContext(!0) || {}, h = [], f = /* @__PURE__ */ new Set();
    let d = {}, p = 1 / 0;
    for (let x = 0; x < el; x++) {
      const g = tl[x], v = n[g], y = u[g] !== void 0 ? u[g] : c[g], P = wt(y), S = g === l ? v.isActive : null;
      S === !1 && (p = x);
      let C = y === c[g] && y !== u[g] && P;
      if (C && s && t.manuallyAnimateOnMount && (C = !1), v.protectedKeys = { ...d }, // If it isn't active and hasn't *just* been set as inactive
      !v.isActive && S === null || // If we didn't and don't have any defined prop for this animation type
      !y && !v.prevProp || // Or if the prop doesn't define an animation
      At(y) || typeof y == "boolean")
        continue;
      let b = il(v.prevProp, y) || // If we're making this variant active, we want to always make it active
      g === l && v.isActive && !C && P || // If we removed a higher-priority variant (i is in reverse order)
      x > p && P, k = !1;
      const O = Array.isArray(y) ? y : [y];
      let ot = O.reduce(i(g), {});
      S === !1 && (ot = {});
      const { prevResolvedValues: an = {} } = v, Qi = {
        ...an,
        ...ot
      }, ln = (L) => {
        b = !0, f.has(L) && (k = !0, f.delete(L)), v.needsAnimating[L] = !0;
        const U = t.getValue(L);
        U && (U.liveStyle = !1);
      };
      for (const L in Qi) {
        const U = ot[L], ee = an[L];
        if (d.hasOwnProperty(L))
          continue;
        let ne = !1;
        ye(U) && ye(ee) ? ne = !Qs(U, ee) : ne = U !== ee, ne ? U != null ? ln(L) : f.add(L) : U !== void 0 && f.has(L) ? ln(L) : v.protectedKeys[L] = !0;
      }
      v.prevProp = y, v.prevResolvedValues = ot, v.isActive && (d = { ...d, ...ot }), s && t.blockInitialAnimation && (b = !1), b && (!C || k) && h.push(...O.map((L) => ({
        animation: L,
        options: { type: g }
      })));
    }
    if (f.size) {
      const x = {};
      f.forEach((g) => {
        const v = t.getBaseTarget(g), y = t.getValue(g);
        y && (y.liveStyle = !0), x[g] = v ?? null;
      }), h.push({ animation: x });
    }
    let m = !!h.length;
    return s && (u.initial === !1 || u.initial === u.animate) && !t.manuallyAnimateOnMount && (m = !1), s = !1, m ? e(h) : Promise.resolve();
  }
  function a(l, u) {
    var c;
    if (n[l].isActive === u)
      return Promise.resolve();
    (c = t.variantChildren) === null || c === void 0 || c.forEach((f) => {
      var d;
      return (d = f.animationState) === null || d === void 0 ? void 0 : d.setActive(l, u);
    }), n[l].isActive = u;
    const h = r(l);
    for (const f in n)
      n[f].protectedKeys = {};
    return h;
  }
  return {
    animateChanges: r,
    setActive: a,
    setAnimateFunction: o,
    getState: () => n,
    reset: () => {
      n = jn(), s = !0;
    }
  };
}
function il(t, e) {
  return typeof e == "string" ? e !== t : Array.isArray(e) ? !Qs(e, t) : !1;
}
function et(t = !1) {
  return {
    isActive: t,
    protectedKeys: {},
    needsAnimating: {},
    prevResolvedValues: {}
  };
}
function jn() {
  return {
    animate: et(!0),
    whileInView: et(),
    whileHover: et(),
    whileTap: et(),
    whileDrag: et(),
    whileFocus: et(),
    exit: et()
  };
}
class rl extends tt {
  /**
   * We dynamically generate the AnimationState manager as it contains a reference
   * to the underlying animation library. We only want to load that if we load this,
   * so people can optionally code split it out using the `m` component.
   */
  constructor(e) {
    super(e), e.animationState || (e.animationState = sl(e));
  }
  updateAnimationControlsSubscription() {
    const { animate: e } = this.node.getProps();
    At(e) && (this.unmountControls = e.subscribe(this.node));
  }
  /**
   * Subscribe any provided AnimationControls to the component's VisualElement
   */
  mount() {
    this.updateAnimationControlsSubscription();
  }
  update() {
    const { animate: e } = this.node.getProps(), { animate: n } = this.node.prevProps || {};
    e !== n && this.updateAnimationControlsSubscription();
  }
  unmount() {
    var e;
    this.node.animationState.reset(), (e = this.unmountControls) === null || e === void 0 || e.call(this);
  }
}
let ol = 0;
class al extends tt {
  constructor() {
    super(...arguments), this.id = ol++;
  }
  update() {
    if (!this.node.presenceContext)
      return;
    const { isPresent: e, onExitComplete: n } = this.node.presenceContext, { isPresent: s } = this.node.prevPresenceContext || {};
    if (!this.node.animationState || e === s)
      return;
    const i = this.node.animationState.setActive("exit", !e);
    n && !e && i.then(() => n(this.id));
  }
  mount() {
    const { register: e } = this.node.presenceContext || {};
    e && (this.unmount = e(this.id));
  }
  unmount() {
  }
}
const ll = {
  animation: {
    Feature: rl
  },
  exit: {
    Feature: al
  }
}, In = (t, e) => Math.abs(t - e);
function ul(t, e) {
  const n = In(t.x, e.x), s = In(t.y, e.y);
  return Math.sqrt(n ** 2 + s ** 2);
}
class Fi {
  constructor(e, n, { transformPagePoint: s, contextWindow: i, dragSnapToOrigin: o = !1 } = {}) {
    if (this.startEvent = null, this.lastMoveEvent = null, this.lastMoveEventInfo = null, this.handlers = {}, this.contextWindow = window, this.updatePoint = () => {
      if (!(this.lastMoveEvent && this.lastMoveEventInfo))
        return;
      const h = he(this.lastMoveEventInfo, this.history), f = this.startEvent !== null, d = ul(h.offset, { x: 0, y: 0 }) >= 3;
      if (!f && !d)
        return;
      const { point: p } = h, { timestamp: m } = D;
      this.history.push({ ...p, timestamp: m });
      const { onStart: x, onMove: g } = this.handlers;
      f || (x && x(this.lastMoveEvent, h), this.startEvent = this.lastMoveEvent), g && g(this.lastMoveEvent, h);
    }, this.handlePointerMove = (h, f) => {
      this.lastMoveEvent = h, this.lastMoveEventInfo = ce(f, this.transformPagePoint), w.update(this.updatePoint, !0);
    }, this.handlePointerUp = (h, f) => {
      this.end();
      const { onEnd: d, onSessionEnd: p, resumeAnimation: m } = this.handlers;
      if (this.dragSnapToOrigin && m && m(), !(this.lastMoveEvent && this.lastMoveEventInfo))
        return;
      const x = he(h.type === "pointercancel" ? this.lastMoveEventInfo : ce(f, this.transformPagePoint), this.history);
      this.startEvent && d && d(h, x), p && p(h, x);
    }, !Ys(e))
      return;
    this.dragSnapToOrigin = o, this.handlers = n, this.transformPagePoint = s, this.contextWindow = i || window;
    const r = Jt(e), a = ce(r, this.transformPagePoint), { point: l } = a, { timestamp: u } = D;
    this.history = [{ ...l, timestamp: u }];
    const { onSessionStart: c } = n;
    c && c(e, he(a, this.history)), this.removeListeners = K(W(this.contextWindow, "pointermove", this.handlePointerMove), W(this.contextWindow, "pointerup", this.handlePointerUp), W(this.contextWindow, "pointercancel", this.handlePointerUp));
  }
  updateHandlers(e) {
    this.handlers = e;
  }
  end() {
    this.removeListeners && this.removeListeners(), z(this.updatePoint);
  }
}
function ce(t, e) {
  return e ? { point: e(t.point) } : t;
}
function Nn(t, e) {
  return { x: t.x - e.x, y: t.y - e.y };
}
function he({ point: t }, e) {
  return {
    point: t,
    delta: Nn(t, Bi(e)),
    offset: Nn(t, cl(e)),
    velocity: hl(e, 0.1)
  };
}
function cl(t) {
  return t[0];
}
function Bi(t) {
  return t[t.length - 1];
}
function hl(t, e) {
  if (t.length < 2)
    return { x: 0, y: 0 };
  let n = t.length - 1, s = null;
  const i = Bi(t);
  for (; n >= 0 && (s = t[n], !(i.timestamp - s.timestamp > $(e))); )
    n--;
  if (!s)
    return { x: 0, y: 0 };
  const o = G(i.timestamp - s.timestamp);
  if (o === 0)
    return { x: 0, y: 0 };
  const r = {
    x: (i.x - s.x) / o,
    y: (i.y - s.y) / o
  };
  return r.x === 1 / 0 && (r.x = 0), r.y === 1 / 0 && (r.y = 0), r;
}
const ki = 1e-4, fl = 1 - ki, dl = 1 + ki, Oi = 0.01, pl = 0 - Oi, ml = 0 + Oi;
function B(t) {
  return t.max - t.min;
}
function gl(t, e, n) {
  return Math.abs(t - e) <= n;
}
function Un(t, e, n, s = 0.5) {
  t.origin = s, t.originPoint = A(e.min, e.max, t.origin), t.scale = B(n) / B(e), t.translate = A(n.min, n.max, t.origin) - t.originPoint, (t.scale >= fl && t.scale <= dl || isNaN(t.scale)) && (t.scale = 1), (t.translate >= pl && t.translate <= ml || isNaN(t.translate)) && (t.translate = 0);
}
function St(t, e, n, s) {
  Un(t.x, e.x, n.x, s ? s.originX : void 0), Un(t.y, e.y, n.y, s ? s.originY : void 0);
}
function _n(t, e, n) {
  t.min = n.min + e.min, t.max = t.min + B(e);
}
function yl(t, e, n) {
  _n(t.x, e.x, n.x), _n(t.y, e.y, n.y);
}
function Wn(t, e, n) {
  t.min = e.min - n.min, t.max = t.min + B(e);
}
function bt(t, e, n) {
  Wn(t.x, e.x, n.x), Wn(t.y, e.y, n.y);
}
function vl(t, { min: e, max: n }, s) {
  return e !== void 0 && t < e ? t = s ? A(e, t, s.min) : Math.max(t, e) : n !== void 0 && t > n && (t = s ? A(n, t, s.max) : Math.min(t, n)), t;
}
function Kn(t, e, n) {
  return {
    min: e !== void 0 ? t.min + e : void 0,
    max: n !== void 0 ? t.max + n - (t.max - t.min) : void 0
  };
}
function xl(t, { top: e, left: n, bottom: s, right: i }) {
  return {
    x: Kn(t.x, n, i),
    y: Kn(t.y, e, s)
  };
}
function $n(t, e) {
  let n = e.min - t.min, s = e.max - t.max;
  return e.max - e.min < t.max - t.min && ([n, s] = [s, n]), { min: n, max: s };
}
function Tl(t, e) {
  return {
    x: $n(t.x, e.x),
    y: $n(t.y, e.y)
  };
}
function Pl(t, e) {
  let n = 0.5;
  const s = B(t), i = B(e);
  return i > s ? n = Ct(e.min, e.max - s, t.min) : s > i && (n = Ct(t.min, t.max - i, e.min)), Z(0, 1, n);
}
function Sl(t, e) {
  const n = {};
  return e.min !== void 0 && (n.min = e.min - t.min), e.max !== void 0 && (n.max = e.max - t.min), n;
}
const Ce = 0.35;
function bl(t = Ce) {
  return t === !1 ? t = 0 : t === !0 && (t = Ce), {
    x: Gn(t, "left", "right"),
    y: Gn(t, "top", "bottom")
  };
}
function Gn(t, e, n) {
  return {
    min: zn(t, e),
    max: zn(t, n)
  };
}
function zn(t, e) {
  return typeof t == "number" ? t : t[e] || 0;
}
const Hn = () => ({
  translate: 0,
  scale: 1,
  origin: 0,
  originPoint: 0
}), ut = () => ({
  x: Hn(),
  y: Hn()
}), Yn = () => ({ min: 0, max: 0 }), V = () => ({
  x: Yn(),
  y: Yn()
});
function I(t) {
  return [t("x"), t("y")];
}
function ji({ top: t, left: e, right: n, bottom: s }) {
  return {
    x: { min: e, max: n },
    y: { min: t, max: s }
  };
}
function wl({ x: t, y: e }) {
  return { top: e.min, right: t.max, bottom: e.max, left: t.min };
}
function Al(t, e) {
  if (!e)
    return t;
  const n = e({ x: t.left, y: t.top }), s = e({ x: t.right, y: t.bottom });
  return {
    top: n.y,
    left: n.x,
    bottom: s.y,
    right: s.x
  };
}
function fe(t) {
  return t === void 0 || t === 1;
}
function De({ scale: t, scaleX: e, scaleY: n }) {
  return !fe(t) || !fe(e) || !fe(n);
}
function nt(t) {
  return De(t) || Ii(t) || t.z || t.rotate || t.rotateX || t.rotateY || t.skewX || t.skewY;
}
function Ii(t) {
  return Xn(t.x) || Xn(t.y);
}
function Xn(t) {
  return t && t !== "0%";
}
function Gt(t, e, n) {
  const s = t - n, i = e * s;
  return n + i;
}
function qn(t, e, n, s, i) {
  return i !== void 0 && (t = Gt(t, i, s)), Gt(t, n, s) + e;
}
function Me(t, e = 0, n = 1, s, i) {
  t.min = qn(t.min, e, n, s, i), t.max = qn(t.max, e, n, s, i);
}
function Ni(t, { x: e, y: n }) {
  Me(t.x, e.translate, e.scale, e.originPoint), Me(t.y, n.translate, n.scale, n.originPoint);
}
const Zn = 0.999999999999, Jn = 1.0000000000001;
function Vl(t, e, n, s = !1) {
  const i = n.length;
  if (!i)
    return;
  e.x = e.y = 1;
  let o, r;
  for (let a = 0; a < i; a++) {
    o = n[a], r = o.projectionDelta;
    const { visualElement: l } = o.options;
    l && l.props.style && l.props.style.display === "contents" || (s && o.options.layoutScroll && o.scroll && o !== o.root && ht(t, {
      x: -o.scroll.offset.x,
      y: -o.scroll.offset.y
    }), r && (e.x *= r.x.scale, e.y *= r.y.scale, Ni(t, r)), s && nt(o.latestValues) && ht(t, o.latestValues));
  }
  e.x < Jn && e.x > Zn && (e.x = 1), e.y < Jn && e.y > Zn && (e.y = 1);
}
function ct(t, e) {
  t.min = t.min + e, t.max = t.max + e;
}
function Qn(t, e, n, s, i = 0.5) {
  const o = A(t.min, t.max, i);
  Me(t, e, n, o, s);
}
function ht(t, e) {
  Qn(t.x, e.x, e.scaleX, e.scale, e.originX), Qn(t.y, e.y, e.scaleY, e.scale, e.originY);
}
function Ui(t, e) {
  return ji(Al(t.getBoundingClientRect(), e));
}
function Cl(t, e, n) {
  const s = Ui(t, n), { scroll: i } = e;
  return i && (ct(s.x, i.offset.x), ct(s.y, i.offset.y)), s;
}
const _i = ({ current: t }) => t ? t.ownerDocument.defaultView : null, Dl = /* @__PURE__ */ new WeakMap();
class Ml {
  constructor(e) {
    this.openGlobalLock = null, this.isDragging = !1, this.currentDirection = null, this.originPoint = { x: 0, y: 0 }, this.constraints = !1, this.hasMutatedConstraints = !1, this.elastic = V(), this.visualElement = e;
  }
  start(e, { snapToCursor: n = !1 } = {}) {
    const { presenceContext: s } = this.visualElement;
    if (s && s.isPresent === !1)
      return;
    const i = (c) => {
      const { dragSnapToOrigin: h } = this.getProps();
      h ? this.pauseAnimation() : this.stopAnimation(), n && this.snapToCursor(Jt(c, "page").point);
    }, o = (c, h) => {
      var f;
      const { drag: d, dragPropagation: p, onDragStart: m } = this.getProps();
      if (d && !p && (this.openGlobalLock && this.openGlobalLock(), this.openGlobalLock = qs(d), !this.openGlobalLock))
        return;
      this.isDragging = !0, this.currentDirection = null, this.resolveConstraints(), this.visualElement.projection && (this.visualElement.projection.isAnimationBlocked = !0, this.visualElement.projection.target = void 0), I((g) => {
        let v = this.getAxisMotionValue(g).get() || 0;
        if (N.test(v)) {
          const { projection: y } = this.visualElement;
          if (y && y.layout) {
            const P = y.layout.layoutBox[g];
            P && (v = B(P) * (parseFloat(v) / 100));
          }
        }
        this.originPoint[g] = v;
      }), m && w.postRender(() => m(c, h)), (f = this.removeWillChange) === null || f === void 0 || f.call(this), this.removeWillChange = Ae(this.visualElement, "transform");
      const { animationState: x } = this.visualElement;
      x && x.setActive("whileDrag", !0);
    }, r = (c, h) => {
      const { dragPropagation: f, dragDirectionLock: d, onDirectionLock: p, onDrag: m } = this.getProps();
      if (!f && !this.openGlobalLock)
        return;
      const { offset: x } = h;
      if (d && this.currentDirection === null) {
        this.currentDirection = Rl(x), this.currentDirection !== null && p && p(this.currentDirection);
        return;
      }
      this.updateAxis("x", h.point, x), this.updateAxis("y", h.point, x), this.visualElement.render(), m && m(c, h);
    }, a = (c, h) => this.stop(c, h), l = () => I((c) => {
      var h;
      return this.getAnimationState(c) === "paused" && ((h = this.getAxisMotionValue(c).animation) === null || h === void 0 ? void 0 : h.play());
    }), { dragSnapToOrigin: u } = this.getProps();
    this.panSession = new Fi(e, {
      onSessionStart: i,
      onStart: o,
      onMove: r,
      onSessionEnd: a,
      resumeAnimation: l
    }, {
      transformPagePoint: this.visualElement.getTransformPagePoint(),
      dragSnapToOrigin: u,
      contextWindow: _i(this.visualElement)
    });
  }
  stop(e, n) {
    var s;
    (s = this.removeWillChange) === null || s === void 0 || s.call(this);
    const i = this.isDragging;
    if (this.cancel(), !i)
      return;
    const { velocity: o } = n;
    this.startAnimation(o);
    const { onDragEnd: r } = this.getProps();
    r && w.postRender(() => r(e, n));
  }
  cancel() {
    this.isDragging = !1;
    const { projection: e, animationState: n } = this.visualElement;
    e && (e.isAnimationBlocked = !1), this.panSession && this.panSession.end(), this.panSession = void 0;
    const { dragPropagation: s } = this.getProps();
    !s && this.openGlobalLock && (this.openGlobalLock(), this.openGlobalLock = null), n && n.setActive("whileDrag", !1);
  }
  updateAxis(e, n, s) {
    const { drag: i } = this.getProps();
    if (!s || !kt(e, i, this.currentDirection))
      return;
    const o = this.getAxisMotionValue(e);
    let r = this.originPoint[e] + s[e];
    this.constraints && this.constraints[e] && (r = vl(r, this.constraints[e], this.elastic[e])), o.set(r);
  }
  resolveConstraints() {
    var e;
    const { dragConstraints: n, dragElastic: s } = this.getProps(), i = this.visualElement.projection && !this.visualElement.projection.layout ? this.visualElement.projection.measure(!1) : (e = this.visualElement.projection) === null || e === void 0 ? void 0 : e.layout, o = this.constraints;
    n && at(n) ? this.constraints || (this.constraints = this.resolveRefConstraints()) : n && i ? this.constraints = xl(i.layoutBox, n) : this.constraints = !1, this.elastic = bl(s), o !== this.constraints && i && this.constraints && !this.hasMutatedConstraints && I((r) => {
      this.constraints !== !1 && this.getAxisMotionValue(r) && (this.constraints[r] = Sl(i.layoutBox[r], this.constraints[r]));
    });
  }
  resolveRefConstraints() {
    const { dragConstraints: e, onMeasureDragConstraints: n } = this.getProps();
    if (!e || !at(e))
      return !1;
    const s = e.current;
    q(s !== null, "If `dragConstraints` is set as a React ref, that ref must be passed to another component's `ref` prop.");
    const { projection: i } = this.visualElement;
    if (!i || !i.layout)
      return !1;
    const o = Cl(s, i.root, this.visualElement.getTransformPagePoint());
    let r = Tl(i.layout.layoutBox, o);
    if (n) {
      const a = n(wl(r));
      this.hasMutatedConstraints = !!a, a && (r = ji(a));
    }
    return r;
  }
  startAnimation(e) {
    const { drag: n, dragMomentum: s, dragElastic: i, dragTransition: o, dragSnapToOrigin: r, onDragTransitionEnd: a } = this.getProps(), l = this.constraints || {}, u = I((c) => {
      if (!kt(c, n, this.currentDirection))
        return;
      let h = l && l[c] || {};
      r && (h = { min: 0, max: 0 });
      const f = i ? 200 : 1e6, d = i ? 40 : 1e7, p = {
        type: "inertia",
        velocity: s ? e[c] : 0,
        bounceStiffness: f,
        bounceDamping: d,
        timeConstant: 750,
        restDelta: 1,
        restSpeed: 10,
        ...o,
        ...h
      };
      return this.startAxisValueAnimation(c, p);
    });
    return Promise.all(u).then(a);
  }
  startAxisValueAnimation(e, n) {
    const s = this.getAxisMotionValue(e);
    return s.start(sn(e, s, 0, n, this.visualElement, !1, Ae(this.visualElement, e)));
  }
  stopAnimation() {
    I((e) => this.getAxisMotionValue(e).stop());
  }
  pauseAnimation() {
    I((e) => {
      var n;
      return (n = this.getAxisMotionValue(e).animation) === null || n === void 0 ? void 0 : n.pause();
    });
  }
  getAnimationState(e) {
    var n;
    return (n = this.getAxisMotionValue(e).animation) === null || n === void 0 ? void 0 : n.state;
  }
  /**
   * Drag works differently depending on which props are provided.
   *
   * - If _dragX and _dragY are provided, we output the gesture delta directly to those motion values.
   * - Otherwise, we apply the delta to the x/y motion values.
   */
  getAxisMotionValue(e) {
    const n = `_drag${e.toUpperCase()}`, s = this.visualElement.getProps(), i = s[n];
    return i || this.visualElement.getValue(e, (s.initial ? s.initial[e] : void 0) || 0);
  }
  snapToCursor(e) {
    I((n) => {
      const { drag: s } = this.getProps();
      if (!kt(n, s, this.currentDirection))
        return;
      const { projection: i } = this.visualElement, o = this.getAxisMotionValue(n);
      if (i && i.layout) {
        const { min: r, max: a } = i.layout.layoutBox[n];
        o.set(e[n] - A(r, a, 0.5));
      }
    });
  }
  /**
   * When the viewport resizes we want to check if the measured constraints
   * have changed and, if so, reposition the element within those new constraints
   * relative to where it was before the resize.
   */
  scalePositionWithinConstraints() {
    if (!this.visualElement.current)
      return;
    const { drag: e, dragConstraints: n } = this.getProps(), { projection: s } = this.visualElement;
    if (!at(n) || !s || !this.constraints)
      return;
    this.stopAnimation();
    const i = { x: 0, y: 0 };
    I((r) => {
      const a = this.getAxisMotionValue(r);
      if (a && this.constraints !== !1) {
        const l = a.get();
        i[r] = Pl({ min: l, max: l }, this.constraints[r]);
      }
    });
    const { transformTemplate: o } = this.visualElement.getProps();
    this.visualElement.current.style.transform = o ? o({}, "") : "none", s.root && s.root.updateScroll(), s.updateLayout(), this.resolveConstraints(), I((r) => {
      if (!kt(r, e, null))
        return;
      const a = this.getAxisMotionValue(r), { min: l, max: u } = this.constraints[r];
      a.set(A(l, u, i[r]));
    });
  }
  addListeners() {
    if (!this.visualElement.current)
      return;
    Dl.set(this.visualElement, this);
    const e = this.visualElement.current, n = W(e, "pointerdown", (l) => {
      const { drag: u, dragListener: c = !0 } = this.getProps();
      u && c && this.start(l);
    }), s = () => {
      const { dragConstraints: l } = this.getProps();
      at(l) && l.current && (this.constraints = this.resolveRefConstraints());
    }, { projection: i } = this.visualElement, o = i.addEventListener("measure", s);
    i && !i.layout && (i.root && i.root.updateScroll(), i.updateLayout()), w.read(s);
    const r = _(window, "resize", () => this.scalePositionWithinConstraints()), a = i.addEventListener("didUpdate", ({ delta: l, hasLayoutChanged: u }) => {
      this.isDragging && u && (I((c) => {
        const h = this.getAxisMotionValue(c);
        h && (this.originPoint[c] += l[c].translate, h.set(h.get() + l[c].translate));
      }), this.visualElement.render());
    });
    return () => {
      r(), n(), o(), a && a();
    };
  }
  getProps() {
    const e = this.visualElement.getProps(), { drag: n = !1, dragDirectionLock: s = !1, dragPropagation: i = !1, dragConstraints: o = !1, dragElastic: r = Ce, dragMomentum: a = !0 } = e;
    return {
      ...e,
      drag: n,
      dragDirectionLock: s,
      dragPropagation: i,
      dragConstraints: o,
      dragElastic: r,
      dragMomentum: a
    };
  }
}
function kt(t, e, n) {
  return (e === !0 || e === t) && (n === null || n === t);
}
function Rl(t, e = 10) {
  let n = null;
  return Math.abs(t.y) > e ? n = "y" : Math.abs(t.x) > e && (n = "x"), n;
}
class El extends tt {
  constructor(e) {
    super(e), this.removeGroupControls = M, this.removeListeners = M, this.controls = new Ml(e);
  }
  mount() {
    const { dragControls: e } = this.node.getProps();
    e && (this.removeGroupControls = e.subscribe(this.controls)), this.removeListeners = this.controls.addListeners() || M;
  }
  unmount() {
    this.removeGroupControls(), this.removeListeners();
  }
}
const ts = (t) => (e, n) => {
  t && w.postRender(() => t(e, n));
};
class Ll extends tt {
  constructor() {
    super(...arguments), this.removePointerDownListener = M;
  }
  onPointerDown(e) {
    this.session = new Fi(e, this.createPanHandlers(), {
      transformPagePoint: this.node.getTransformPagePoint(),
      contextWindow: _i(this.node)
    });
  }
  createPanHandlers() {
    const { onPanSessionStart: e, onPanStart: n, onPan: s, onPanEnd: i } = this.node.getProps();
    return {
      onSessionStart: ts(e),
      onStart: ts(n),
      onMove: s,
      onEnd: (o, r) => {
        delete this.session, i && w.postRender(() => i(o, r));
      }
    };
  }
  mount() {
    this.removePointerDownListener = W(this.node.current, "pointerdown", (e) => this.onPointerDown(e));
  }
  update() {
    this.session && this.session.updateHandlers(this.createPanHandlers());
  }
  unmount() {
    this.removePointerDownListener(), this.session && this.session.end();
  }
}
function Fl() {
  const t = F(Fe);
  if (t === null)
    return [!0, null];
  const { isPresent: e, onExitComplete: n, register: s } = t, i = rr();
  Ee(() => s(i), []);
  const o = As(() => n && n(i), [i, n]);
  return !e && n ? [!1, o] : [!0];
}
const It = {
  /**
   * Global flag as to whether the tree has animated since the last time
   * we resized the window
   */
  hasAnimatedSinceResize: !0,
  /**
   * We set this to true once, on the first update. Any nodes added to the tree beyond that
   * update will be given a `data-projection-id` attribute.
   */
  hasEverUpdated: !1
};
function es(t, e) {
  return e.max === e.min ? 0 : t / (e.max - e.min) * 100;
}
const yt = {
  correct: (t, e) => {
    if (!e.target)
      return t;
    if (typeof t == "string")
      if (T.test(t))
        t = parseFloat(t);
      else
        return t;
    const n = es(t, e.target.x), s = es(t, e.target.y);
    return `${n}% ${s}%`;
  }
}, Bl = {
  correct: (t, { treeScale: e, projectionDelta: n }) => {
    const s = t, i = J.parse(t);
    if (i.length > 5)
      return s;
    const o = J.createTransformer(t), r = typeof i[0] != "number" ? 1 : 0, a = n.x.scale * e.x, l = n.y.scale * e.y;
    i[0 + r] /= a, i[1 + r] /= l;
    const u = A(a, l, 0.5);
    return typeof i[2 + r] == "number" && (i[2 + r] /= u), typeof i[3 + r] == "number" && (i[3 + r] /= u), o(i);
  }
};
class kl extends or {
  /**
   * This only mounts projection nodes for components that
   * need measuring, we might want to do it for all components
   * in order to incorporate transforms
   */
  componentDidMount() {
    const { visualElement: e, layoutGroup: n, switchLayoutGroup: s, layoutId: i } = this.props, { projection: o } = e;
    Cr(Ol), o && (n.group && n.group.add(o), s && s.register && i && s.register(o), o.root.didUpdate(), o.addEventListener("animationComplete", () => {
      this.safeToRemove();
    }), o.setOptions({
      ...o.options,
      onExitComplete: () => this.safeToRemove()
    })), It.hasEverUpdated = !0;
  }
  getSnapshotBeforeUpdate(e) {
    const { layoutDependency: n, visualElement: s, drag: i, isPresent: o } = this.props, r = s.projection;
    return r && (r.isPresent = o, i || e.layoutDependency !== n || n === void 0 ? r.willUpdate() : this.safeToRemove(), e.isPresent !== o && (o ? r.promote() : r.relegate() || w.postRender(() => {
      const a = r.getStack();
      (!a || !a.members.length) && this.safeToRemove();
    }))), null;
  }
  componentDidUpdate() {
    const { projection: e } = this.props.visualElement;
    e && (e.root.didUpdate(), ke.postRender(() => {
      !e.currentAnimation && e.isLead() && this.safeToRemove();
    }));
  }
  componentWillUnmount() {
    const { visualElement: e, layoutGroup: n, switchLayoutGroup: s } = this.props, { projection: i } = e;
    i && (i.scheduleCheckAfterUnmount(), n && n.group && n.group.remove(i), s && s.deregister && s.deregister(i));
  }
  safeToRemove() {
    const { safeToRemove: e } = this.props;
    e && e();
  }
  render() {
    return null;
  }
}
function Wi(t) {
  const [e, n] = Fl(), s = F(Fs);
  return Nt(kl, { ...t, layoutGroup: s, switchLayoutGroup: F(Rs), isPresent: e, safeToRemove: n });
}
const Ol = {
  borderRadius: {
    ...yt,
    applyTo: [
      "borderTopLeftRadius",
      "borderTopRightRadius",
      "borderBottomLeftRadius",
      "borderBottomRightRadius"
    ]
  },
  borderTopLeftRadius: yt,
  borderTopRightRadius: yt,
  borderBottomLeftRadius: yt,
  borderBottomRightRadius: yt,
  boxShadow: Bl
}, Ki = ["TopLeft", "TopRight", "BottomLeft", "BottomRight"], jl = Ki.length, ns = (t) => typeof t == "string" ? parseFloat(t) : t, ss = (t) => typeof t == "number" || T.test(t);
function Il(t, e, n, s, i, o) {
  i ? (t.opacity = A(
    0,
    // TODO Reinstate this if only child
    n.opacity !== void 0 ? n.opacity : 1,
    Nl(s)
  ), t.opacityExit = A(e.opacity !== void 0 ? e.opacity : 1, 0, Ul(s))) : o && (t.opacity = A(e.opacity !== void 0 ? e.opacity : 1, n.opacity !== void 0 ? n.opacity : 1, s));
  for (let r = 0; r < jl; r++) {
    const a = `border${Ki[r]}Radius`;
    let l = is(e, a), u = is(n, a);
    if (l === void 0 && u === void 0)
      continue;
    l || (l = 0), u || (u = 0), l === 0 || u === 0 || ss(l) === ss(u) ? (t[a] = Math.max(A(ns(l), ns(u), s), 0), (N.test(u) || N.test(l)) && (t[a] += "%")) : t[a] = u;
  }
  (e.rotate || n.rotate) && (t.rotate = A(e.rotate || 0, n.rotate || 0, s));
}
function is(t, e) {
  return t[e] !== void 0 ? t[e] : t.borderRadius;
}
const Nl = $i(0, 0.5, bi), Ul = $i(0.5, 0.95, M);
function $i(t, e, n) {
  return (s) => s < t ? 0 : s > e ? 1 : n(Ct(t, e, s));
}
function rs(t, e) {
  t.min = e.min, t.max = e.max;
}
function j(t, e) {
  rs(t.x, e.x), rs(t.y, e.y);
}
function os(t, e) {
  t.translate = e.translate, t.scale = e.scale, t.originPoint = e.originPoint, t.origin = e.origin;
}
function as(t, e, n, s, i) {
  return t -= e, t = Gt(t, 1 / n, s), i !== void 0 && (t = Gt(t, 1 / i, s)), t;
}
function _l(t, e = 0, n = 1, s = 0.5, i, o = t, r = t) {
  if (N.test(e) && (e = parseFloat(e), e = A(r.min, r.max, e / 100) - r.min), typeof e != "number")
    return;
  let a = A(o.min, o.max, s);
  t === o && (a -= e), t.min = as(t.min, e, n, a, i), t.max = as(t.max, e, n, a, i);
}
function ls(t, e, [n, s, i], o, r) {
  _l(t, e[n], e[s], e[i], e.scale, o, r);
}
const Wl = ["x", "scaleX", "originX"], Kl = ["y", "scaleY", "originY"];
function us(t, e, n, s) {
  ls(t.x, e, Wl, n ? n.x : void 0, s ? s.x : void 0), ls(t.y, e, Kl, n ? n.y : void 0, s ? s.y : void 0);
}
function cs(t) {
  return t.translate === 0 && t.scale === 1;
}
function Gi(t) {
  return cs(t.x) && cs(t.y);
}
function hs(t, e) {
  return t.min === e.min && t.max === e.max;
}
function $l(t, e) {
  return hs(t.x, e.x) && hs(t.y, e.y);
}
function fs(t, e) {
  return Math.round(t.min) === Math.round(e.min) && Math.round(t.max) === Math.round(e.max);
}
function zi(t, e) {
  return fs(t.x, e.x) && fs(t.y, e.y);
}
function ds(t) {
  return B(t.x) / B(t.y);
}
function ps(t, e) {
  return t.translate === e.translate && t.scale === e.scale && t.originPoint === e.originPoint;
}
class Gl {
  constructor() {
    this.members = [];
  }
  add(e) {
    qt(this.members, e), e.scheduleRender();
  }
  remove(e) {
    if (Zt(this.members, e), e === this.prevLead && (this.prevLead = void 0), e === this.lead) {
      const n = this.members[this.members.length - 1];
      n && this.promote(n);
    }
  }
  relegate(e) {
    const n = this.members.findIndex((i) => e === i);
    if (n === 0)
      return !1;
    let s;
    for (let i = n; i >= 0; i--) {
      const o = this.members[i];
      if (o.isPresent !== !1) {
        s = o;
        break;
      }
    }
    return s ? (this.promote(s), !0) : !1;
  }
  promote(e, n) {
    const s = this.lead;
    if (e !== s && (this.prevLead = s, this.lead = e, e.show(), s)) {
      s.instance && s.scheduleRender(), e.scheduleRender(), e.resumeFrom = s, n && (e.resumeFrom.preserveOpacity = !0), s.snapshot && (e.snapshot = s.snapshot, e.snapshot.latestValues = s.animationValues || s.latestValues), e.root && e.root.isUpdating && (e.isLayoutDirty = !0);
      const { crossfade: i } = e.options;
      i === !1 && s.hide();
    }
  }
  exitAnimationComplete() {
    this.members.forEach((e) => {
      const { options: n, resumingFrom: s } = e;
      n.onExitComplete && n.onExitComplete(), s && s.options.onExitComplete && s.options.onExitComplete();
    });
  }
  scheduleRender() {
    this.members.forEach((e) => {
      e.instance && e.scheduleRender(!1);
    });
  }
  /**
   * Clear any leads that have been removed this render to prevent them from being
   * used in future animations and to prevent memory leaks
   */
  removeLeadSnapshot() {
    this.lead && this.lead.snapshot && (this.lead.snapshot = void 0);
  }
}
function zl(t, e, n) {
  let s = "";
  const i = t.x.translate / e.x, o = t.y.translate / e.y, r = (n == null ? void 0 : n.z) || 0;
  if ((i || o || r) && (s = `translate3d(${i}px, ${o}px, ${r}px) `), (e.x !== 1 || e.y !== 1) && (s += `scale(${1 / e.x}, ${1 / e.y}) `), n) {
    const { transformPerspective: u, rotate: c, rotateX: h, rotateY: f, skewX: d, skewY: p } = n;
    u && (s = `perspective(${u}px) ${s}`), c && (s += `rotate(${c}deg) `), h && (s += `rotateX(${h}deg) `), f && (s += `rotateY(${f}deg) `), d && (s += `skewX(${d}deg) `), p && (s += `skewY(${p}deg) `);
  }
  const a = t.x.scale * e.x, l = t.y.scale * e.y;
  return (a !== 1 || l !== 1) && (s += `scale(${a}, ${l})`), s || "none";
}
const Hl = (t, e) => t.depth - e.depth;
class Yl {
  constructor() {
    this.children = [], this.isDirty = !1;
  }
  add(e) {
    qt(this.children, e), this.isDirty = !0;
  }
  remove(e) {
    Zt(this.children, e), this.isDirty = !0;
  }
  forEach(e) {
    this.isDirty && this.children.sort(Hl), this.isDirty = !1, this.children.forEach(e);
  }
}
function Xl(t, e) {
  const n = X.now(), s = ({ timestamp: i }) => {
    const o = i - n;
    o >= e && (z(s), t(o - e));
  };
  return w.read(s, !0), () => z(s);
}
function ql(t) {
  return t instanceof SVGElement && t.tagName !== "svg";
}
function Zl(t, e, n) {
  const s = E(t) ? t : Dt(t);
  return s.start(sn("", s, e, n)), s.animation;
}
const st = {
  type: "projectionFrame",
  totalNodes: 0,
  resolvedTargetDeltas: 0,
  recalculatedProjection: 0
}, xt = typeof window < "u" && window.MotionDebug !== void 0, de = ["", "X", "Y", "Z"], Jl = { visibility: "hidden" }, ms = 1e3;
let Ql = 0;
function pe(t, e, n, s) {
  const { latestValues: i } = e;
  i[t] && (n[t] = i[t], e.setStaticValue(t, 0), s && (s[t] = 0));
}
function Hi(t) {
  if (t.hasCheckedOptimisedAppear = !0, t.root === t)
    return !1;
  const { visualElement: e } = t.options;
  return e ? Ei(e) ? !0 : t.parent && !t.parent.hasCheckedOptimisedAppear ? Hi(t.parent) : !1 : !1;
}
function Yi({ attachResizeListener: t, defaultParent: e, measureScroll: n, checkIsScrollRoot: s, resetTransform: i }) {
  return class {
    constructor(r = {}, a = e == null ? void 0 : e()) {
      this.id = Ql++, this.animationId = 0, this.children = /* @__PURE__ */ new Set(), this.options = {}, this.isTreeAnimating = !1, this.isAnimationBlocked = !1, this.isLayoutDirty = !1, this.isProjectionDirty = !1, this.isSharedProjectionDirty = !1, this.isTransformDirty = !1, this.updateManuallyBlocked = !1, this.updateBlockedByResize = !1, this.isUpdating = !1, this.isSVG = !1, this.needsReset = !1, this.shouldResetTransform = !1, this.hasCheckedOptimisedAppear = !1, this.treeScale = { x: 1, y: 1 }, this.eventHandlers = /* @__PURE__ */ new Map(), this.hasTreeAnimated = !1, this.updateScheduled = !1, this.scheduleUpdate = () => this.update(), this.projectionUpdateScheduled = !1, this.checkUpdateFailed = () => {
        this.isUpdating && (this.isUpdating = !1, this.clearAllSnapshots());
      }, this.updateProjection = () => {
        this.projectionUpdateScheduled = !1, xt && (st.totalNodes = st.resolvedTargetDeltas = st.recalculatedProjection = 0), this.nodes.forEach(nu), this.nodes.forEach(au), this.nodes.forEach(lu), this.nodes.forEach(su), xt && window.MotionDebug.record(st);
      }, this.resolvedRelativeTargetAt = 0, this.hasProjected = !1, this.isVisible = !0, this.animationProgress = 0, this.sharedNodes = /* @__PURE__ */ new Map(), this.latestValues = r, this.root = a ? a.root || a : this, this.path = a ? [...a.path, a] : [], this.parent = a, this.depth = a ? a.depth + 1 : 0;
      for (let l = 0; l < this.path.length; l++)
        this.path[l].shouldResetTransform = !0;
      this.root === this && (this.nodes = new Yl());
    }
    addEventListener(r, a) {
      return this.eventHandlers.has(r) || this.eventHandlers.set(r, new rn()), this.eventHandlers.get(r).add(a);
    }
    notifyListeners(r, ...a) {
      const l = this.eventHandlers.get(r);
      l && l.notify(...a);
    }
    hasListeners(r) {
      return this.eventHandlers.has(r);
    }
    /**
     * Lifecycles
     */
    mount(r, a = this.root.hasTreeAnimated) {
      if (this.instance)
        return;
      this.isSVG = ql(r), this.instance = r;
      const { layoutId: l, layout: u, visualElement: c } = this.options;
      if (c && !c.current && c.mount(r), this.root.nodes.add(this), this.parent && this.parent.children.add(this), a && (u || l) && (this.isLayoutDirty = !0), t) {
        let h;
        const f = () => this.root.updateBlockedByResize = !1;
        t(r, () => {
          this.root.updateBlockedByResize = !0, h && h(), h = Xl(f, 250), It.hasAnimatedSinceResize && (It.hasAnimatedSinceResize = !1, this.nodes.forEach(ys));
        });
      }
      l && this.root.registerSharedNode(l, this), this.options.animate !== !1 && c && (l || u) && this.addEventListener("didUpdate", ({ delta: h, hasLayoutChanged: f, hasRelativeTargetChanged: d, layout: p }) => {
        if (this.isTreeAnimationBlocked()) {
          this.target = void 0, this.relativeTarget = void 0;
          return;
        }
        const m = this.options.transition || c.getDefaultTransition() || du, { onLayoutAnimationStart: x, onLayoutAnimationComplete: g } = c.getProps(), v = !this.targetLayout || !zi(this.targetLayout, p) || d, y = !f && d;
        if (this.options.layoutRoot || this.resumeFrom && this.resumeFrom.instance || y || f && (v || !this.currentAnimation)) {
          this.resumeFrom && (this.resumingFrom = this.resumeFrom, this.resumingFrom.resumingFrom = void 0), this.setAnimationOrigin(h, y);
          const P = {
            ...Ye(m, "layout"),
            onPlay: x,
            onComplete: g
          };
          (c.shouldReduceMotion || this.options.layoutRoot) && (P.delay = 0, P.type = !1), this.startAnimation(P);
        } else
          f || ys(this), this.isLead() && this.options.onExitComplete && this.options.onExitComplete();
        this.targetLayout = p;
      });
    }
    unmount() {
      this.options.layoutId && this.willUpdate(), this.root.nodes.remove(this);
      const r = this.getStack();
      r && r.remove(this), this.parent && this.parent.children.delete(this), this.instance = void 0, z(this.updateProjection);
    }
    // only on the root
    blockUpdate() {
      this.updateManuallyBlocked = !0;
    }
    unblockUpdate() {
      this.updateManuallyBlocked = !1;
    }
    isUpdateBlocked() {
      return this.updateManuallyBlocked || this.updateBlockedByResize;
    }
    isTreeAnimationBlocked() {
      return this.isAnimationBlocked || this.parent && this.parent.isTreeAnimationBlocked() || !1;
    }
    // Note: currently only running on root node
    startUpdate() {
      this.isUpdateBlocked() || (this.isUpdating = !0, this.nodes && this.nodes.forEach(uu), this.animationId++);
    }
    getTransformTemplate() {
      const { visualElement: r } = this.options;
      return r && r.getProps().transformTemplate;
    }
    willUpdate(r = !0) {
      if (this.root.hasTreeAnimated = !0, this.root.isUpdateBlocked()) {
        this.options.onExitComplete && this.options.onExitComplete();
        return;
      }
      if (window.HandoffCancelAllAnimations && Hi(this) && window.HandoffCancelAllAnimations(), !this.root.isUpdating && this.root.startUpdate(), this.isLayoutDirty)
        return;
      this.isLayoutDirty = !0;
      for (let c = 0; c < this.path.length; c++) {
        const h = this.path[c];
        h.shouldResetTransform = !0, h.updateScroll("snapshot"), h.options.layoutRoot && h.willUpdate(!1);
      }
      const { layoutId: a, layout: l } = this.options;
      if (a === void 0 && !l)
        return;
      const u = this.getTransformTemplate();
      this.prevTransformTemplateValue = u ? u(this.latestValues, "") : void 0, this.updateSnapshot(), r && this.notifyListeners("willUpdate");
    }
    update() {
      if (this.updateScheduled = !1, this.isUpdateBlocked()) {
        this.unblockUpdate(), this.clearAllSnapshots(), this.nodes.forEach(gs);
        return;
      }
      this.isUpdating || this.nodes.forEach(ru), this.isUpdating = !1, this.nodes.forEach(ou), this.nodes.forEach(tu), this.nodes.forEach(eu), this.clearAllSnapshots();
      const a = X.now();
      D.delta = Z(0, 1e3 / 60, a - D.timestamp), D.timestamp = a, D.isProcessing = !0, se.update.process(D), se.preRender.process(D), se.render.process(D), D.isProcessing = !1;
    }
    didUpdate() {
      this.updateScheduled || (this.updateScheduled = !0, ke.read(this.scheduleUpdate));
    }
    clearAllSnapshots() {
      this.nodes.forEach(iu), this.sharedNodes.forEach(cu);
    }
    scheduleUpdateProjection() {
      this.projectionUpdateScheduled || (this.projectionUpdateScheduled = !0, w.preRender(this.updateProjection, !1, !0));
    }
    scheduleCheckAfterUnmount() {
      w.postRender(() => {
        this.isLayoutDirty ? this.root.didUpdate() : this.root.checkUpdateFailed();
      });
    }
    /**
     * Update measurements
     */
    updateSnapshot() {
      this.snapshot || !this.instance || (this.snapshot = this.measure());
    }
    updateLayout() {
      if (!this.instance || (this.updateScroll(), !(this.options.alwaysMeasureLayout && this.isLead()) && !this.isLayoutDirty))
        return;
      if (this.resumeFrom && !this.resumeFrom.instance)
        for (let l = 0; l < this.path.length; l++)
          this.path[l].updateScroll();
      const r = this.layout;
      this.layout = this.measure(!1), this.layoutCorrected = V(), this.isLayoutDirty = !1, this.projectionDelta = void 0, this.notifyListeners("measure", this.layout.layoutBox);
      const { visualElement: a } = this.options;
      a && a.notify("LayoutMeasure", this.layout.layoutBox, r ? r.layoutBox : void 0);
    }
    updateScroll(r = "measure") {
      let a = !!(this.options.layoutScroll && this.instance);
      if (this.scroll && this.scroll.animationId === this.root.animationId && this.scroll.phase === r && (a = !1), a) {
        const l = s(this.instance);
        this.scroll = {
          animationId: this.root.animationId,
          phase: r,
          isRoot: l,
          offset: n(this.instance),
          wasRoot: this.scroll ? this.scroll.isRoot : l
        };
      }
    }
    resetTransform() {
      if (!i)
        return;
      const r = this.isLayoutDirty || this.shouldResetTransform || this.options.alwaysMeasureLayout, a = this.projectionDelta && !Gi(this.projectionDelta), l = this.getTransformTemplate(), u = l ? l(this.latestValues, "") : void 0, c = u !== this.prevTransformTemplateValue;
      r && (a || nt(this.latestValues) || c) && (i(this.instance, u), this.shouldResetTransform = !1, this.scheduleRender());
    }
    measure(r = !0) {
      const a = this.measurePageBox();
      let l = this.removeElementScroll(a);
      return r && (l = this.removeTransform(l)), pu(l), {
        animationId: this.root.animationId,
        measuredBox: a,
        layoutBox: l,
        latestValues: {},
        source: this.id
      };
    }
    measurePageBox() {
      var r;
      const { visualElement: a } = this.options;
      if (!a)
        return V();
      const l = a.measureViewportBox();
      if (!(((r = this.scroll) === null || r === void 0 ? void 0 : r.wasRoot) || this.path.some(mu))) {
        const { scroll: c } = this.root;
        c && (ct(l.x, c.offset.x), ct(l.y, c.offset.y));
      }
      return l;
    }
    removeElementScroll(r) {
      var a;
      const l = V();
      if (j(l, r), !((a = this.scroll) === null || a === void 0) && a.wasRoot)
        return l;
      for (let u = 0; u < this.path.length; u++) {
        const c = this.path[u], { scroll: h, options: f } = c;
        c !== this.root && h && f.layoutScroll && (h.wasRoot && j(l, r), ct(l.x, h.offset.x), ct(l.y, h.offset.y));
      }
      return l;
    }
    applyTransform(r, a = !1) {
      const l = V();
      j(l, r);
      for (let u = 0; u < this.path.length; u++) {
        const c = this.path[u];
        !a && c.options.layoutScroll && c.scroll && c !== c.root && ht(l, {
          x: -c.scroll.offset.x,
          y: -c.scroll.offset.y
        }), nt(c.latestValues) && ht(l, c.latestValues);
      }
      return nt(this.latestValues) && ht(l, this.latestValues), l;
    }
    removeTransform(r) {
      const a = V();
      j(a, r);
      for (let l = 0; l < this.path.length; l++) {
        const u = this.path[l];
        if (!u.instance || !nt(u.latestValues))
          continue;
        De(u.latestValues) && u.updateSnapshot();
        const c = V(), h = u.measurePageBox();
        j(c, h), us(a, u.latestValues, u.snapshot ? u.snapshot.layoutBox : void 0, c);
      }
      return nt(this.latestValues) && us(a, this.latestValues), a;
    }
    setTargetDelta(r) {
      this.targetDelta = r, this.root.scheduleUpdateProjection(), this.isProjectionDirty = !0;
    }
    setOptions(r) {
      this.options = {
        ...this.options,
        ...r,
        crossfade: r.crossfade !== void 0 ? r.crossfade : !0
      };
    }
    clearMeasurements() {
      this.scroll = void 0, this.layout = void 0, this.snapshot = void 0, this.prevTransformTemplateValue = void 0, this.targetDelta = void 0, this.target = void 0, this.isLayoutDirty = !1;
    }
    forceRelativeParentToResolveTarget() {
      this.relativeParent && this.relativeParent.resolvedRelativeTargetAt !== D.timestamp && this.relativeParent.resolveTargetDelta(!0);
    }
    resolveTargetDelta(r = !1) {
      var a;
      const l = this.getLead();
      this.isProjectionDirty || (this.isProjectionDirty = l.isProjectionDirty), this.isTransformDirty || (this.isTransformDirty = l.isTransformDirty), this.isSharedProjectionDirty || (this.isSharedProjectionDirty = l.isSharedProjectionDirty);
      const u = !!this.resumingFrom || this !== l;
      if (!(r || u && this.isSharedProjectionDirty || this.isProjectionDirty || !((a = this.parent) === null || a === void 0) && a.isProjectionDirty || this.attemptToResolveRelativeTarget || this.root.updateBlockedByResize))
        return;
      const { layout: h, layoutId: f } = this.options;
      if (!(!this.layout || !(h || f))) {
        if (this.resolvedRelativeTargetAt = D.timestamp, !this.targetDelta && !this.relativeTarget) {
          const d = this.getClosestProjectingParent();
          d && d.layout && this.animationProgress !== 1 ? (this.relativeParent = d, this.forceRelativeParentToResolveTarget(), this.relativeTarget = V(), this.relativeTargetOrigin = V(), bt(this.relativeTargetOrigin, this.layout.layoutBox, d.layout.layoutBox), j(this.relativeTarget, this.relativeTargetOrigin)) : this.relativeParent = this.relativeTarget = void 0;
        }
        if (!(!this.relativeTarget && !this.targetDelta)) {
          if (this.target || (this.target = V(), this.targetWithTransforms = V()), this.relativeTarget && this.relativeTargetOrigin && this.relativeParent && this.relativeParent.target ? (this.forceRelativeParentToResolveTarget(), yl(this.target, this.relativeTarget, this.relativeParent.target)) : this.targetDelta ? (this.resumingFrom ? this.target = this.applyTransform(this.layout.layoutBox) : j(this.target, this.layout.layoutBox), Ni(this.target, this.targetDelta)) : j(this.target, this.layout.layoutBox), this.attemptToResolveRelativeTarget) {
            this.attemptToResolveRelativeTarget = !1;
            const d = this.getClosestProjectingParent();
            d && !!d.resumingFrom == !!this.resumingFrom && !d.options.layoutScroll && d.target && this.animationProgress !== 1 ? (this.relativeParent = d, this.forceRelativeParentToResolveTarget(), this.relativeTarget = V(), this.relativeTargetOrigin = V(), bt(this.relativeTargetOrigin, this.target, d.target), j(this.relativeTarget, this.relativeTargetOrigin)) : this.relativeParent = this.relativeTarget = void 0;
          }
          xt && st.resolvedTargetDeltas++;
        }
      }
    }
    getClosestProjectingParent() {
      if (!(!this.parent || De(this.parent.latestValues) || Ii(this.parent.latestValues)))
        return this.parent.isProjecting() ? this.parent : this.parent.getClosestProjectingParent();
    }
    isProjecting() {
      return !!((this.relativeTarget || this.targetDelta || this.options.layoutRoot) && this.layout);
    }
    calcProjection() {
      var r;
      const a = this.getLead(), l = !!this.resumingFrom || this !== a;
      let u = !0;
      if ((this.isProjectionDirty || !((r = this.parent) === null || r === void 0) && r.isProjectionDirty) && (u = !1), l && (this.isSharedProjectionDirty || this.isTransformDirty) && (u = !1), this.resolvedRelativeTargetAt === D.timestamp && (u = !1), u)
        return;
      const { layout: c, layoutId: h } = this.options;
      if (this.isTreeAnimating = !!(this.parent && this.parent.isTreeAnimating || this.currentAnimation || this.pendingAnimation), this.isTreeAnimating || (this.targetDelta = this.relativeTarget = void 0), !this.layout || !(c || h))
        return;
      j(this.layoutCorrected, this.layout.layoutBox);
      const f = this.treeScale.x, d = this.treeScale.y;
      Vl(this.layoutCorrected, this.treeScale, this.path, l), a.layout && !a.target && (this.treeScale.x !== 1 || this.treeScale.y !== 1) && (a.target = a.layout.layoutBox, a.targetWithTransforms = V());
      const { target: p } = a;
      if (!p) {
        this.prevProjectionDelta && (this.createProjectionDeltas(), this.scheduleRender());
        return;
      }
      !this.projectionDelta || !this.prevProjectionDelta ? this.createProjectionDeltas() : (os(this.prevProjectionDelta.x, this.projectionDelta.x), os(this.prevProjectionDelta.y, this.projectionDelta.y)), St(this.projectionDelta, this.layoutCorrected, p, this.latestValues), (this.treeScale.x !== f || this.treeScale.y !== d || !ps(this.projectionDelta.x, this.prevProjectionDelta.x) || !ps(this.projectionDelta.y, this.prevProjectionDelta.y)) && (this.hasProjected = !0, this.scheduleRender(), this.notifyListeners("projectionUpdate", p)), xt && st.recalculatedProjection++;
    }
    hide() {
      this.isVisible = !1;
    }
    show() {
      this.isVisible = !0;
    }
    scheduleRender(r = !0) {
      var a;
      if ((a = this.options.visualElement) === null || a === void 0 || a.scheduleRender(), r) {
        const l = this.getStack();
        l && l.scheduleRender();
      }
      this.resumingFrom && !this.resumingFrom.instance && (this.resumingFrom = void 0);
    }
    createProjectionDeltas() {
      this.prevProjectionDelta = ut(), this.projectionDelta = ut(), this.projectionDeltaWithTransform = ut();
    }
    setAnimationOrigin(r, a = !1) {
      const l = this.snapshot, u = l ? l.latestValues : {}, c = { ...this.latestValues }, h = ut();
      (!this.relativeParent || !this.relativeParent.options.layoutRoot) && (this.relativeTarget = this.relativeTargetOrigin = void 0), this.attemptToResolveRelativeTarget = !a;
      const f = V(), d = l ? l.source : void 0, p = this.layout ? this.layout.source : void 0, m = d !== p, x = this.getStack(), g = !x || x.members.length <= 1, v = !!(m && !g && this.options.crossfade === !0 && !this.path.some(fu));
      this.animationProgress = 0;
      let y;
      this.mixTargetDelta = (P) => {
        const S = P / 1e3;
        vs(h.x, r.x, S), vs(h.y, r.y, S), this.setTargetDelta(h), this.relativeTarget && this.relativeTargetOrigin && this.layout && this.relativeParent && this.relativeParent.layout && (bt(f, this.layout.layoutBox, this.relativeParent.layout.layoutBox), hu(this.relativeTarget, this.relativeTargetOrigin, f, S), y && $l(this.relativeTarget, y) && (this.isProjectionDirty = !1), y || (y = V()), j(y, this.relativeTarget)), m && (this.animationValues = c, Il(c, u, this.latestValues, S, v, g)), this.root.scheduleUpdateProjection(), this.scheduleRender(), this.animationProgress = S;
      }, this.mixTargetDelta(this.options.layoutRoot ? 1e3 : 0);
    }
    startAnimation(r) {
      this.notifyListeners("animationStart"), this.currentAnimation && this.currentAnimation.stop(), this.resumingFrom && this.resumingFrom.currentAnimation && this.resumingFrom.currentAnimation.stop(), this.pendingAnimation && (z(this.pendingAnimation), this.pendingAnimation = void 0), this.pendingAnimation = w.update(() => {
        It.hasAnimatedSinceResize = !0, this.currentAnimation = Zl(0, ms, {
          ...r,
          onUpdate: (a) => {
            this.mixTargetDelta(a), r.onUpdate && r.onUpdate(a);
          },
          onComplete: () => {
            r.onComplete && r.onComplete(), this.completeAnimation();
          }
        }), this.resumingFrom && (this.resumingFrom.currentAnimation = this.currentAnimation), this.pendingAnimation = void 0;
      });
    }
    completeAnimation() {
      this.resumingFrom && (this.resumingFrom.currentAnimation = void 0, this.resumingFrom.preserveOpacity = void 0);
      const r = this.getStack();
      r && r.exitAnimationComplete(), this.resumingFrom = this.currentAnimation = this.animationValues = void 0, this.notifyListeners("animationComplete");
    }
    finishAnimation() {
      this.currentAnimation && (this.mixTargetDelta && this.mixTargetDelta(ms), this.currentAnimation.stop()), this.completeAnimation();
    }
    applyTransformsToTarget() {
      const r = this.getLead();
      let { targetWithTransforms: a, target: l, layout: u, latestValues: c } = r;
      if (!(!a || !l || !u)) {
        if (this !== r && this.layout && u && Xi(this.options.animationType, this.layout.layoutBox, u.layoutBox)) {
          l = this.target || V();
          const h = B(this.layout.layoutBox.x);
          l.x.min = r.target.x.min, l.x.max = l.x.min + h;
          const f = B(this.layout.layoutBox.y);
          l.y.min = r.target.y.min, l.y.max = l.y.min + f;
        }
        j(a, l), ht(a, c), St(this.projectionDeltaWithTransform, this.layoutCorrected, a, c);
      }
    }
    registerSharedNode(r, a) {
      this.sharedNodes.has(r) || this.sharedNodes.set(r, new Gl()), this.sharedNodes.get(r).add(a);
      const u = a.options.initialPromotionConfig;
      a.promote({
        transition: u ? u.transition : void 0,
        preserveFollowOpacity: u && u.shouldPreserveFollowOpacity ? u.shouldPreserveFollowOpacity(a) : void 0
      });
    }
    isLead() {
      const r = this.getStack();
      return r ? r.lead === this : !0;
    }
    getLead() {
      var r;
      const { layoutId: a } = this.options;
      return a ? ((r = this.getStack()) === null || r === void 0 ? void 0 : r.lead) || this : this;
    }
    getPrevLead() {
      var r;
      const { layoutId: a } = this.options;
      return a ? (r = this.getStack()) === null || r === void 0 ? void 0 : r.prevLead : void 0;
    }
    getStack() {
      const { layoutId: r } = this.options;
      if (r)
        return this.root.sharedNodes.get(r);
    }
    promote({ needsReset: r, transition: a, preserveFollowOpacity: l } = {}) {
      const u = this.getStack();
      u && u.promote(this, l), r && (this.projectionDelta = void 0, this.needsReset = !0), a && this.setOptions({ transition: a });
    }
    relegate() {
      const r = this.getStack();
      return r ? r.relegate(this) : !1;
    }
    resetSkewAndRotation() {
      const { visualElement: r } = this.options;
      if (!r)
        return;
      let a = !1;
      const { latestValues: l } = r;
      if ((l.z || l.rotate || l.rotateX || l.rotateY || l.rotateZ || l.skewX || l.skewY) && (a = !0), !a)
        return;
      const u = {};
      l.z && pe("z", r, u, this.animationValues);
      for (let c = 0; c < de.length; c++)
        pe(`rotate${de[c]}`, r, u, this.animationValues), pe(`skew${de[c]}`, r, u, this.animationValues);
      r.render();
      for (const c in u)
        r.setStaticValue(c, u[c]), this.animationValues && (this.animationValues[c] = u[c]);
      r.scheduleRender();
    }
    getProjectionStyles(r) {
      var a, l;
      if (!this.instance || this.isSVG)
        return;
      if (!this.isVisible)
        return Jl;
      const u = {
        visibility: ""
      }, c = this.getTransformTemplate();
      if (this.needsReset)
        return this.needsReset = !1, u.opacity = "", u.pointerEvents = Ot(r == null ? void 0 : r.pointerEvents) || "", u.transform = c ? c(this.latestValues, "") : "none", u;
      const h = this.getLead();
      if (!this.projectionDelta || !this.layout || !h.target) {
        const m = {};
        return this.options.layoutId && (m.opacity = this.latestValues.opacity !== void 0 ? this.latestValues.opacity : 1, m.pointerEvents = Ot(r == null ? void 0 : r.pointerEvents) || ""), this.hasProjected && !nt(this.latestValues) && (m.transform = c ? c({}, "") : "none", this.hasProjected = !1), m;
      }
      const f = h.animationValues || h.latestValues;
      this.applyTransformsToTarget(), u.transform = zl(this.projectionDeltaWithTransform, this.treeScale, f), c && (u.transform = c(f, u.transform));
      const { x: d, y: p } = this.projectionDelta;
      u.transformOrigin = `${d.origin * 100}% ${p.origin * 100}% 0`, h.animationValues ? u.opacity = h === this ? (l = (a = f.opacity) !== null && a !== void 0 ? a : this.latestValues.opacity) !== null && l !== void 0 ? l : 1 : this.preserveOpacity ? this.latestValues.opacity : f.opacityExit : u.opacity = h === this ? f.opacity !== void 0 ? f.opacity : "" : f.opacityExit !== void 0 ? f.opacityExit : 0;
      for (const m in Ut) {
        if (f[m] === void 0)
          continue;
        const { correct: x, applyTo: g } = Ut[m], v = u.transform === "none" ? f[m] : x(f[m], h);
        if (g) {
          const y = g.length;
          for (let P = 0; P < y; P++)
            u[g[P]] = v;
        } else
          u[m] = v;
      }
      return this.options.layoutId && (u.pointerEvents = h === this ? Ot(r == null ? void 0 : r.pointerEvents) || "" : "none"), u;
    }
    clearSnapshot() {
      this.resumeFrom = this.snapshot = void 0;
    }
    // Only run on root
    resetTree() {
      this.root.nodes.forEach((r) => {
        var a;
        return (a = r.currentAnimation) === null || a === void 0 ? void 0 : a.stop();
      }), this.root.nodes.forEach(gs), this.root.sharedNodes.clear();
    }
  };
}
function tu(t) {
  t.updateLayout();
}
function eu(t) {
  var e;
  const n = ((e = t.resumeFrom) === null || e === void 0 ? void 0 : e.snapshot) || t.snapshot;
  if (t.isLead() && t.layout && n && t.hasListeners("didUpdate")) {
    const { layoutBox: s, measuredBox: i } = t.layout, { animationType: o } = t.options, r = n.source !== t.layout.source;
    o === "size" ? I((h) => {
      const f = r ? n.measuredBox[h] : n.layoutBox[h], d = B(f);
      f.min = s[h].min, f.max = f.min + d;
    }) : Xi(o, n.layoutBox, s) && I((h) => {
      const f = r ? n.measuredBox[h] : n.layoutBox[h], d = B(s[h]);
      f.max = f.min + d, t.relativeTarget && !t.currentAnimation && (t.isProjectionDirty = !0, t.relativeTarget[h].max = t.relativeTarget[h].min + d);
    });
    const a = ut();
    St(a, s, n.layoutBox);
    const l = ut();
    r ? St(l, t.applyTransform(i, !0), n.measuredBox) : St(l, s, n.layoutBox);
    const u = !Gi(a);
    let c = !1;
    if (!t.resumeFrom) {
      const h = t.getClosestProjectingParent();
      if (h && !h.resumeFrom) {
        const { snapshot: f, layout: d } = h;
        if (f && d) {
          const p = V();
          bt(p, n.layoutBox, f.layoutBox);
          const m = V();
          bt(m, s, d.layoutBox), zi(p, m) || (c = !0), h.options.layoutRoot && (t.relativeTarget = m, t.relativeTargetOrigin = p, t.relativeParent = h);
        }
      }
    }
    t.notifyListeners("didUpdate", {
      layout: s,
      snapshot: n,
      delta: l,
      layoutDelta: a,
      hasLayoutChanged: u,
      hasRelativeTargetChanged: c
    });
  } else if (t.isLead()) {
    const { onExitComplete: s } = t.options;
    s && s();
  }
  t.options.transition = void 0;
}
function nu(t) {
  xt && st.totalNodes++, t.parent && (t.isProjecting() || (t.isProjectionDirty = t.parent.isProjectionDirty), t.isSharedProjectionDirty || (t.isSharedProjectionDirty = !!(t.isProjectionDirty || t.parent.isProjectionDirty || t.parent.isSharedProjectionDirty)), t.isTransformDirty || (t.isTransformDirty = t.parent.isTransformDirty));
}
function su(t) {
  t.isProjectionDirty = t.isSharedProjectionDirty = t.isTransformDirty = !1;
}
function iu(t) {
  t.clearSnapshot();
}
function gs(t) {
  t.clearMeasurements();
}
function ru(t) {
  t.isLayoutDirty = !1;
}
function ou(t) {
  const { visualElement: e } = t.options;
  e && e.getProps().onBeforeLayoutMeasure && e.notify("BeforeLayoutMeasure"), t.resetTransform();
}
function ys(t) {
  t.finishAnimation(), t.targetDelta = t.relativeTarget = t.target = void 0, t.isProjectionDirty = !0;
}
function au(t) {
  t.resolveTargetDelta();
}
function lu(t) {
  t.calcProjection();
}
function uu(t) {
  t.resetSkewAndRotation();
}
function cu(t) {
  t.removeLeadSnapshot();
}
function vs(t, e, n) {
  t.translate = A(e.translate, 0, n), t.scale = A(e.scale, 1, n), t.origin = e.origin, t.originPoint = e.originPoint;
}
function xs(t, e, n, s) {
  t.min = A(e.min, n.min, s), t.max = A(e.max, n.max, s);
}
function hu(t, e, n, s) {
  xs(t.x, e.x, n.x, s), xs(t.y, e.y, n.y, s);
}
function fu(t) {
  return t.animationValues && t.animationValues.opacityExit !== void 0;
}
const du = {
  duration: 0.45,
  ease: [0.4, 0, 0.1, 1]
}, Ts = (t) => typeof navigator < "u" && navigator.userAgent && navigator.userAgent.toLowerCase().includes(t), Ps = Ts("applewebkit/") && !Ts("chrome/") ? Math.round : M;
function Ss(t) {
  t.min = Ps(t.min), t.max = Ps(t.max);
}
function pu(t) {
  Ss(t.x), Ss(t.y);
}
function Xi(t, e, n) {
  return t === "position" || t === "preserve-aspect" && !gl(ds(e), ds(n), 0.2);
}
function mu(t) {
  var e;
  return t !== t.root && ((e = t.scroll) === null || e === void 0 ? void 0 : e.wasRoot);
}
const gu = Yi({
  attachResizeListener: (t, e) => _(t, "resize", e),
  measureScroll: () => ({
    x: document.documentElement.scrollLeft || document.body.scrollLeft,
    y: document.documentElement.scrollTop || document.body.scrollTop
  }),
  checkIsScrollRoot: () => !0
}), me = {
  current: void 0
}, qi = Yi({
  measureScroll: (t) => ({
    x: t.scrollLeft,
    y: t.scrollTop
  }),
  defaultParent: () => {
    if (!me.current) {
      const t = new gu({});
      t.mount(window), t.setOptions({ layoutScroll: !0 }), me.current = t;
    }
    return me.current;
  },
  resetTransform: (t, e) => {
    t.style.transform = e !== void 0 ? e : "none";
  },
  checkIsScrollRoot: (t) => window.getComputedStyle(t).position === "fixed"
}), yu = {
  pan: {
    Feature: Ll
  },
  drag: {
    Feature: El,
    ProjectionNode: qi,
    MeasureLayout: Wi
  }
}, Re = { current: null }, Zi = { current: !1 };
function vu() {
  if (Zi.current = !0, !!Be)
    if (window.matchMedia) {
      const t = window.matchMedia("(prefers-reduced-motion)"), e = () => Re.current = t.matches;
      t.addListener(e), e();
    } else
      Re.current = !1;
}
function xu(t, e, n) {
  for (const s in e) {
    const i = e[s], o = n[s];
    if (E(i))
      t.addValue(s, i), process.env.NODE_ENV === "development" && on(i.version === "11.3.20", `Attempting to mix Framer Motion versions ${i.version} with 11.3.20 may not work as expected.`);
    else if (E(o))
      t.addValue(s, Dt(i, { owner: t }));
    else if (o !== i)
      if (t.hasValue(s)) {
        const r = t.getValue(s);
        r.liveStyle === !0 ? r.jump(i) : r.hasAnimated || r.set(i);
      } else {
        const r = t.getStaticValue(s);
        t.addValue(s, Dt(r !== void 0 ? r : i, { owner: t }));
      }
  }
  for (const s in n)
    e[s] === void 0 && t.removeValue(s);
  return e;
}
const bs = /* @__PURE__ */ new WeakMap(), Tu = [...ii, R, J], Pu = (t) => Tu.find(si(t)), ws = [
  "AnimationStart",
  "AnimationComplete",
  "Update",
  "BeforeLayoutMeasure",
  "LayoutMeasure",
  "LayoutAnimationStart",
  "LayoutAnimationComplete"
], Su = je.length;
class bu {
  /**
   * This method takes React props and returns found MotionValues. For example, HTML
   * MotionValues will be found within the style prop, whereas for Three.js within attribute arrays.
   *
   * This isn't an abstract method as it needs calling in the constructor, but it is
   * intended to be one.
   */
  scrapeMotionValuesFromProps(e, n, s) {
    return {};
  }
  constructor({ parent: e, props: n, presenceContext: s, reducedMotionConfig: i, blockInitialAnimation: o, visualState: r }, a = {}) {
    this.applyWillChange = !1, this.resolveKeyframes = (f, d, p, m) => new this.KeyframeResolver(f, d, p, m, this), this.current = null, this.children = /* @__PURE__ */ new Set(), this.isVariantNode = !1, this.isControllingVariants = !1, this.shouldReduceMotion = null, this.values = /* @__PURE__ */ new Map(), this.KeyframeResolver = Xe, this.features = {}, this.valueSubscriptions = /* @__PURE__ */ new Map(), this.prevMotionValues = {}, this.events = {}, this.propEventSubscriptions = {}, this.notifyUpdate = () => this.notify("Update", this.latestValues), this.render = () => {
      this.isRenderScheduled = !1, this.current && (this.triggerBuild(), this.renderInstance(this.current, this.renderState, this.props.style, this.projection));
    }, this.isRenderScheduled = !1, this.scheduleRender = () => {
      this.isRenderScheduled || (this.isRenderScheduled = !0, w.render(this.render, !1, !0));
    };
    const { latestValues: l, renderState: u } = r;
    this.latestValues = l, this.baseTarget = { ...l }, this.initialValues = n.initial ? { ...l } : {}, this.renderState = u, this.parent = e, this.props = n, this.presenceContext = s, this.depth = e ? e.depth + 1 : 0, this.reducedMotionConfig = i, this.options = a, this.blockInitialAnimation = !!o, this.isControllingVariants = Xt(n), this.isVariantNode = Ls(n), this.isVariantNode && (this.variantChildren = /* @__PURE__ */ new Set()), this.manuallyAnimateOnMount = !!(e && e.current);
    const { willChange: c, ...h } = this.scrapeMotionValuesFromProps(n, {}, this);
    for (const f in h) {
      const d = h[f];
      l[f] !== void 0 && E(d) && d.set(l[f], !1);
    }
  }
  mount(e) {
    this.current = e, bs.set(e, this), this.projection && !this.projection.instance && this.projection.mount(e), this.parent && this.isVariantNode && !this.isControllingVariants && (this.removeFromVariantTree = this.parent.addVariantChild(this)), this.values.forEach((n, s) => this.bindToMotionValue(s, n)), Zi.current || vu(), this.shouldReduceMotion = this.reducedMotionConfig === "never" ? !1 : this.reducedMotionConfig === "always" ? !0 : Re.current, process.env.NODE_ENV !== "production" && on(this.shouldReduceMotion !== !0, "You have Reduced Motion enabled on your device. Animations may not appear as expected."), this.parent && this.parent.children.add(this), this.update(this.props, this.presenceContext);
  }
  unmount() {
    bs.delete(this.current), this.projection && this.projection.unmount(), z(this.notifyUpdate), z(this.render), this.valueSubscriptions.forEach((e) => e()), this.removeFromVariantTree && this.removeFromVariantTree(), this.parent && this.parent.children.delete(this);
    for (const e in this.events)
      this.events[e].clear();
    for (const e in this.features) {
      const n = this.features[e];
      n && (n.unmount(), n.isMounted = !1);
    }
    this.current = null;
  }
  bindToMotionValue(e, n) {
    const s = Q.has(e), i = n.on("change", (r) => {
      this.latestValues[e] = r, this.props.onUpdate && w.preRender(this.notifyUpdate), s && this.projection && (this.projection.isTransformDirty = !0);
    }), o = n.on("renderRequest", this.scheduleRender);
    this.valueSubscriptions.set(e, () => {
      i(), o(), n.owner && n.stop();
    });
  }
  sortNodePosition(e) {
    return !this.current || !this.sortInstanceNodePosition || this.type !== e.type ? 0 : this.sortInstanceNodePosition(this.current, e.current);
  }
  updateFeatures() {
    let e = "animation";
    for (e in ft) {
      const n = ft[e];
      if (!n)
        continue;
      const { isEnabled: s, Feature: i } = n;
      if (!this.features[e] && i && s(this.props) && (this.features[e] = new i(this)), this.features[e]) {
        const o = this.features[e];
        o.isMounted ? o.update() : (o.mount(), o.isMounted = !0);
      }
    }
  }
  triggerBuild() {
    this.build(this.renderState, this.latestValues, this.props);
  }
  /**
   * Measure the current viewport box with or without transforms.
   * Only measures axis-aligned boxes, rotate and skew must be manually
   * removed with a re-render to work.
   */
  measureViewportBox() {
    return this.current ? this.measureInstanceViewportBox(this.current, this.props) : V();
  }
  getStaticValue(e) {
    return this.latestValues[e];
  }
  setStaticValue(e, n) {
    this.latestValues[e] = n;
  }
  /**
   * Update the provided props. Ensure any newly-added motion values are
   * added to our map, old ones removed, and listeners updated.
   */
  update(e, n) {
    (e.transformTemplate || this.props.transformTemplate) && this.scheduleRender(), this.prevProps = this.props, this.props = e, this.prevPresenceContext = this.presenceContext, this.presenceContext = n;
    for (let s = 0; s < ws.length; s++) {
      const i = ws[s];
      this.propEventSubscriptions[i] && (this.propEventSubscriptions[i](), delete this.propEventSubscriptions[i]);
      const o = "on" + i, r = e[o];
      r && (this.propEventSubscriptions[i] = this.on(i, r));
    }
    this.prevMotionValues = xu(this, this.scrapeMotionValuesFromProps(e, this.prevProps, this), this.prevMotionValues), this.handleChildMotionValue && this.handleChildMotionValue();
  }
  getProps() {
    return this.props;
  }
  /**
   * Returns the variant definition with a given name.
   */
  getVariant(e) {
    return this.props.variants ? this.props.variants[e] : void 0;
  }
  /**
   * Returns the defined default transition on this component.
   */
  getDefaultTransition() {
    return this.props.transition;
  }
  getTransformPagePoint() {
    return this.props.transformPagePoint;
  }
  getClosestVariantNode() {
    return this.isVariantNode ? this : this.parent ? this.parent.getClosestVariantNode() : void 0;
  }
  getVariantContext(e = !1) {
    if (e)
      return this.parent ? this.parent.getVariantContext() : void 0;
    if (!this.isControllingVariants) {
      const s = this.parent ? this.parent.getVariantContext() || {} : {};
      return this.props.initial !== void 0 && (s.initial = this.props.initial), s;
    }
    const n = {};
    for (let s = 0; s < Su; s++) {
      const i = je[s], o = this.props[i];
      (wt(o) || o === !1) && (n[i] = o);
    }
    return n;
  }
  /**
   * Add a child visual element to our set of children.
   */
  addVariantChild(e) {
    const n = this.getClosestVariantNode();
    if (n)
      return n.variantChildren && n.variantChildren.add(e), () => n.variantChildren.delete(e);
  }
  /**
   * Add a motion value and bind it to this visual element.
   */
  addValue(e, n) {
    const s = this.values.get(e);
    n !== s && (s && this.removeValue(e), this.bindToMotionValue(e, n), this.values.set(e, n), this.latestValues[e] = n.get());
  }
  /**
   * Remove a motion value and unbind any active subscriptions.
   */
  removeValue(e) {
    this.values.delete(e);
    const n = this.valueSubscriptions.get(e);
    n && (n(), this.valueSubscriptions.delete(e)), delete this.latestValues[e], this.removeValueFromRenderState(e, this.renderState);
  }
  /**
   * Check whether we have a motion value for this key
   */
  hasValue(e) {
    return this.values.has(e);
  }
  getValue(e, n) {
    if (this.props.values && this.props.values[e])
      return this.props.values[e];
    let s = this.values.get(e);
    return s === void 0 && n !== void 0 && (s = Dt(n === null ? void 0 : n, { owner: this }), this.addValue(e, s)), s;
  }
  /**
   * If we're trying to animate to a previously unencountered value,
   * we need to check for it in our state and as a last resort read it
   * directly from the instance (which might have performance implications).
   */
  readValue(e, n) {
    var s;
    let i = this.latestValues[e] !== void 0 || !this.current ? this.latestValues[e] : (s = this.getBaseTargetFromProps(this.props, e)) !== null && s !== void 0 ? s : this.readValueFromInstance(this.current, e, this.options);
    return i != null && (typeof i == "string" && (ei(i) || ti(i)) ? i = parseFloat(i) : !Pu(i) && J.test(n) && (i = fi(e, n)), this.setBaseTarget(e, E(i) ? i.get() : i)), E(i) ? i.get() : i;
  }
  /**
   * Set the base target to later animate back to. This is currently
   * only hydrated on creation and when we first read a value.
   */
  setBaseTarget(e, n) {
    this.baseTarget[e] = n;
  }
  /**
   * Find the base target for a value thats been removed from all animation
   * props.
   */
  getBaseTarget(e) {
    var n;
    const { initial: s } = this.props;
    let i;
    if (typeof s == "string" || typeof s == "object") {
      const r = He(this.props, s, (n = this.presenceContext) === null || n === void 0 ? void 0 : n.custom);
      r && (i = r[e]);
    }
    if (s && i !== void 0)
      return i;
    const o = this.getBaseTargetFromProps(this.props, e);
    return o !== void 0 && !E(o) ? o : this.initialValues[e] !== void 0 && i === void 0 ? void 0 : this.baseTarget[e];
  }
  on(e, n) {
    return this.events[e] || (this.events[e] = new rn()), this.events[e].add(n);
  }
  notify(e, ...n) {
    this.events[e] && this.events[e].notify(...n);
  }
}
class Ji extends bu {
  constructor() {
    super(...arguments), this.KeyframeResolver = di;
  }
  sortInstanceNodePosition(e, n) {
    return e.compareDocumentPosition(n) & 2 ? 1 : -1;
  }
  getBaseTargetFromProps(e, n) {
    return e.style ? e.style[n] : void 0;
  }
  removeValueFromRenderState(e, { vars: n, style: s }) {
    delete n[e], delete s[e];
  }
}
function wu(t) {
  return window.getComputedStyle(t);
}
class Au extends Ji {
  constructor() {
    super(...arguments), this.type = "html", this.applyWillChange = !0, this.renderInstance = _s;
  }
  readValueFromInstance(e, n) {
    if (Q.has(n)) {
      const s = Ze(n);
      return s && s.default || 0;
    } else {
      const s = wu(e), i = (js(n) ? s.getPropertyValue(n) : s[n]) || 0;
      return typeof i == "string" ? i.trim() : i;
    }
  }
  measureInstanceViewportBox(e, { transformPagePoint: n }) {
    return Ui(e, n);
  }
  build(e, n, s) {
    We(e, n, s.transformTemplate);
  }
  scrapeMotionValuesFromProps(e, n, s) {
    return ze(e, n, s);
  }
  handleChildMotionValue() {
    this.childSubscription && (this.childSubscription(), delete this.childSubscription);
    const { children: e } = this.props;
    E(e) && (this.childSubscription = e.on("change", (n) => {
      this.current && (this.current.textContent = `${n}`);
    }));
  }
}
class Vu extends Ji {
  constructor() {
    super(...arguments), this.type = "svg", this.isSVGTag = !1, this.measureInstanceViewportBox = V;
  }
  getBaseTargetFromProps(e, n) {
    return e[n];
  }
  readValueFromInstance(e, n) {
    if (Q.has(n)) {
      const s = Ze(n);
      return s && s.default || 0;
    }
    return n = Ws.has(n) ? n : Yt(n), e.getAttribute(n);
  }
  scrapeMotionValuesFromProps(e, n, s) {
    return $s(e, n, s);
  }
  build(e, n, s) {
    $e(e, n, this.isSVGTag, s.transformTemplate);
  }
  renderInstance(e, n, s, i) {
    Ks(e, n, s, i);
  }
  mount(e) {
    this.isSVGTag = Ge(e.tagName), super.mount(e);
  }
}
const Cu = (t, e) => Ie(t) ? new Vu(e) : new Au(e, {
  allowProjection: t !== Le
}), Du = {
  layout: {
    ProjectionNode: qi,
    MeasureLayout: Wi
  }
}, Mu = {
  ...ll,
  ...vo,
  ...yu,
  ...Du
}, Ou = /* @__PURE__ */ Ar((t, e) => io(t, e, Mu, Cu)), ju = ({ spinnerType: t = "ClipLoader", spinnerCss: e = {}, color: n = "blue", size: s = "2rem", aspectRatio: i = "16/9", wrapperHeight: o = "" }) => {
  const r = t === "none" ? Le : ar[t];
  return (
    //Spinner Wrapper
    Nt("div", { style: {
      width: "100%",
      aspectRatio: i,
      height: o,
      display: "flex",
      justifyContent: "center",
      alignItems: "center"
    }, children: Nt(r, { color: n, cssOverride: e, size: s }) })
  );
};
export {
  ju as B,
  Fu as g,
  Ou as m
};
