// react-image-displayer/dist/index-C47FX26S.js
import * as o from "react";
var ue = {
  cm: !0,
  mm: !0,
  in: !0,
  px: !0,
  pt: !0,
  pc: !0,
  em: !0,
  ex: !0,
  ch: !0,
  rem: !0,
  vw: !0,
  vh: !0,
  vmin: !0,
  vmax: !0,
  "%": !0
};
function M(e) {
  if (typeof e == "number")
    return {
      value: e,
      unit: "px"
    };
  var n, a = (e.match(/^[0-9.]*/) || "").toString();
  a.includes(".") ? n = parseFloat(a) : n = parseInt(a, 10);
  var t = (e.match(/[^0-9]*$/) || "").toString();
  return ue[t] ? {
    value: n,
    unit: t
  } : (console.warn("React Spinners: ".concat(e, " is not a valid css value. Defaulting to ").concat(n, "px.")), {
    value: n,
    unit: "px"
  });
}
function u(e) {
  var n = M(e);
  return "".concat(n.value).concat(n.unit);
}
var _ = function(e, n, a) {
  var t = "react-spinners-".concat(e, "-").concat(a);
  if (typeof window > "u" || !window.document)
    return t;
  var r = document.createElement("style");
  document.head.appendChild(r);
  var c = r.sheet, g = `
    @keyframes `.concat(t, ` {
      `).concat(n, `
    }
  `);
  return c && c.insertRule(g, 0), t;
}, R;
(function(e) {
  e.maroon = "#800000", e.red = "#FF0000", e.orange = "#FFA500", e.yellow = "#FFFF00", e.olive = "#808000", e.green = "#008000", e.purple = "#800080", e.fuchsia = "#FF00FF", e.lime = "#00FF00", e.teal = "#008080", e.aqua = "#00FFFF", e.blue = "#0000FF", e.navy = "#000080", e.black = "#000000", e.gray = "#808080", e.silver = "#C0C0C0", e.white = "#FFFFFF";
})(R || (R = {}));
var fe = function(e, n) {
  if (e.includes("/"))
    return e.replace("rgb(", "rgba(");
  var a = e.substring(e.startsWith("rgba(") ? 5 : 4, e.length - 1).trim(), t = a.split(",");
  return t.length === 4 ? e.replace("rgb(", "rgba(") : t.length === 3 ? "rgba(".concat(a, ", ").concat(n, ")") : "rgba(".concat(a, " / ").concat(n, ")");
}, oe = function(e, n) {
  if (e.startsWith("rgb"))
    return fe(e, n);
  if (Object.keys(R).includes(e) && (e = R[e]), e[0] === "#" && (e = e.slice(1)), e.length === 3) {
    var a = "";
    e.split("").forEach(function(r) {
      a += r, a += r;
    }), e = a;
  }
  var t = (e.match(/.{2}/g) || []).map(function(r) {
    return parseInt(r, 16);
  }).join(", ");
  return "rgba(".concat(t, ", ").concat(n, ")");
}, X = function() {
  return X = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, X.apply(this, arguments);
}, ve = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ge = _("BarLoader", "0% {left: -35%;right: 100%} 60% {left: 100%;right: -90%} 100% {left: 100%;right: -90%}", "long"), me = _("BarLoader", "0% {left: -200%;right: 100%} 60% {left: 107%;right: -8%} 100% {left: 107%;right: -8%}", "short");
function ye(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.height, i = s === void 0 ? 4 : s, m = e.width, y = m === void 0 ? 100 : m, d = ve(e, ["loading", "color", "speedMultiplier", "cssOverride", "height", "width"]), f = X({ display: "inherit", position: "relative", width: u(y), height: u(i), overflow: "hidden", backgroundColor: oe(r, 0.2), backgroundClip: "padding-box" }, h), p = function(v) {
    return {
      position: "absolute",
      height: u(i),
      overflow: "hidden",
      backgroundColor: r,
      backgroundClip: "padding-box",
      display: "block",
      borderRadius: 2,
      willChange: "left, right",
      animationFillMode: "forwards",
      animation: "".concat(v === 1 ? ge : me, " ").concat(2.1 / g, "s ").concat(v === 2 ? "".concat(1.15 / g, "s") : "", " ").concat(v === 1 ? "cubic-bezier(0.65, 0.815, 0.735, 0.395)" : "cubic-bezier(0.165, 0.84, 0.44, 1)", " infinite")
    };
  };
  return a ? o.createElement(
    "span",
    X({ style: f }, d),
    o.createElement("span", { style: p(1) }),
    o.createElement("span", { style: p(2) })
  ) : null;
}
var C = function() {
  return C = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, C.apply(this, arguments);
}, be = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, he = _("BeatLoader", "50% {transform: scale(0.75);opacity: 0.2} 100% {transform: scale(1);opacity: 1}", "beat");
function Oe(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = be(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = C({ display: "inherit" }, h), p = function(v) {
    return {
      display: "inline-block",
      backgroundColor: r,
      width: u(i),
      height: u(i),
      margin: u(y),
      borderRadius: "100%",
      animation: "".concat(he, " ").concat(0.7 / g, "s ").concat(v % 2 ? "0s" : "".concat(0.35 / g, "s"), " infinite linear"),
      animationFillMode: "both"
    };
  };
  return a ? o.createElement(
    "span",
    C({ style: f }, d),
    o.createElement("span", { style: p(1) }),
    o.createElement("span", { style: p(2) }),
    o.createElement("span", { style: p(3) })
  ) : null;
}
var I = function() {
  return I = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, I.apply(this, arguments);
}, we = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, _e = _("BounceLoader", "0% {transform: scale(0)} 50% {transform: scale(1.0)} 100% {transform: scale(0)}", "bounce");
function xe(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 60 : s, m = we(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = function(f) {
    var p = f === 1 ? "".concat(1 / g, "s") : "0s";
    return {
      position: "absolute",
      height: u(i),
      width: u(i),
      backgroundColor: r,
      borderRadius: "100%",
      opacity: 0.6,
      top: 0,
      left: 0,
      animationFillMode: "both",
      animation: "".concat(_e, " ").concat(2.1 / g, "s ").concat(p, " infinite ease-in-out")
    };
  }, d = I({ display: "inherit", position: "relative", width: u(i), height: u(i) }, h);
  return a ? o.createElement(
    "span",
    I({ style: d }, m),
    o.createElement("span", { style: y(1) }),
    o.createElement("span", { style: y(2) })
  ) : null;
}
var Y = function() {
  return Y = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, Y.apply(this, arguments);
}, je = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Ee = _("CircleLoader", "0% {transform: rotate(0deg)} 50% {transform: rotate(180deg)} 100% {transform: rotate(360deg)}", "circle");
function Pe(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 50 : s, m = je(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = Y({ display: "inherit", position: "relative", width: u(i), height: u(i) }, h), d = function(f) {
    var p = M(i), v = p.value, b = p.unit;
    return {
      position: "absolute",
      height: "".concat(v * (1 - f / 10)).concat(b),
      width: "".concat(v * (1 - f / 10)).concat(b),
      borderTop: "1px solid ".concat(r),
      borderBottom: "none",
      borderLeft: "1px solid ".concat(r),
      borderRight: "none",
      borderRadius: "100%",
      transition: "2s",
      top: "".concat(f * 0.7 * 2.5, "%"),
      left: "".concat(f * 0.35 * 2.5, "%"),
      animation: "".concat(Ee, " ").concat(1 / g, "s ").concat(f * 0.2 / g, "s infinite linear")
    };
  };
  return a ? o.createElement(
    "span",
    Y({ style: y }, m),
    o.createElement("span", { style: d(0) }),
    o.createElement("span", { style: d(1) }),
    o.createElement("span", { style: d(2) }),
    o.createElement("span", { style: d(3) }),
    o.createElement("span", { style: d(4) })
  ) : null;
}
var T = function() {
  return T = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, T.apply(this, arguments);
}, Me = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Se = _("ClimbingBoxLoader", `0% {transform:translate(0, -1em) rotate(-45deg)}
  5% {transform:translate(0, -1em) rotate(-50deg)}
  20% {transform:translate(1em, -2em) rotate(47deg)}
  25% {transform:translate(1em, -2em) rotate(45deg)}
  30% {transform:translate(1em, -2em) rotate(40deg)}
  45% {transform:translate(2em, -3em) rotate(137deg)}
  50% {transform:translate(2em, -3em) rotate(135deg)}
  55% {transform:translate(2em, -3em) rotate(130deg)}
  70% {transform:translate(3em, -4em) rotate(217deg)}
  75% {transform:translate(3em, -4em) rotate(220deg)}
  100% {transform:translate(0, -1em) rotate(-225deg)}`, "climbingBox");
function ze(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = Me(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = T({ display: "inherit", position: "relative", width: "7.1em", height: "7.1em" }, h), d = {
    position: "absolute",
    top: "50%",
    left: "50%",
    marginTop: "-2.7em",
    marginLeft: "-2.7em",
    width: "5.4em",
    height: "5.4em",
    fontSize: u(i)
  }, f = {
    position: "absolute",
    left: "0",
    bottom: "-0.1em",
    height: "1em",
    width: "1em",
    backgroundColor: "transparent",
    borderRadius: "15%",
    border: "0.25em solid ".concat(r),
    transform: "translate(0, -1em) rotate(-45deg)",
    animationFillMode: "both",
    animation: "".concat(Se, " ").concat(2.5 / g, "s infinite cubic-bezier(0.79, 0, 0.47, 0.97)")
  }, p = {
    position: "absolute",
    width: "7.1em",
    height: "7.1em",
    top: "1.7em",
    left: "1.7em",
    borderLeft: "0.25em solid ".concat(r),
    transform: "rotate(45deg)"
  };
  return a ? o.createElement(
    "span",
    T({ style: y }, m),
    o.createElement(
      "span",
      { style: d },
      o.createElement("span", { style: f }),
      o.createElement("span", { style: p })
    )
  ) : null;
}
var q = function() {
  return q = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, q.apply(this, arguments);
}, Le = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Fe = _("ClipLoader", "0% {transform: rotate(0deg) scale(1)} 50% {transform: rotate(180deg) scale(0.8)} 100% {transform: rotate(360deg) scale(1)}", "clip");
function ke(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 35 : s, m = Le(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = q({ background: "transparent !important", width: u(i), height: u(i), borderRadius: "100%", border: "2px solid", borderTopColor: r, borderBottomColor: "transparent", borderLeftColor: r, borderRightColor: r, display: "inline-block", animation: "".concat(Fe, " ").concat(0.75 / g, "s 0s infinite linear"), animationFillMode: "both" }, h);
  return a ? o.createElement("span", q({ style: y }, m)) : null;
}
var D = function() {
  return D = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, D.apply(this, arguments);
}, $e = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, re = _("ClockLoader", "100% { transform: rotate(360deg) }", "rotate");
function Re(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 50 : s, m = $e(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = M(i), d = y.value, f = y.unit, p = D({ display: "inherit", position: "relative", width: "".concat(d).concat(f), height: "".concat(d).concat(f), backgroundColor: "transparent", boxShadow: "inset 0px 0px 0px 2px ".concat(r), borderRadius: "50%" }, h), v = {
    position: "absolute",
    backgroundColor: r,
    width: "".concat(d / 3, "px"),
    height: "2px",
    top: "".concat(d / 2 - 1, "px"),
    left: "".concat(d / 2 - 1, "px"),
    transformOrigin: "1px 1px",
    animation: "".concat(re, " ").concat(8 / g, "s linear infinite")
  }, b = {
    position: "absolute",
    backgroundColor: r,
    width: "".concat(d / 2.4, "px"),
    height: "2px",
    top: "".concat(d / 2 - 1, "px"),
    left: "".concat(d / 2 - 1, "px"),
    transformOrigin: "1px 1px",
    animation: "".concat(re, " ").concat(2 / g, "s linear infinite")
  };
  return a ? o.createElement(
    "span",
    D({ style: p }, m),
    o.createElement("span", { style: b }),
    o.createElement("span", { style: v })
  ) : null;
}
var V = function() {
  return V = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, V.apply(this, arguments);
}, Xe = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Ce = _("DotLoader", "100% {transform: rotate(360deg)}", "rotate"), Ie = _("DotLoader", "0%, 100% {transform: scale(0)} 50% {transform: scale(1.0)}", "bounce");
function Ye(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 60 : s, m = Xe(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = V({ display: "inherit", position: "relative", width: u(i), height: u(i), animationFillMode: "forwards", animation: "".concat(Ce, " ").concat(2 / g, "s 0s infinite linear") }, h), d = function(f) {
    var p = M(i), v = p.value, b = p.unit;
    return {
      position: "absolute",
      top: f % 2 ? "0" : "auto",
      bottom: f % 2 ? "auto" : "0",
      height: "".concat(v / 2).concat(b),
      width: "".concat(v / 2).concat(b),
      backgroundColor: r,
      borderRadius: "100%",
      animationFillMode: "forwards",
      animation: "".concat(Ie, " ").concat(2 / g, "s ").concat(f === 2 ? "1s" : "0s", " infinite linear")
    };
  };
  return a ? o.createElement(
    "span",
    V({ style: y }, m),
    o.createElement("span", { style: d(1) }),
    o.createElement("span", { style: d(2) })
  ) : null;
}
var j = function() {
  return j = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, j.apply(this, arguments);
}, Te = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, qe = _("FadeLoader", "50% {opacity: 0.3} 100% {opacity: 1}", "fade");
function De(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.height, i = s === void 0 ? 15 : s, m = e.width, y = m === void 0 ? 5 : m, d = e.radius, f = d === void 0 ? 2 : d, p = e.margin, v = p === void 0 ? 2 : p, b = Te(e, ["loading", "color", "speedMultiplier", "cssOverride", "height", "width", "radius", "margin"]), w = M(v).value, O = w + 18, x = O / 2 + O / 5.5, z = j({ display: "inherit", position: "relative", fontSize: "0", top: O, left: O, width: "".concat(O * 3, "px"), height: "".concat(O * 3, "px") }, h), P = function(pe) {
    return {
      position: "absolute",
      width: u(y),
      height: u(i),
      margin: u(v),
      backgroundColor: r,
      borderRadius: u(f),
      transition: "2s",
      animationFillMode: "both",
      animation: "".concat(qe, " ").concat(1.2 / g, "s ").concat(pe * 0.12, "s infinite ease-in-out")
    };
  }, $ = j(j({}, P(1)), { top: "".concat(O, "px"), left: "0" }), te = j(j({}, P(2)), { top: "".concat(x, "px"), left: "".concat(x, "px"), transform: "rotate(-45deg)" }), k = j(j({}, P(3)), { top: "0", left: "".concat(O, "px"), transform: "rotate(90deg)" }), ie = j(j({}, P(4)), { top: "".concat(-1 * x, "px"), left: "".concat(x, "px"), transform: "rotate(45deg)" }), ce = j(j({}, P(5)), { top: "".concat(-1 * O, "px"), left: "0" }), le = j(j({}, P(6)), { top: "".concat(-1 * x, "px"), left: "".concat(-1 * x, "px"), transform: "rotate(-45deg)" }), se = j(j({}, P(7)), { top: "0", left: "".concat(-1 * O, "px"), transform: "rotate(90deg)" }), de = j(j({}, P(8)), { top: "".concat(x, "px"), left: "".concat(-1 * x, "px"), transform: "rotate(45deg)" });
  return a ? o.createElement(
    "span",
    j({ style: z }, b),
    o.createElement("span", { style: $ }),
    o.createElement("span", { style: te }),
    o.createElement("span", { style: k }),
    o.createElement("span", { style: ie }),
    o.createElement("span", { style: ce }),
    o.createElement("span", { style: le }),
    o.createElement("span", { style: se }),
    o.createElement("span", { style: de })
  ) : null;
}
var W = function() {
  return W = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, W.apply(this, arguments);
}, Ve = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, We = _("GridLoader", "0% {transform: scale(1)} 50% {transform: scale(0.5); opacity: 0.7} 100% {transform: scale(1); opacity: 1}", "grid"), L = function(e) {
  return Math.random() * e;
};
function Ue(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = Ve(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = M(i), p = M(y), v = parseFloat(f.value.toString()) * 3 + parseFloat(p.value.toString()) * 6, b = W({ width: "".concat(v).concat(f.unit), fontSize: 0, display: "inline-block" }, h), w = function(O) {
    return {
      display: "inline-block",
      backgroundColor: r,
      width: "".concat(u(i)),
      height: "".concat(u(i)),
      margin: u(y),
      borderRadius: "100%",
      animationFillMode: "both",
      animation: "".concat(We, " ").concat((O / 100 + 0.6) / g, "s ").concat(O / 100 - 0.2, "s infinite ease")
    };
  };
  return a ? o.createElement(
    "span",
    W({ style: b }, d, { ref: function(O) {
      O && O.style.setProperty("width", "".concat(v).concat(f.unit), "important");
    } }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) }),
    o.createElement("span", { style: w(L(100)) })
  ) : null;
}
var U = function() {
  return U = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, U.apply(this, arguments);
}, Ze = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
};
function Ae(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 50 : s, m = Ze(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = M(i), d = y.value, f = y.unit, p = U({ display: "inherit", position: "relative", width: u(i), height: u(i), transform: "rotate(165deg)" }, h), v = d / 5, b = (d - v) / 2, w = b - v, O = oe(r, 0.75), x = _("HashLoader", "0% {width: ".concat(v, "px; box-shadow: ").concat(b, "px ").concat(-w, "px ").concat(O, ", ").concat(-b, "px ").concat(w, "px ").concat(O, `}
    35% {width: `).concat(u(i), "; box-shadow: 0 ").concat(-w, "px ").concat(O, ", 0 ").concat(w, "px ").concat(O, `}
    70% {width: `).concat(v, "px; box-shadow: ").concat(-b, "px ").concat(-w, "px ").concat(O, ", ").concat(b, "px ").concat(w, "px ").concat(O, `}
    100% {box-shadow: `).concat(b, "px ").concat(-w, "px ").concat(O, ", ").concat(-b, "px ").concat(w, "px ").concat(O, "}"), "before"), z = _("HashLoader", "0% {height: ".concat(v, "px; box-shadow: ").concat(w, "px ").concat(b, "px ").concat(r, ", ").concat(-w, "px ").concat(-b, "px ").concat(r, `}
    35% {height: `).concat(u(i), "; box-shadow: ").concat(w, "px 0 ").concat(r, ", ").concat(-w, "px 0 ").concat(r, `}
    70% {height: `).concat(v, "px; box-shadow: ").concat(w, "px ").concat(-b, "px ").concat(r, ", ").concat(-w, "px ").concat(b, "px ").concat(r, `}
    100% {box-shadow: `).concat(w, "px ").concat(b, "px ").concat(r, ", ").concat(-w, "px ").concat(-b, "px ").concat(r, "}"), "after"), P = function($) {
    return {
      position: "absolute",
      top: "50%",
      left: "50%",
      display: "block",
      width: "".concat(d / 5).concat(f),
      height: "".concat(d / 5).concat(f),
      borderRadius: "".concat(d / 10).concat(f),
      transform: "translate(-50%, -50%)",
      animationFillMode: "none",
      animation: "".concat($ === 1 ? x : z, " ").concat(2 / g, "s infinite")
    };
  };
  return a ? o.createElement(
    "span",
    U({ style: p }, m),
    o.createElement("span", { style: P(1) }),
    o.createElement("span", { style: P(2) })
  ) : null;
}
var F = function() {
  return F = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, F.apply(this, arguments);
}, He = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ne = _("MoonLoader", "100% {transform: rotate(360deg)}", "moon");
function Ge(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 60 : s, m = He(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = M(i), d = y.value, f = y.unit, p = Math.round(d / 7), v = F({ display: "inherit", position: "relative", width: "".concat("".concat(d + p * 2).concat(f)), height: "".concat("".concat(d + p * 2).concat(f)), animation: "".concat(ne, " ").concat(0.6 / g, "s 0s infinite linear"), animationFillMode: "forwards" }, h), b = function(x) {
    return {
      width: u(x),
      height: u(x),
      borderRadius: "100%"
    };
  }, w = F(F({}, b(p)), { backgroundColor: "".concat(r), opacity: "0.8", position: "absolute", top: "".concat("".concat(d / 2 - p / 2).concat(f)), animation: "".concat(ne, " ").concat(0.6 / g, "s 0s infinite linear"), animationFillMode: "forwards" }), O = F(F({}, b(d)), { border: "".concat(p, "px solid ").concat(r), opacity: "0.1", boxSizing: "content-box", position: "absolute" });
  return a ? o.createElement(
    "span",
    F({ style: v }, m),
    o.createElement("span", { style: w }),
    o.createElement("span", { style: O })
  ) : null;
}
var Z = function() {
  return Z = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, Z.apply(this, arguments);
}, Be = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Ne = [
  _("PacmanLoader", "0% {transform: rotate(0deg)} 50% {transform: rotate(-44deg)}", "pacman-1"),
  _("PacmanLoader", "0% {transform: rotate(0deg)} 50% {transform: rotate(44deg)}", "pacman-2")
];
function Je(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 25 : s, m = e.margin, y = m === void 0 ? 2 : m, d = Be(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = M(i), p = f.value, v = f.unit, b = Z({ display: "inherit", position: "relative", fontSize: 0, height: "".concat(p * 2).concat(v), width: "".concat(p * 2).concat(v) }, h), w = _("PacmanLoader", `75% {opacity: 0.7}
    100% {transform: translate(`.concat("".concat(-4 * p).concat(v), ", ").concat("".concat(-p / 4).concat(v), ")}"), "ball"), O = function(k) {
    return {
      width: "".concat(p / 3).concat(v),
      height: "".concat(p / 3).concat(v),
      backgroundColor: r,
      margin: u(y),
      borderRadius: "100%",
      transform: "translate(0, ".concat("".concat(-p / 4).concat(v), ")"),
      position: "absolute",
      top: "".concat(p).concat(v),
      left: "".concat(p * 4).concat(v),
      animation: "".concat(w, " ").concat(1 / g, "s ").concat(k * 0.25, "s infinite linear"),
      animationFillMode: "both"
    };
  }, x = "".concat(u(i), " solid transparent"), z = "".concat(u(i), " solid ").concat(r), P = function(k) {
    return {
      width: 0,
      height: 0,
      borderRight: x,
      borderTop: k === 0 ? x : z,
      borderLeft: z,
      borderBottom: k === 0 ? z : x,
      borderRadius: u(i),
      position: "absolute",
      animation: "".concat(Ne[k], " ").concat(0.8 / g, "s infinite ease-in-out"),
      animationFillMode: "both"
    };
  }, $ = P(0), te = P(1);
  return a ? o.createElement(
    "span",
    Z({ style: b }, d),
    o.createElement("span", { style: $ }),
    o.createElement("span", { style: te }),
    o.createElement("span", { style: O(2) }),
    o.createElement("span", { style: O(3) }),
    o.createElement("span", { style: O(4) }),
    o.createElement("span", { style: O(5) })
  ) : null;
}
var A = function() {
  return A = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, A.apply(this, arguments);
}, Ke = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, E = [1, 3, 5], Qe = [
  _("PropagateLoader", "25% {transform: translateX(-".concat(E[0], `rem) scale(0.75)}
    50% {transform: translateX(-`).concat(E[1], `rem) scale(0.6)}
    75% {transform: translateX(-`).concat(E[2], `rem) scale(0.5)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-0"),
  _("PropagateLoader", "25% {transform: translateX(-".concat(E[0], `rem) scale(0.75)}
    50% {transform: translateX(-`).concat(E[1], `rem) scale(0.6)}
    75% {transform: translateX(-`).concat(E[1], `rem) scale(0.6)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-1"),
  _("PropagateLoader", "25% {transform: translateX(-".concat(E[0], `rem) scale(0.75)}
    75% {transform: translateX(-`).concat(E[0], `rem) scale(0.75)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-2"),
  _("PropagateLoader", "25% {transform: translateX(".concat(E[0], `rem) scale(0.75)}
    75% {transform: translateX(`).concat(E[0], `rem) scale(0.75)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-3"),
  _("PropagateLoader", "25% {transform: translateX(".concat(E[0], `rem) scale(0.75)}
    50% {transform: translateX(`).concat(E[1], `rem) scale(0.6)}
    75% {transform: translateX(`).concat(E[1], `rem) scale(0.6)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-4"),
  _("PropagateLoader", "25% {transform: translateX(".concat(E[0], `rem) scale(0.75)}
    50% {transform: translateX(`).concat(E[1], `rem) scale(0.6)}
    75% {transform: translateX(`).concat(E[2], `rem) scale(0.5)}
    95% {transform: translateX(0rem) scale(1)}`), "propogate-5")
];
function et(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = Ke(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = M(i), d = y.value, f = y.unit, p = A({ display: "inherit", position: "relative" }, h), v = function(b) {
    return {
      position: "absolute",
      fontSize: "".concat(d / 3).concat(f),
      width: "".concat(d).concat(f),
      height: "".concat(d).concat(f),
      background: r,
      borderRadius: "50%",
      animation: "".concat(Qe[b], " ").concat(1.5 / g, "s infinite"),
      animationFillMode: "forwards"
    };
  };
  return a ? o.createElement(
    "span",
    A({ style: p }, m),
    o.createElement("span", { style: v(0) }),
    o.createElement("span", { style: v(1) }),
    o.createElement("span", { style: v(2) }),
    o.createElement("span", { style: v(3) }),
    o.createElement("span", { style: v(4) }),
    o.createElement("span", { style: v(5) })
  ) : null;
}
var H = function() {
  return H = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, H.apply(this, arguments);
}, tt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, rt = _("PulseLoader", "0% {transform: scale(1); opacity: 1} 45% {transform: scale(0.1); opacity: 0.7} 80% {transform: scale(1); opacity: 1}", "pulse");
function nt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = tt(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = H({ display: "inherit" }, h), p = function(v) {
    return {
      backgroundColor: r,
      width: u(i),
      height: u(i),
      margin: u(y),
      borderRadius: "100%",
      display: "inline-block",
      animation: "".concat(rt, " ").concat(0.75 / g, "s ").concat(v * 0.12 / g, "s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08)"),
      animationFillMode: "both"
    };
  };
  return a ? o.createElement(
    "span",
    H({ style: f }, d),
    o.createElement("span", { style: p(1) }),
    o.createElement("span", { style: p(2) }),
    o.createElement("span", { style: p(3) })
  ) : null;
}
var G = function() {
  return G = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, G.apply(this, arguments);
}, at = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ae = [
  _("PuffLoader", "0% {transform: scale(0)} 100% {transform: scale(1.0)}", "puff-1"),
  _("PuffLoader", "0% {opacity: 1} 100% {opacity: 0}", "puff-2")
];
function ot(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 60 : s, m = at(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = G({ display: "inherit", position: "relative", width: u(i), height: u(i) }, h), d = function(f) {
    return {
      position: "absolute",
      height: u(i),
      width: u(i),
      border: "thick solid ".concat(r),
      borderRadius: "50%",
      opacity: "1",
      top: "0",
      left: "0",
      animationFillMode: "both",
      animation: "".concat(ae[0], ", ").concat(ae[1]),
      animationDuration: "".concat(2 / g, "s"),
      animationIterationCount: "infinite",
      animationTimingFunction: "cubic-bezier(0.165, 0.84, 0.44, 1), cubic-bezier(0.3, 0.61, 0.355, 1)",
      animationDelay: f === 1 ? "-1s" : "0s"
    };
  };
  return a ? o.createElement(
    "span",
    G({ style: y }, m),
    o.createElement("span", { style: d(1) }),
    o.createElement("span", { style: d(2) })
  ) : null;
}
var B = function() {
  return B = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, B.apply(this, arguments);
}, it = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ct = _("RingLoader", "0% {transform: rotateX(0deg) rotateY(0deg) rotateZ(0deg)} 100% {transform: rotateX(180deg) rotateY(360deg) rotateZ(360deg)}", "right"), lt = _("RingLoader", "0% {transform: rotateX(0deg) rotateY(0deg) rotateZ(0deg)} 100% {transform: rotateX(360deg) rotateY(180deg) rotateZ(360deg)}", "left");
function st(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 60 : s, m = it(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = M(i), d = y.value, f = y.unit, p = B({ display: "inherit", width: u(i), height: u(i), position: "relative" }, h), v = function(b) {
    return {
      position: "absolute",
      top: "0",
      left: "0",
      width: "".concat(d).concat(f),
      height: "".concat(d).concat(f),
      border: "".concat(d / 10).concat(f, " solid ").concat(r),
      opacity: "0.4",
      borderRadius: "100%",
      animationFillMode: "forwards",
      perspective: "800px",
      animation: "".concat(b === 1 ? ct : lt, " ").concat(2 / g, "s 0s infinite linear")
    };
  };
  return a ? o.createElement(
    "span",
    B({ style: p }, m),
    o.createElement("span", { style: v(1) }),
    o.createElement("span", { style: v(2) })
  ) : null;
}
var N = function() {
  return N = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, N.apply(this, arguments);
}, dt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
};
function pt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = dt(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = N({ display: "inherit" }, h), p = _("RiseLoader", `0% {transform: scale(1.1)}
    25% {transform: translateY(-`.concat(i, `px)}
    50% {transform: scale(0.4)}
    75% {transform: translateY(`).concat(i, `px)}
    100% {transform: translateY(0) scale(1.0)}`), "even"), v = _("RiseLoader", `0% {transform: scale(0.4)}
    25% {transform: translateY(`.concat(i, `px)}
    50% {transform: scale(1.1)}
    75% {transform: translateY(`).concat(-i, `px)}
    100% {transform: translateY(0) scale(0.75)}`), "odd"), b = function(w) {
    return {
      backgroundColor: r,
      width: u(i),
      height: u(i),
      margin: u(y),
      borderRadius: "100%",
      display: "inline-block",
      animation: "".concat(w % 2 === 0 ? p : v, " ").concat(1 / g, "s 0s infinite cubic-bezier(0.15, 0.46, 0.9, 0.6)"),
      animationFillMode: "both"
    };
  };
  return a ? o.createElement(
    "span",
    N({ style: f }, d),
    o.createElement("span", { style: b(1) }),
    o.createElement("span", { style: b(2) }),
    o.createElement("span", { style: b(3) }),
    o.createElement("span", { style: b(4) }),
    o.createElement("span", { style: b(5) })
  ) : null;
}
var S = function() {
  return S = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, S.apply(this, arguments);
}, ut = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ft = _("RotateLoader", "0% {transform: rotate(0deg)} 50% {transform: rotate(180deg)} 100% {transform: rotate(360deg)}", "rotate");
function vt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = ut(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = M(y), p = f.value, v = f.unit, b = {
    backgroundColor: r,
    width: u(i),
    height: u(i),
    borderRadius: "100%"
  }, w = S(S(S({}, b), { display: "inline-block", position: "relative", animationFillMode: "both", animation: "".concat(ft, " ").concat(1 / g, "s 0s infinite cubic-bezier(0.7, -0.13, 0.22, 0.86)") }), h), O = function(x) {
    var z = (x % 2 ? -1 : 1) * (26 + p);
    return {
      opacity: "0.8",
      position: "absolute",
      top: "0",
      left: "".concat(z).concat(v)
    };
  };
  return a ? o.createElement(
    "span",
    S({ style: w }, d),
    o.createElement("span", { style: S(S({}, b), O(1)) }),
    o.createElement("span", { style: S(S({}, b), O(2)) })
  ) : null;
}
var J = function() {
  return J = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, J.apply(this, arguments);
}, gt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, mt = _("ScaleLoader", "0% {transform: scaley(1.0)} 50% {transform: scaley(0.4)} 100% {transform: scaley(1.0)}", "scale");
function yt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.height, i = s === void 0 ? 35 : s, m = e.width, y = m === void 0 ? 4 : m, d = e.radius, f = d === void 0 ? 2 : d, p = e.margin, v = p === void 0 ? 2 : p, b = gt(e, ["loading", "color", "speedMultiplier", "cssOverride", "height", "width", "radius", "margin"]), w = J({ display: "inherit" }, h), O = function(x) {
    return {
      backgroundColor: r,
      width: u(y),
      height: u(i),
      margin: u(v),
      borderRadius: u(f),
      display: "inline-block",
      animation: "".concat(mt, " ").concat(1 / g, "s ").concat(x * 0.1, "s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08)"),
      animationFillMode: "both"
    };
  };
  return a ? o.createElement(
    "span",
    J({ style: w }, b),
    o.createElement("span", { style: O(1) }),
    o.createElement("span", { style: O(2) }),
    o.createElement("span", { style: O(3) }),
    o.createElement("span", { style: O(4) }),
    o.createElement("span", { style: O(5) })
  ) : null;
}
var K = function() {
  return K = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, K.apply(this, arguments);
}, bt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, ht = _("SkewLoader", "25% {transform: perspective(100px) rotateX(180deg) rotateY(0)} 50% {transform: perspective(100px) rotateX(180deg) rotateY(180deg)} 75% {transform: perspective(100px) rotateX(0) rotateY(180deg)} 100% {transform: perspective(100px) rotateX(0) rotateY(0)}", "skew");
function Ot(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 20 : s, m = bt(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = K({ width: "0", height: "0", borderLeft: "".concat(u(i), " solid transparent"), borderRight: "".concat(u(i), " solid transparent"), borderBottom: "".concat(u(i), " solid ").concat(r), display: "inline-block", animation: "".concat(ht, " ").concat(3 / g, "s 0s infinite cubic-bezier(0.09, 0.57, 0.49, 0.9)"), animationFillMode: "both" }, h);
  return a ? o.createElement("span", K({ style: y }, m)) : null;
}
var Q = function() {
  return Q = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, Q.apply(this, arguments);
}, wt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, _t = _("SquareLoader", `25% {transform: rotateX(180deg) rotateY(0)}
  50% {transform: rotateX(180deg) rotateY(180deg)} 
  75% {transform: rotateX(0) rotateY(180deg)} 
  100% {transform: rotateX(0) rotateY(0)}`, "square");
function xt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 50 : s, m = wt(e, ["loading", "color", "speedMultiplier", "cssOverride", "size"]), y = Q({ backgroundColor: r, width: u(i), height: u(i), display: "inline-block", animation: "".concat(_t, " ").concat(3 / g, "s 0s infinite cubic-bezier(0.09, 0.57, 0.49, 0.9)"), animationFillMode: "both" }, h);
  return a ? o.createElement("span", Q({ style: y }, m)) : null;
}
var ee = function() {
  return ee = Object.assign || function(e) {
    for (var n, a = 1, t = arguments.length; a < t; a++) {
      n = arguments[a];
      for (var r in n) Object.prototype.hasOwnProperty.call(n, r) && (e[r] = n[r]);
    }
    return e;
  }, ee.apply(this, arguments);
}, jt = function(e, n) {
  var a = {};
  for (var t in e) Object.prototype.hasOwnProperty.call(e, t) && n.indexOf(t) < 0 && (a[t] = e[t]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var r = 0, t = Object.getOwnPropertySymbols(e); r < t.length; r++)
      n.indexOf(t[r]) < 0 && Object.prototype.propertyIsEnumerable.call(e, t[r]) && (a[t[r]] = e[t[r]]);
  return a;
}, Et = _("SyncLoader", `33% {transform: translateY(10px)}
  66% {transform: translateY(-10px)}
  100% {transform: translateY(0)}`, "sync");
function Pt(e) {
  var n = e.loading, a = n === void 0 ? !0 : n, t = e.color, r = t === void 0 ? "#000000" : t, c = e.speedMultiplier, g = c === void 0 ? 1 : c, l = e.cssOverride, h = l === void 0 ? {} : l, s = e.size, i = s === void 0 ? 15 : s, m = e.margin, y = m === void 0 ? 2 : m, d = jt(e, ["loading", "color", "speedMultiplier", "cssOverride", "size", "margin"]), f = ee({ display: "inherit" }, h), p = function(v) {
    return {
      backgroundColor: r,
      width: u(i),
      height: u(i),
      margin: u(y),
      borderRadius: "100%",
      display: "inline-block",
      animation: "".concat(Et, " ").concat(0.6 / g, "s ").concat(v * 0.07, "s infinite ease-in-out"),
      animationFillMode: "both"
    };
  };
  return a ? o.createElement(
    "span",
    ee({ style: f }, d),
    o.createElement("span", { style: p(1) }),
    o.createElement("span", { style: p(2) }),
    o.createElement("span", { style: p(3) })
  ) : null;
}
const Mt = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  BarLoader: ye,
  BeatLoader: Oe,
  BounceLoader: xe,
  CircleLoader: Pe,
  ClimbingBoxLoader: ze,
  ClipLoader: ke,
  ClockLoader: Re,
  DotLoader: Ye,
  FadeLoader: De,
  GridLoader: Ue,
  HashLoader: Ae,
  MoonLoader: Ge,
  PacmanLoader: Je,
  PropagateLoader: et,
  PuffLoader: ot,
  PulseLoader: nt,
  RingLoader: st,
  RiseLoader: pt,
  RotateLoader: vt,
  ScaleLoader: yt,
  SkewLoader: Ot,
  SquareLoader: xt,
  SyncLoader: Pt
}, Symbol.toStringTag, { value: "Module" }));
export {
  Mt as r
};
